// @spartacus/storefront/esm2022/shared/components/media/media.component.mjs
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { ChangeDetectionStrategy, Component, EventEmitter, HostBinding, inject, Input, Output, } from '@angular/core';
import { Config } from '@spartacus/core';
import { ImageLoadingStrategy } from './media.model';
import { USE_LEGACY_MEDIA_COMPONENT } from './media.token';
import * as i0 from "@angular/core";
import * as i1 from "./media.service";
import * as i2 from "@angular/common";
import * as i3 from "./media-sources.pipe";
export class MediaComponent {
    constructor(mediaService) {
        this.mediaService = mediaService;
        /**
         * Set the loading strategy of the media. Defaults to global loading strategy.
         * Use 'lazy' or 'eager' strategies.
         */
        this.loading = this.loadingStrategy;
        /**
         * Once the media is loaded, we emit an event.
         */
        this.loaded = new EventEmitter();
        /**
         * The `cx-media` component has an `is-initialized` class as long as the
         * media is being initialized.
         */
        this.isInitialized = false;
        /**
         * The `cx-media` component has a `is-loading` class as long as the
         * media is loaded. Wehn the media is loaded, the `is-initialized` class
         * is added.
         */
        this.isLoading = true;
        /**
         * When there's no media provided for the content, or in case an error
         * happened during loading, we add the `is-missing` class. Visual effects
         * can be controlled by CSS.
         */
        this.isMissing = false;
        this.trackByMedia = (_, item) => item.media;
        this.isLegacy = inject(USE_LEGACY_MEDIA_COMPONENT, { optional: true }) ||
            inject(Config)['useLegacyMediaComponent'] ||
            false;
    }
    ngOnChanges() {
        this.create();
    }
    /**
     * Creates the `Media` object
     */
    create() {
        this.media = this.mediaService.getMedia(this.container instanceof Array ? this.container[0] : this.container, this.format, this.alt, this.role);
        if (!this.media?.src) {
            this.handleMissing();
        }
    }
    /**
     * This handler is called from the UI when the image is loaded.
     */
    loadHandler() {
        this.isLoading = false;
        this.isInitialized = true;
        this.isMissing = false;
        this.loaded.emit(true);
    }
    /**
     * Indicates whether the browser should lazy load the image.
     */
    get loadingStrategy() {
        return this.mediaService.loadingStrategy === ImageLoadingStrategy.LAZY
            ? ImageLoadingStrategy.LAZY
            : null;
    }
    /**
     * Whenever an error happens during load, we mark the component
     * with css classes to have a missing media.
     */
    errorHandler() {
        this.handleMissing();
    }
    handleMissing() {
        this.isLoading = false;
        this.isInitialized = true;
        this.isMissing = true;
        this.loaded.emit(false);
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: MediaComponent, deps: [{ token: i1.MediaService }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "17.0.5", type: MediaComponent, selector: "cx-media", inputs: { container: "container", format: "format", alt: "alt", role: "role", loading: "loading" }, outputs: { loaded: "loaded" }, host: { properties: { "class.is-initialized": "this.isInitialized", "class.is-loading": "this.isLoading", "class.is-missing": "this.isMissing" } }, usesOnChanges: true, ngImport: i0, template: "<picture *ngIf=\"media?.srcset && !isLegacy; else tmpImg\">\n  <source\n    *ngFor=\"\n      let source of media!.srcset! | cxMediaSources;\n      trackBy: trackByMedia\n    \"\n    [srcset]=\"source.srcset\"\n    [media]=\"source.media\"\n  />\n\n  <ng-container [ngTemplateOutlet]=\"tmpImg\"></ng-container>\n</picture>\n\n<ng-template #tmpImg>\n  <img\n    *ngIf=\"media && media.src\"\n    [alt]=\"media.alt\"\n    [src]=\"media.src\"\n    [srcset]=\"media.srcset || media.src\"\n    [attr.role]=\"media.role\"\n    [loading]=\"loading\"\n    (load)=\"loadHandler()\"\n    (error)=\"errorHandler()\"\n  />\n</ng-template>\n", dependencies: [{ kind: "directive", type: i2.NgForOf, selector: "[ngFor][ngForOf]", inputs: ["ngForOf", "ngForTrackBy", "ngForTemplate"] }, { kind: "directive", type: i2.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { kind: "directive", type: i2.NgTemplateOutlet, selector: "[ngTemplateOutlet]", inputs: ["ngTemplateOutletContext", "ngTemplateOutlet", "ngTemplateOutletInjector"] }, { kind: "pipe", type: i3.MediaSourcesPipe, name: "cxMediaSources" }], changeDetection: i0.ChangeDetectionStrategy.OnPush }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: MediaComponent, decorators: [{
            type: Component,
            args: [{ selector: 'cx-media', changeDetection: ChangeDetectionStrategy.OnPush, template: "<picture *ngIf=\"media?.srcset && !isLegacy; else tmpImg\">\n  <source\n    *ngFor=\"\n      let source of media!.srcset! | cxMediaSources;\n      trackBy: trackByMedia\n    \"\n    [srcset]=\"source.srcset\"\n    [media]=\"source.media\"\n  />\n\n  <ng-container [ngTemplateOutlet]=\"tmpImg\"></ng-container>\n</picture>\n\n<ng-template #tmpImg>\n  <img\n    *ngIf=\"media && media.src\"\n    [alt]=\"media.alt\"\n    [src]=\"media.src\"\n    [srcset]=\"media.srcset || media.src\"\n    [attr.role]=\"media.role\"\n    [loading]=\"loading\"\n    (load)=\"loadHandler()\"\n    (error)=\"errorHandler()\"\n  />\n</ng-template>\n" }]
        }], ctorParameters: () => [{ type: i1.MediaService }], propDecorators: { container: [{
                type: Input
            }], format: [{
                type: Input
            }], alt: [{
                type: Input
            }], role: [{
                type: Input
            }], loading: [{
                type: Input
            }], loaded: [{
                type: Output
            }], isInitialized: [{
                type: HostBinding,
                args: ['class.is-initialized']
            }], isLoading: [{
                type: HostBinding,
                args: ['class.is-loading']
            }], isMissing: [{
                type: HostBinding,
                args: ['class.is-missing']
            }] } });
