// @spartacus/storefront/esm2022/cms-structure/services/cms-guards.service.mjs
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { Injectable } from '@angular/core';
import { getLastValueSync, } from '@spartacus/core';
import { concat, from, isObservable, of } from 'rxjs';
import { endWith, first, skipWhile } from 'rxjs/operators';
import * as i0 from "@angular/core";
import * as i1 from "./cms-components.service";
import * as i2 from "@spartacus/core";
export class CmsGuardsService {
    constructor(cmsComponentsService, 
    // TODO:#checkout - handle breaking changes in schematics
    unifiedInjector) {
        this.cmsComponentsService = cmsComponentsService;
        this.unifiedInjector = unifiedInjector;
    }
    cmsPageCanActivate(componentTypes, route, state) {
        const guards = this.cmsComponentsService.getGuards(componentTypes);
        if (guards.length) {
            const canActivateObservables = guards.map((guard) => this.canActivateGuard(guard, route, state));
            return concat(...canActivateObservables).pipe(skipWhile((canActivate) => canActivate === true), endWith(true), first());
        }
        else {
            return of(true);
        }
    }
    canActivateGuard(guardClass, route, state) {
        const guard = getLastValueSync(this.unifiedInjector.get(guardClass));
        if (isCanActivate(guard)) {
            return wrapIntoObservable(guard.canActivate(route, state)).pipe(first());
        }
        else {
            throw new Error('Invalid CanActivate guard in cmsMapping');
        }
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: CmsGuardsService, deps: [{ token: i1.CmsComponentsService }, { token: i2.UnifiedInjector }], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: CmsGuardsService, providedIn: 'root' }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: CmsGuardsService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }], ctorParameters: () => [{ type: i1.CmsComponentsService }, { type: i2.UnifiedInjector }] });
function wrapIntoObservable(value) {
    if (isObservable(value)) {
        return value;
    }
    if (isPromise(value)) {
        return from(Promise.resolve(value));
    }
    return of(value);
}
function isPromise(obj) {
    return !!obj && typeof obj.then === 'function';
}
function isCanActivate(guard) {
    return (guard &&
        isFunction(guard.canActivate));
}
function isFunction(v) {
    return typeof v === 'function';
}
