// @spartacus/user/esm2022/profile/components/address-book/address-book.module.mjs
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { AuthGuard, I18nModule, provideDefaultConfig, UserAddressService, } from '@spartacus/core';
import { AddressBookComponent } from './address-book.component';
import { AddressFormModule } from './address-form/address-form.module';
import { defaultSuggestedAddressesDialogLayoutConfig } from './address-form/suggested-addresses-dialog/default-suggested-addresses-dialog-layout.config';
import { CardModule, SpinnerModule } from '@spartacus/storefront';
import * as i0 from "@angular/core";
export class AddressBookModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: AddressBookModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "17.0.5", ngImport: i0, type: AddressBookModule, declarations: [AddressBookComponent], imports: [CommonModule,
            CardModule,
            AddressFormModule,
            SpinnerModule,
            I18nModule], exports: [AddressBookComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: AddressBookModule, providers: [
            provideDefaultConfig({
                cmsComponents: {
                    AccountAddressBookComponent: {
                        component: AddressBookComponent,
                        guards: [AuthGuard],
                    },
                },
            }),
            provideDefaultConfig(defaultSuggestedAddressesDialogLayoutConfig),
            UserAddressService,
        ], imports: [CommonModule,
            CardModule,
            AddressFormModule,
            SpinnerModule,
            I18nModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: AddressBookModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        CardModule,
                        AddressFormModule,
                        SpinnerModule,
                        I18nModule,
                    ],
                    declarations: [AddressBookComponent],
                    exports: [AddressBookComponent],
                    providers: [
                        provideDefaultConfig({
                            cmsComponents: {
                                AccountAddressBookComponent: {
                                    component: AddressBookComponent,
                                    guards: [AuthGuard],
                                },
                            },
                        }),
                        provideDefaultConfig(defaultSuggestedAddressesDialogLayoutConfig),
                        UserAddressService,
                    ],
                }]
        }] });
