// @spartacus/asm/esm2022/customer-360/occ/asm-customer-360-occ.module.mjs
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { AsmCustomer360Adapter } from '@spartacus/asm/customer-360/core';
import { provideDefaultConfig } from '@spartacus/core';
import { defaultOccAsmCustomer360Config } from './adapters/default-occ-asm-customer-360-config';
import { OccAsmCustomer360Adapter } from './adapters/occ-asm-customer-360.adapter';
import * as i0 from "@angular/core";
export class AsmCustomer360OccModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: AsmCustomer360OccModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "17.0.5", ngImport: i0, type: AsmCustomer360OccModule, imports: [CommonModule] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: AsmCustomer360OccModule, providers: [
            provideDefaultConfig(defaultOccAsmCustomer360Config),
            {
                provide: AsmCustomer360Adapter,
                useClass: OccAsmCustomer360Adapter,
            },
        ], imports: [CommonModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: AsmCustomer360OccModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    providers: [
                        provideDefaultConfig(defaultOccAsmCustomer360Config),
                        {
                            provide: AsmCustomer360Adapter,
                            useClass: OccAsmCustomer360Adapter,
                        },
                    ],
                }]
        }] });
