// @spartacus/checkout/esm2022/base/root/http-interceptors/checkout-cart.interceptor.mjs
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { HttpErrorResponse, } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { switchMap, take, tap } from 'rxjs/operators';
import * as i0 from "@angular/core";
import * as i1 from "@spartacus/core";
import * as i2 from "@spartacus/cart/base/root";
/**
 * Interceptor that handles "Cart not found" errors while a user is in a checkout step.
 *
 * When a user doing a checkout has a "Cart not found" error, he is redirected to checkout and the cart is reloaded.
 * If a "Cart not found" error happens and the user is not on checkout, this interceptor does not perform any actions.
 */
export class CheckoutCartInterceptor {
    constructor(routingService, multiCartFacade) {
        this.routingService = routingService;
        this.multiCartFacade = multiCartFacade;
    }
    intercept(request, next) {
        return this.routingService.getRouterState().pipe(take(1), switchMap((state) => {
            return next.handle(request).pipe(tap({
                error: (response) => {
                    if (response instanceof HttpErrorResponse &&
                        this.isUserInCheckoutRoute(state.state?.semanticRoute)) {
                        if (this.isCartNotFoundError(response)) {
                            this.routingService.go({ cxRoute: 'cart' });
                            const cartCode = this.getCartIdFromError(response);
                            if (cartCode) {
                                this.multiCartFacade.reloadCart(cartCode);
                            }
                        }
                    }
                },
            }));
        }));
    }
    /**
     * Returns true if the parameter semantic route is part of "checkout"
     * Checkout semantic routes:
     * checkout
     * checkoutPaymentType
     * CheckoutDeliveryAddress
     * checkoutDeliveryMode
     * checkoutPaymentDetails
     * checkoutReviewOrder
     * checkoutLogin
     * @param semanticRoute
     */
    isUserInCheckoutRoute(semanticRoute) {
        return semanticRoute?.toLowerCase().startsWith('checkout') ?? false;
    }
    /**
     * Checks of the error is for a cart not found, i.e. the cart doesn't exist anymore
     *
     * @param response
     */
    isCartNotFoundError(response) {
        return (response.status === 400 &&
            response.error?.errors?.[0]?.type === 'CartError' &&
            response.error?.errors?.[0]?.reason === 'notFound');
    }
    getCartIdFromError(response) {
        return response.error?.errors?.[0]?.subject;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: CheckoutCartInterceptor, deps: [{ token: i1.RoutingService }, { token: i2.MultiCartFacade }], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: CheckoutCartInterceptor, providedIn: 'root' }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: CheckoutCartInterceptor, decorators: [{
            type: Injectable,
            args: [{ providedIn: 'root' }]
        }], ctorParameters: () => [{ type: i1.RoutingService }, { type: i2.MultiCartFacade }] });
