// @spartacus/storefront/esm2022/router/on-navigate.service.mjs
import { ApplicationRef, Injectable, inject, } from '@angular/core';
import { ROUTER_CONFIGURATION, Scroll, } from '@angular/router';
import { filter, pairwise } from 'rxjs/operators';
import * as i0 from "@angular/core";
import * as i1 from "./config";
import * as i2 from "@angular/router";
import * as i3 from "@angular/common";
export class OnNavigateService {
    get hostComponent() {
        return this.injector.get(ApplicationRef)?.components?.[0];
    }
    constructor(config, router, viewportScroller, injector) {
        this.config = config;
        this.router = router;
        this.viewportScroller = viewportScroller;
        this.injector = injector;
        this.routerConfiguration = inject(ROUTER_CONFIGURATION, { optional: true }) || {};
    }
    /**
     * Reads configuration and enables features based on flags set.
     */
    initializeWithConfig() {
        if (this.config?.enableResetViewOnNavigate?.active) {
            this.setResetViewOnNavigate(this.config.enableResetViewOnNavigate.active);
        }
    }
    /**
     * Resets view back to the original position when performing a back navigation and to the top when performing a front navigation
     * and sets the focus back to the top of the page before skiplinks for any type of navigation
     * @param enable Enable or disable this feature
     */
    setResetViewOnNavigate(enable) {
        this.subscription?.unsubscribe();
        if (enable) {
            // Disable automatic scroll restoration to avoid race conditions
            this.viewportScroller.setHistoryScrollRestoration('manual');
            this.subscription = this.router.events
                .pipe(filter((event) => event instanceof Scroll), pairwise())
                .subscribe((event) => {
                const previousRoute = event[0];
                const currentRoute = event[1];
                const position = currentRoute.position;
                if (position) {
                    // allow the pages to be repainted before scrolling to proper position
                    this.scrollToPosition(currentRoute, position);
                }
                else {
                    if (this.config.enableResetViewOnNavigate?.ignoreQueryString &&
                        this.isPathEqual(previousRoute, currentRoute)) {
                        return;
                    }
                    if (this.isChildRoute(currentRoute)) {
                        return;
                    }
                    this.scrollToPosition(currentRoute, position);
                }
                this.hostComponent?.location?.nativeElement.focus();
            });
        }
    }
    /**
     * Scrolls to a specified position or anchor based on the current route and configuration.
     * @param currentRoute The current route containing scroll information.
     * @param position The target scroll position as [x, y] coordinates, or null.
     */
    scrollToPosition(currentRoute, position) {
        const scrollTo = (anchor, scrollPosition) => {
            if (anchor && this.routerConfiguration.anchorScrolling === 'enabled') {
                this.viewportScroller.scrollToAnchor(anchor);
            }
            else {
                this.viewportScroller.scrollToPosition(scrollPosition);
            }
        };
        const defaultPosition = [0, 0];
        setTimeout(() => {
            const anchor = currentRoute.anchor;
            const positionToScroll = position || defaultPosition;
            scrollTo(anchor, positionToScroll);
        }, position ? 0 : 100);
    }
    /**
     * Verifies if the current route is a child route from the given ignore config route
     *
     * @param route
     * @returns boolean whether the route is a child route
     */
    isChildRoute(route) {
        return (this.config.enableResetViewOnNavigate?.ignoreRoutes?.some((configRoute) => this.isNavigationEnd(route.routerEvent) &&
            route.routerEvent.urlAfterRedirects.split('/').includes(configRoute)) ?? false);
    }
    /**
     * Verifies if the previous and current route are the same without the query string
     *
     * @param previousRoute
     * @param currentRoute
     * @returns boolean depending on the previous and current route are equal without the query strings
     */
    isPathEqual(previousRoute, currentRoute) {
        if (!this.isNavigationEnd(previousRoute.routerEvent) ||
            !this.isNavigationEnd(currentRoute.routerEvent)) {
            return false;
        }
        return (previousRoute.routerEvent.urlAfterRedirects.split('?')[0] ===
            currentRoute.routerEvent.urlAfterRedirects.split('?')[0]);
    }
    /**
     * Verifies if router event is a navigation end event
     * @private
     */
    isNavigationEnd(event) {
        return event.type === 1 /* EventType.NavigationEnd */;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: OnNavigateService, deps: [{ token: i1.OnNavigateConfig }, { token: i2.Router }, { token: i3.ViewportScroller }, { token: i0.Injector }], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: OnNavigateService, providedIn: 'root' }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: OnNavigateService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }], ctorParameters: () => [{ type: i1.OnNavigateConfig }, { type: i2.Router }, { type: i3.ViewportScroller }, { type: i0.Injector }] });
