// @spartacus/storefront/esm2022/cms-components/myaccount/notification-preference/notification-preference.module.mjs
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { CommonModule } from '@angular/common';
import { NgModule, inject } from '@angular/core';
import { RouterModule } from '@angular/router';
import { AuthGuard, I18nModule, provideDefaultConfig, provideDefaultConfigFactory, } from '@spartacus/core';
import { CmsPageGuard } from '../../../cms-structure/guards/cms-page.guard';
import { PageLayoutComponent } from '../../../cms-structure/page/page-layout/page-layout.component';
import { SpinnerModule } from '../../../shared/components/spinner/spinner.module';
import { NotificationPreferenceComponent } from './notification-preference.component';
import { USE_MY_ACCOUNT_V2_NOTIFICATION_PREFERENCE } from '../my-account-v2/use-my-account-v2-consent-notification-perference';
import { MyAccountV2NotificationPreferenceComponent } from '../my-account-v2';
import * as i0 from "@angular/core";
import * as i1 from "@angular/router";
const myAccountV2CmsMapping = {
    cmsComponents: {
        NotificationPreferenceComponent: {
            component: MyAccountV2NotificationPreferenceComponent,
            //guards: inherited from standard config,
        },
    },
};
export class NotificationPreferenceModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: NotificationPreferenceModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "17.0.5", ngImport: i0, type: NotificationPreferenceModule, declarations: [NotificationPreferenceComponent], imports: [CommonModule,
            SpinnerModule,
            I18nModule, i1.RouterModule], exports: [NotificationPreferenceComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: NotificationPreferenceModule, providers: [
            provideDefaultConfig({
                cmsComponents: {
                    NotificationPreferenceComponent: {
                        component: NotificationPreferenceComponent,
                        guards: [AuthGuard],
                    },
                },
            }),
            provideDefaultConfigFactory(() => inject(USE_MY_ACCOUNT_V2_NOTIFICATION_PREFERENCE)
                ? myAccountV2CmsMapping
                : {}),
        ], imports: [CommonModule,
            SpinnerModule,
            I18nModule,
            RouterModule.forChild([
                {
                    // @ts-ignore
                    path: null,
                    canActivate: [AuthGuard, CmsPageGuard],
                    component: PageLayoutComponent,
                    data: { cxRoute: 'notificationPreference' },
                },
            ])] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: NotificationPreferenceModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [NotificationPreferenceComponent],
                    imports: [
                        CommonModule,
                        SpinnerModule,
                        I18nModule,
                        RouterModule.forChild([
                            {
                                // @ts-ignore
                                path: null,
                                canActivate: [AuthGuard, CmsPageGuard],
                                component: PageLayoutComponent,
                                data: { cxRoute: 'notificationPreference' },
                            },
                        ]),
                    ],
                    providers: [
                        provideDefaultConfig({
                            cmsComponents: {
                                NotificationPreferenceComponent: {
                                    component: NotificationPreferenceComponent,
                                    guards: [AuthGuard],
                                },
                            },
                        }),
                        provideDefaultConfigFactory(() => inject(USE_MY_ACCOUNT_V2_NOTIFICATION_PREFERENCE)
                            ? myAccountV2CmsMapping
                            : {}),
                    ],
                    exports: [NotificationPreferenceComponent],
                }]
        }] });
