// @spartacus/storefront/esm2022/cms-components/product/product-summary/product-summary.module.mjs
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { FeaturesConfigModule, I18nModule, provideDefaultConfig, } from '@spartacus/core';
import { OutletModule } from '../../../cms-structure/outlet/outlet.module';
import { ProductSummaryComponent } from './product-summary.component';
import { PromotionsModule } from '../../misc/promotions/promotions.module';
import * as i0 from "@angular/core";
export class ProductSummaryModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: ProductSummaryModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "17.0.5", ngImport: i0, type: ProductSummaryModule, declarations: [ProductSummaryComponent], imports: [CommonModule,
            OutletModule,
            I18nModule,
            PromotionsModule,
            FeaturesConfigModule], exports: [ProductSummaryComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: ProductSummaryModule, providers: [
            provideDefaultConfig({
                cmsComponents: {
                    ProductSummaryComponent: {
                        component: ProductSummaryComponent,
                    },
                },
                features: {
                    showPromotionsInPDP: false,
                },
            }),
        ], imports: [CommonModule,
            OutletModule,
            I18nModule,
            PromotionsModule,
            FeaturesConfigModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: ProductSummaryModule, decorators: [{
            type: NgModule,
            args: [{
                    providers: [
                        provideDefaultConfig({
                            cmsComponents: {
                                ProductSummaryComponent: {
                                    component: ProductSummaryComponent,
                                },
                            },
                            features: {
                                showPromotionsInPDP: false,
                            },
                        }),
                    ],
                    declarations: [ProductSummaryComponent],
                    exports: [ProductSummaryComponent],
                    imports: [
                        CommonModule,
                        OutletModule,
                        I18nModule,
                        PromotionsModule,
                        FeaturesConfigModule,
                    ],
                }]
        }] });
