// @spartacus/order/esm2022/components/my-account-v2/my-account-v2-orders.module.mjs
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { FormsModule } from '@angular/forms';
import { RouterModule } from '@angular/router';
import { UrlModule, I18nModule, provideDefaultConfig, AuthGuard, } from '@spartacus/core';
import { MediaModule, SpinnerModule } from '@spartacus/storefront';
import { MyAccountV2OrdersComponent } from './my-account-v2-orders.component';
import * as i0 from "@angular/core";
export class MyAccountV2OrdersModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: MyAccountV2OrdersModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "17.0.5", ngImport: i0, type: MyAccountV2OrdersModule, declarations: [MyAccountV2OrdersComponent], imports: [CommonModule,
            RouterModule,
            FormsModule,
            SpinnerModule,
            UrlModule,
            I18nModule,
            MediaModule], exports: [MyAccountV2OrdersComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: MyAccountV2OrdersModule, providers: [
            provideDefaultConfig({
                cmsComponents: {
                    MyAccountViewOrderComponent: {
                        component: MyAccountV2OrdersComponent,
                        guards: [AuthGuard],
                    },
                },
            }),
        ], imports: [CommonModule,
            RouterModule,
            FormsModule,
            SpinnerModule,
            UrlModule,
            I18nModule,
            MediaModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: MyAccountV2OrdersModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        RouterModule,
                        FormsModule,
                        SpinnerModule,
                        UrlModule,
                        I18nModule,
                        MediaModule,
                    ],
                    declarations: [MyAccountV2OrdersComponent],
                    exports: [MyAccountV2OrdersComponent],
                    providers: [
                        provideDefaultConfig({
                            cmsComponents: {
                                MyAccountViewOrderComponent: {
                                    component: MyAccountV2OrdersComponent,
                                    guards: [AuthGuard],
                                },
                            },
                        }),
                    ],
                }]
        }] });
