// @spartacus/storefront/esm2022/layout/a11y/keyboard-focus/keyboard-focus.module.mjs
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { FocusDirective } from './focus.directive';
import * as i0 from "@angular/core";
const directives = [
    // PersistFocusDirective,
    // VisibleFocusDirective,
    // BlockFocusDirective,
    // AutoFocusDirective,
    // EscapeFocusDirective,
    // LockFocusDirective,
    // TrapFocusDirective,
    // TabFocusDirective,
    FocusDirective,
];
export class KeyboardFocusModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: KeyboardFocusModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "17.0.5", ngImport: i0, type: KeyboardFocusModule, declarations: [
            // PersistFocusDirective,
            // VisibleFocusDirective,
            // BlockFocusDirective,
            // AutoFocusDirective,
            // EscapeFocusDirective,
            // LockFocusDirective,
            // TrapFocusDirective,
            // TabFocusDirective,
            FocusDirective], imports: [CommonModule], exports: [
            // PersistFocusDirective,
            // VisibleFocusDirective,
            // BlockFocusDirective,
            // AutoFocusDirective,
            // EscapeFocusDirective,
            // LockFocusDirective,
            // TrapFocusDirective,
            // TabFocusDirective,
            FocusDirective] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: KeyboardFocusModule, imports: [CommonModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: KeyboardFocusModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [...directives],
                    exports: [...directives],
                }]
        }] });
