// @spartacus/product/esm2022/image-zoom/components/product-image-zoom/product-image-zoom.module.mjs
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { RouterModule } from '@angular/router';
import { I18nModule, provideDefaultConfig } from '@spartacus/core';
import { CarouselModule, IconModule, KeyboardFocusModule, MediaModule, OutletModule, } from '@spartacus/storefront';
import { defaultProductImageZoomLayoutConfig } from './default-product-image-zoom-layout.config';
import { ProductImageZoomDialogComponent } from './product-image-zoom-dialog/product-image-zoom-dialog.component';
import { ProductImageZoomProductImagesComponent } from './product-image-zoom-product-images/product-image-zoom-product-images.component';
import { ProductImageZoomThumbnailsComponent } from './product-image-zoom-thumbnails/product-image-zoom-thumbnails.component';
import { ProductImageZoomTriggerComponent } from './product-image-zoom-trigger/product-image-zoom-trigger.component';
import { ProductImageZoomViewComponent } from './product-image-zoom-view/product-image-zoom-view.component';
import * as i0 from "@angular/core";
export class ProductImageZoomModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: ProductImageZoomModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "17.0.5", ngImport: i0, type: ProductImageZoomModule, declarations: [ProductImageZoomDialogComponent,
            ProductImageZoomProductImagesComponent,
            ProductImageZoomThumbnailsComponent,
            ProductImageZoomTriggerComponent,
            ProductImageZoomViewComponent], imports: [CarouselModule,
            CommonModule,
            I18nModule,
            IconModule,
            KeyboardFocusModule,
            MediaModule,
            OutletModule,
            RouterModule], exports: [ProductImageZoomDialogComponent,
            ProductImageZoomProductImagesComponent,
            ProductImageZoomThumbnailsComponent,
            ProductImageZoomTriggerComponent,
            ProductImageZoomViewComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: ProductImageZoomModule, providers: [
            provideDefaultConfig(defaultProductImageZoomLayoutConfig),
            provideDefaultConfig({
                cmsComponents: {
                    ProductImagesComponent: {
                        component: ProductImageZoomProductImagesComponent,
                    },
                },
            }),
        ], imports: [CarouselModule,
            CommonModule,
            I18nModule,
            IconModule,
            KeyboardFocusModule,
            MediaModule,
            OutletModule,
            RouterModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: ProductImageZoomModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CarouselModule,
                        CommonModule,
                        I18nModule,
                        IconModule,
                        KeyboardFocusModule,
                        MediaModule,
                        OutletModule,
                        RouterModule,
                    ],
                    providers: [
                        provideDefaultConfig(defaultProductImageZoomLayoutConfig),
                        provideDefaultConfig({
                            cmsComponents: {
                                ProductImagesComponent: {
                                    component: ProductImageZoomProductImagesComponent,
                                },
                            },
                        }),
                    ],
                    declarations: [
                        ProductImageZoomDialogComponent,
                        ProductImageZoomProductImagesComponent,
                        ProductImageZoomThumbnailsComponent,
                        ProductImageZoomTriggerComponent,
                        ProductImageZoomViewComponent,
                    ],
                    exports: [
                        ProductImageZoomDialogComponent,
                        ProductImageZoomProductImagesComponent,
                        ProductImageZoomThumbnailsComponent,
                        ProductImageZoomTriggerComponent,
                        ProductImageZoomViewComponent,
                    ],
                }]
        }] });
