// @spartacus/cart/esm2022/quick-order/components/cart-quick-order-form/cart-quick-order-form.module.mjs
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { ReactiveFormsModule } from '@angular/forms';
import { FeaturesConfigModule, I18nModule, provideDefaultConfig, } from '@spartacus/core';
import { FormErrorsModule } from '@spartacus/storefront';
import { CartQuickOrderFormComponent } from './cart-quick-order-form.component';
import * as i0 from "@angular/core";
export class CartQuickOrderFormModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: CartQuickOrderFormModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "17.0.5", ngImport: i0, type: CartQuickOrderFormModule, declarations: [CartQuickOrderFormComponent], imports: [CommonModule,
            ReactiveFormsModule,
            I18nModule,
            FormErrorsModule,
            FeaturesConfigModule], exports: [CartQuickOrderFormComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: CartQuickOrderFormModule, providers: [
            provideDefaultConfig({
                cmsComponents: {
                    CartQuickOrderFormComponent: {
                        component: CartQuickOrderFormComponent,
                    },
                },
            }),
        ], imports: [CommonModule,
            ReactiveFormsModule,
            I18nModule,
            FormErrorsModule,
            FeaturesConfigModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: CartQuickOrderFormModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        ReactiveFormsModule,
                        I18nModule,
                        FormErrorsModule,
                        FeaturesConfigModule,
                    ],
                    providers: [
                        provideDefaultConfig({
                            cmsComponents: {
                                CartQuickOrderFormComponent: {
                                    component: CartQuickOrderFormComponent,
                                },
                            },
                        }),
                    ],
                    declarations: [CartQuickOrderFormComponent],
                    exports: [CartQuickOrderFormComponent],
                }]
        }] });
