// @spartacus/core/esm2022/src/routing/configurable-routes/secure-portal-config/secure-portal-config-initializer.mjs
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { Injectable } from '@angular/core';
import { lastValueFrom } from 'rxjs';
import { map, switchMap, take, tap } from 'rxjs/operators';
import { BASE_SITE_CONTEXT_ID } from '../../../site-context/providers/context-ids';
import * as i0 from "@angular/core";
import * as i1 from "../../../site-context/facade/base-site.service";
import * as i2 from "../../../config/config-initializer/config-initializer.service";
export class SecurePortalConfigInitializer {
    constructor(baseSiteService, configInit) {
        this.baseSiteService = baseSiteService;
        this.configInit = configInit;
        this.scopes = ['routing'];
        this.configFactory = () => lastValueFrom(this.resolveConfig());
    }
    /**
     * Emits the Routing config basing on the current base site data.
     *
     * Completes after emitting the value.
     */
    resolveConfig() {
        return this.configInit.getStable('context').pipe(switchMap((config) => {
            const siteUid = config?.context?.[BASE_SITE_CONTEXT_ID]?.[0];
            return this.baseSiteService.get(siteUid).pipe(tap((baseSite) => {
                if (!baseSite) {
                    throw new Error(`Error: Cannot get base site config for ${siteUid}.`);
                }
            }), map((baseSite) => this.getRoutingConfig(baseSite)), take(1));
        }));
    }
    getRoutingConfig(source) {
        const result = {
            routing: {
                protected: source?.requiresAuthentication,
            },
        };
        return result;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: SecurePortalConfigInitializer, deps: [{ token: i1.BaseSiteService }, { token: i2.ConfigInitializerService }], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: SecurePortalConfigInitializer, providedIn: 'root' }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: SecurePortalConfigInitializer, decorators: [{
            type: Injectable,
            args: [{ providedIn: 'root' }]
        }], ctorParameters: () => [{ type: i1.BaseSiteService }, { type: i2.ConfigInitializerService }] });
