// @spartacus/checkout/esm2022/b2b/occ/checkout-b2b-occ.module.mjs
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { CheckoutCostCenterAdapter, CheckoutPaymentTypeAdapter, } from '@spartacus/checkout/b2b/core';
import { provideDefaultConfig } from '@spartacus/core';
import { OccCheckoutCostCenterAdapter } from './adapters/occ-checkout-cost-center.adapter';
import { OccCheckoutPaymentTypeAdapter } from './adapters/occ-checkout-payment-type.adapter';
import { defaultOccCheckoutB2BConfig } from './config/default-occ-checkout-b2b-config';
import * as i0 from "@angular/core";
export class CheckoutB2BOccModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: CheckoutB2BOccModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "17.0.5", ngImport: i0, type: CheckoutB2BOccModule, imports: [CommonModule] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: CheckoutB2BOccModule, providers: [
            provideDefaultConfig(defaultOccCheckoutB2BConfig),
            {
                provide: CheckoutPaymentTypeAdapter,
                useClass: OccCheckoutPaymentTypeAdapter,
            },
            {
                provide: CheckoutCostCenterAdapter,
                useClass: OccCheckoutCostCenterAdapter,
            },
        ], imports: [CommonModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: CheckoutB2BOccModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    providers: [
                        provideDefaultConfig(defaultOccCheckoutB2BConfig),
                        {
                            provide: CheckoutPaymentTypeAdapter,
                            useClass: OccCheckoutPaymentTypeAdapter,
                        },
                        {
                            provide: CheckoutCostCenterAdapter,
                            useClass: OccCheckoutCostCenterAdapter,
                        },
                    ],
                }]
        }] });
