// @spartacus/cart/esm2022/quick-order/components/quick-order/quick-order-list.module.mjs
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { ReactiveFormsModule } from '@angular/forms';
import { RouterModule } from '@angular/router';
import { I18nModule, provideDefaultConfig, UrlModule, } from '@spartacus/core';
import { AtMessageModule, FormErrorsModule, IconModule, ItemCounterModule, MediaModule, MessageComponentModule, ProgressButtonModule, } from '@spartacus/storefront';
import { QuickOrderComponent } from './quick-order.component';
import { QuickOrderFormComponent } from './form/quick-order-form.component';
import { QuickOrderItemComponent } from './table/item/quick-order-item.component';
import { QuickOrderTableComponent } from './table/quick-order-table.component';
import * as i0 from "@angular/core";
export class QuickOrderListModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: QuickOrderListModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "17.0.5", ngImport: i0, type: QuickOrderListModule, declarations: [QuickOrderComponent,
            QuickOrderFormComponent,
            QuickOrderItemComponent,
            QuickOrderTableComponent], imports: [AtMessageModule,
            CommonModule,
            FormErrorsModule,
            I18nModule,
            IconModule,
            ItemCounterModule,
            MediaModule,
            MessageComponentModule,
            ProgressButtonModule,
            ReactiveFormsModule,
            RouterModule,
            UrlModule], exports: [QuickOrderComponent,
            QuickOrderFormComponent,
            QuickOrderItemComponent,
            QuickOrderTableComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: QuickOrderListModule, providers: [
            provideDefaultConfig({
                cmsComponents: {
                    QuickOrderComponent: {
                        component: QuickOrderComponent,
                        data: {
                            quickOrderListLimit: 10,
                        },
                    },
                },
            }),
        ], imports: [AtMessageModule,
            CommonModule,
            FormErrorsModule,
            I18nModule,
            IconModule,
            ItemCounterModule,
            MediaModule,
            MessageComponentModule,
            ProgressButtonModule,
            ReactiveFormsModule,
            RouterModule,
            UrlModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: QuickOrderListModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        AtMessageModule,
                        CommonModule,
                        FormErrorsModule,
                        I18nModule,
                        IconModule,
                        ItemCounterModule,
                        MediaModule,
                        MessageComponentModule,
                        ProgressButtonModule,
                        ReactiveFormsModule,
                        RouterModule,
                        UrlModule,
                    ],
                    providers: [
                        provideDefaultConfig({
                            cmsComponents: {
                                QuickOrderComponent: {
                                    component: QuickOrderComponent,
                                    data: {
                                        quickOrderListLimit: 10,
                                    },
                                },
                            },
                        }),
                    ],
                    declarations: [
                        QuickOrderComponent,
                        QuickOrderFormComponent,
                        QuickOrderItemComponent,
                        QuickOrderTableComponent,
                    ],
                    exports: [
                        QuickOrderComponent,
                        QuickOrderFormComponent,
                        QuickOrderItemComponent,
                        QuickOrderTableComponent,
                    ],
                }]
        }] });
