// @spartacus/cart/esm2022/wish-list/root/facade/wish-list.facade.mjs
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { Injectable } from '@angular/core';
import { facadeFactory } from '@spartacus/core';
import { CART_WISH_LIST_CORE_FEATURE } from '../feature-name';
import * as i0 from "@angular/core";
export class WishListFacade {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: WishListFacade, deps: [], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: WishListFacade, providedIn: 'root', useFactory: () => facadeFactory({
            facade: WishListFacade,
            feature: CART_WISH_LIST_CORE_FEATURE,
            methods: [
                'createWishList',
                'getWishList',
                'loadWishList',
                'addEntry',
                'removeEntry',
                'getWishListLoading',
            ],
            async: true,
        }) }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: WishListFacade, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                    useFactory: () => facadeFactory({
                        facade: WishListFacade,
                        feature: CART_WISH_LIST_CORE_FEATURE,
                        methods: [
                            'createWishList',
                            'getWishList',
                            'loadWishList',
                            'addEntry',
                            'removeEntry',
                            'getWishListLoading',
                        ],
                        async: true,
                    }),
                }]
        }] });
