// @spartacus/storefront/esm2022/cms-components/product/product-list/product-list.module.mjs
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { RouterModule } from '@angular/router';
import { FeaturesConfigModule, I18nModule, provideDefaultConfig, UrlModule, } from '@spartacus/core';
import { InfiniteScrollModule } from 'ngx-infinite-scroll';
import { OutletModule } from '../../../cms-structure/outlet/outlet.module';
import { PageComponentModule } from '../../../cms-structure/page/component/page-component.module';
import { AtMessageModule, ItemCounterModule, ListNavigationModule, MediaModule, SpinnerModule, StarRatingModule, } from '../../../shared/index';
import { IconModule } from '../../misc/icon/index';
import { defaultViewConfig } from '../config/default-view-config';
import { ProductListComponent } from './container/product-list.component';
import { ProductScrollComponent } from './container/product-scroll/product-scroll.component';
import { ProductGridItemComponent } from './product-grid-item/product-grid-item.component';
import { ProductListItemComponent } from './product-list-item/product-list-item.component';
import { ProductViewComponent } from './product-view/product-view.component';
import * as i0 from "@angular/core";
export class ProductListModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: ProductListModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "17.0.5", ngImport: i0, type: ProductListModule, declarations: [ProductListComponent,
            ProductListItemComponent,
            ProductGridItemComponent,
            ProductViewComponent,
            ProductScrollComponent], imports: [AtMessageModule,
            CommonModule,
            I18nModule,
            IconModule,
            InfiniteScrollModule,
            ItemCounterModule,
            ListNavigationModule,
            MediaModule,
            OutletModule,
            PageComponentModule,
            RouterModule,
            SpinnerModule,
            StarRatingModule,
            UrlModule,
            FeaturesConfigModule], exports: [ProductListComponent,
            ProductListItemComponent,
            ProductGridItemComponent,
            ProductViewComponent,
            ProductScrollComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: ProductListModule, providers: [
            provideDefaultConfig(defaultViewConfig),
            provideDefaultConfig({
                cmsComponents: {
                    CMSProductListComponent: {
                        component: ProductListComponent,
                        data: {
                            composition: {
                                inner: ['ProductAddToCartComponent'],
                            },
                        },
                    },
                    ProductGridComponent: {
                        component: ProductListComponent,
                        data: {
                            composition: {
                                inner: ['ProductAddToCartComponent'],
                            },
                        },
                    },
                    SearchResultsListComponent: {
                        component: ProductListComponent,
                        data: {
                            composition: {
                                inner: ['ProductAddToCartComponent'],
                            },
                        },
                    },
                },
            }),
        ], imports: [AtMessageModule,
            CommonModule,
            I18nModule,
            IconModule,
            InfiniteScrollModule,
            ItemCounterModule,
            ListNavigationModule,
            MediaModule,
            OutletModule,
            PageComponentModule,
            RouterModule,
            SpinnerModule,
            StarRatingModule,
            UrlModule,
            FeaturesConfigModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: ProductListModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        AtMessageModule,
                        CommonModule,
                        I18nModule,
                        IconModule,
                        InfiniteScrollModule,
                        ItemCounterModule,
                        ListNavigationModule,
                        MediaModule,
                        OutletModule,
                        PageComponentModule,
                        RouterModule,
                        SpinnerModule,
                        StarRatingModule,
                        UrlModule,
                        FeaturesConfigModule,
                    ],
                    providers: [
                        provideDefaultConfig(defaultViewConfig),
                        provideDefaultConfig({
                            cmsComponents: {
                                CMSProductListComponent: {
                                    component: ProductListComponent,
                                    data: {
                                        composition: {
                                            inner: ['ProductAddToCartComponent'],
                                        },
                                    },
                                },
                                ProductGridComponent: {
                                    component: ProductListComponent,
                                    data: {
                                        composition: {
                                            inner: ['ProductAddToCartComponent'],
                                        },
                                    },
                                },
                                SearchResultsListComponent: {
                                    component: ProductListComponent,
                                    data: {
                                        composition: {
                                            inner: ['ProductAddToCartComponent'],
                                        },
                                    },
                                },
                            },
                        }),
                    ],
                    declarations: [
                        ProductListComponent,
                        ProductListItemComponent,
                        ProductGridItemComponent,
                        ProductViewComponent,
                        ProductScrollComponent,
                    ],
                    exports: [
                        ProductListComponent,
                        ProductListItemComponent,
                        ProductGridItemComponent,
                        ProductViewComponent,
                        ProductScrollComponent,
                    ],
                }]
        }] });
