// @spartacus/storefront/esm2022/cms-components/myaccount/my-account-v2/my-account-v2-notification-preference/my-account-v2-notification-preference.module.mjs
/*
 * SPDX-FileCopyrightText: 2023 SAP Spartacus team <spartacus-team@sap.com>
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { RouterModule } from '@angular/router';
import { AuthGuard, I18nModule } from '@spartacus/core';
import { SpinnerModule } from '../../../../shared/components/spinner/spinner.module';
import { CmsPageGuard } from '../../../../cms-structure/guards/cms-page.guard';
import { PageLayoutComponent } from '../../../../cms-structure/page/page-layout/page-layout.component';
import { MyAccountV2NotificationPreferenceComponent } from './my-account-v2-notification-preference.component';
import * as i0 from "@angular/core";
import * as i1 from "@angular/router";
export class MyAccountV2NotificationPreferenceModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: MyAccountV2NotificationPreferenceModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "17.0.5", ngImport: i0, type: MyAccountV2NotificationPreferenceModule, declarations: [MyAccountV2NotificationPreferenceComponent], imports: [CommonModule,
            SpinnerModule,
            I18nModule, i1.RouterModule], exports: [MyAccountV2NotificationPreferenceComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: MyAccountV2NotificationPreferenceModule, imports: [CommonModule,
            SpinnerModule,
            I18nModule,
            RouterModule.forChild([
                {
                    // @ts-ignore
                    path: null,
                    canActivate: [AuthGuard, CmsPageGuard],
                    component: PageLayoutComponent,
                    data: { cxRoute: 'notificationPreference' },
                },
            ])] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: MyAccountV2NotificationPreferenceModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [MyAccountV2NotificationPreferenceComponent],
                    imports: [
                        CommonModule,
                        SpinnerModule,
                        I18nModule,
                        RouterModule.forChild([
                            {
                                // @ts-ignore
                                path: null,
                                canActivate: [AuthGuard, CmsPageGuard],
                                component: PageLayoutComponent,
                                data: { cxRoute: 'notificationPreference' },
                            },
                        ]),
                    ],
                    exports: [MyAccountV2NotificationPreferenceComponent],
                }]
        }] });
