// @spartacus/checkout/esm2022/base/components/checkout-review/checkout-review-shipping/checkout-review-shipping.module.mjs
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { RouterModule } from '@angular/router';
import { I18nModule, provideDefaultConfig, UrlModule, } from '@spartacus/core';
import { CardModule, IconModule, OutletModule } from '@spartacus/storefront';
import { CartNotEmptyGuard } from '../../guards/cart-not-empty.guard';
import { CheckoutAuthGuard } from '../../guards/checkout-auth.guard';
import { CheckoutReviewShippingComponent } from './checkout-review-shipping.component';
import * as i0 from "@angular/core";
export class CheckoutReviewShippingModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: CheckoutReviewShippingModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "17.0.5", ngImport: i0, type: CheckoutReviewShippingModule, declarations: [CheckoutReviewShippingComponent], imports: [CommonModule,
            I18nModule,
            CardModule,
            UrlModule,
            RouterModule,
            IconModule,
            OutletModule], exports: [CheckoutReviewShippingComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: CheckoutReviewShippingModule, providers: [
            provideDefaultConfig({
                cmsComponents: {
                    CheckoutReviewShipping: {
                        component: CheckoutReviewShippingComponent,
                        guards: [CheckoutAuthGuard, CartNotEmptyGuard],
                    },
                },
            }),
        ], imports: [CommonModule,
            I18nModule,
            CardModule,
            UrlModule,
            RouterModule,
            IconModule,
            OutletModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: CheckoutReviewShippingModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [CheckoutReviewShippingComponent],
                    exports: [CheckoutReviewShippingComponent],
                    imports: [
                        CommonModule,
                        I18nModule,
                        CardModule,
                        UrlModule,
                        RouterModule,
                        IconModule,
                        OutletModule,
                    ],
                    providers: [
                        provideDefaultConfig({
                            cmsComponents: {
                                CheckoutReviewShipping: {
                                    component: CheckoutReviewShippingComponent,
                                    guards: [CheckoutAuthGuard, CartNotEmptyGuard],
                                },
                            },
                        }),
                    ],
                }]
        }] });
