// @spartacus/checkout/esm2022/b2b/components/checkout-b2b-components.module.mjs
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { CheckoutAuthGuard, CheckoutStepsSetGuard, } from '@spartacus/checkout/base/components';
import { CheckoutCostCenterModule } from './checkout-cost-center/checkout-cost-center.module';
import { B2BCheckoutDeliveryAddressModule } from './checkout-delivery-address/checkout-delivery-address.module';
import { CheckoutPaymentTypeModule } from './checkout-payment-type/checkout-payment-type.module';
import { B2BCheckoutReviewSubmitModule } from './checkout-review-submit/checkout-review-submit.module';
import { CheckoutB2BAuthGuard } from './guards/checkout-b2b-auth.guard';
import { CheckoutB2BStepsSetGuard } from './guards/checkout-b2b-steps-set.guard';
import * as i0 from "@angular/core";
export class CheckoutB2BComponentsModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: CheckoutB2BComponentsModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "17.0.5", ngImport: i0, type: CheckoutB2BComponentsModule, imports: [CommonModule,
            CheckoutCostCenterModule,
            CheckoutPaymentTypeModule,
            B2BCheckoutDeliveryAddressModule,
            B2BCheckoutReviewSubmitModule] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: CheckoutB2BComponentsModule, providers: [
            {
                provide: CheckoutAuthGuard,
                useExisting: CheckoutB2BAuthGuard,
            },
            {
                provide: CheckoutStepsSetGuard,
                useExisting: CheckoutB2BStepsSetGuard,
            },
        ], imports: [CommonModule,
            CheckoutCostCenterModule,
            CheckoutPaymentTypeModule,
            B2BCheckoutDeliveryAddressModule,
            B2BCheckoutReviewSubmitModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: CheckoutB2BComponentsModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        CheckoutCostCenterModule,
                        CheckoutPaymentTypeModule,
                        B2BCheckoutDeliveryAddressModule,
                        B2BCheckoutReviewSubmitModule,
                    ],
                    providers: [
                        {
                            provide: CheckoutAuthGuard,
                            useExisting: CheckoutB2BAuthGuard,
                        },
                        {
                            provide: CheckoutStepsSetGuard,
                            useExisting: CheckoutB2BStepsSetGuard,
                        },
                    ],
                }]
        }] });
