// @spartacus/cart/esm2022/base/components/abstract-order-context/abstract-order-context.directive.mjs
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { Directive, Input, inject, } from '@angular/core';
import { AbstractOrderType } from '@spartacus/cart/base/root';
import { AbstractOrderContextSource } from './abstract-order-context-source.model';
import { AbstractOrderContext } from './abstract-order-context.model';
import * as i0 from "@angular/core";
export class AbstractOrderContextDirective {
    constructor() {
        this.abstractOrderContextSource = inject(AbstractOrderContextSource);
    }
    ngOnChanges(changes) {
        const contextChanges = changes.cxAbstractOrderContext;
        if (contextChanges) {
            let abstractOrderKey;
            const type = contextChanges.currentValue.type;
            if (type === AbstractOrderType.CART) {
                abstractOrderKey = { type };
            }
            else {
                const id = contextChanges.currentValue.id;
                if (id) {
                    abstractOrderKey = { type, id };
                }
                else {
                    throw new Error('id required for order, quote or saved cart');
                }
            }
            this.abstractOrderContextSource.key$.next(abstractOrderKey);
        }
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: AbstractOrderContextDirective, deps: [], target: i0.ɵɵFactoryTarget.Directive }); }
    static { this.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "17.0.5", type: AbstractOrderContextDirective, selector: "[cxAbstractOrderContext]", inputs: { cxAbstractOrderContext: "cxAbstractOrderContext" }, providers: [
            AbstractOrderContextSource,
            { provide: AbstractOrderContext, useExisting: AbstractOrderContextSource },
        ], usesOnChanges: true, ngImport: i0 }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: AbstractOrderContextDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[cxAbstractOrderContext]',
                    providers: [
                        AbstractOrderContextSource,
                        { provide: AbstractOrderContext, useExisting: AbstractOrderContextSource },
                    ],
                }]
        }], propDecorators: { cxAbstractOrderContext: [{
                type: Input
            }] } });
