// @spartacus/checkout/esm2022/base/core/facade/checkout-payment.service.mjs
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { Injectable } from '@angular/core';
import { CheckoutPaymentCardTypesQueryReloadEvent, CheckoutPaymentCardTypesQueryResetEvent, CheckoutPaymentDetailsCreatedEvent, CheckoutPaymentDetailsSetEvent, } from '@spartacus/checkout/base/root';
import { CommandStrategy, OCC_USER_ID_ANONYMOUS, } from '@spartacus/core';
import { combineLatest } from 'rxjs';
import { map, switchMap, take, tap } from 'rxjs/operators';
import * as i0 from "@angular/core";
import * as i1 from "@spartacus/cart/base/root";
import * as i2 from "@spartacus/core";
import * as i3 from "../connectors/checkout-payment/checkout-payment.connector";
import * as i4 from "@spartacus/checkout/base/root";
export class CheckoutPaymentService {
    /**
     * Returns the reload events for the cardTypes query
     */
    getCheckoutPaymentCardTypesQueryReloadEvents() {
        return [CheckoutPaymentCardTypesQueryReloadEvent];
    }
    /**
     * Returns the reset events for the cardTypes query
     */
    getCheckoutPaymentCardTypesQueryResetEvents() {
        return [CheckoutPaymentCardTypesQueryResetEvent];
    }
    constructor(activeCartFacade, userIdService, queryService, commandService, eventService, checkoutPaymentConnector, checkoutQueryFacade) {
        this.activeCartFacade = activeCartFacade;
        this.userIdService = userIdService;
        this.queryService = queryService;
        this.commandService = commandService;
        this.eventService = eventService;
        this.checkoutPaymentConnector = checkoutPaymentConnector;
        this.checkoutQueryFacade = checkoutQueryFacade;
        this.paymentCardTypesQuery = this.queryService.create(() => this.checkoutPaymentConnector.getPaymentCardTypes(), {
            reloadOn: this.getCheckoutPaymentCardTypesQueryReloadEvents(),
            resetOn: this.getCheckoutPaymentCardTypesQueryResetEvents(),
        });
        this.createPaymentMethodCommand = this.commandService.create((paymentDetails) => this.checkoutPreconditions().pipe(switchMap(([userId, cartId]) => this.checkoutPaymentConnector
            .createPaymentDetails(userId, cartId, paymentDetails)
            .pipe(tap((response) => this.eventService.dispatch({ userId, cartId, paymentDetails: response }, CheckoutPaymentDetailsCreatedEvent))))), {
            strategy: CommandStrategy.CancelPrevious,
        });
        this.setPaymentMethodCommand = this.commandService.create((paymentDetails) => this.checkoutPreconditions().pipe(switchMap(([userId, cartId]) => {
            const paymentDetailsId = paymentDetails?.id;
            if (!paymentDetailsId) {
                throw new Error('Checkout conditions not met');
            }
            return this.checkoutPaymentConnector
                .setPaymentDetails(userId, cartId, paymentDetailsId)
                .pipe(tap(() => this.eventService.dispatch({
                userId,
                cartId,
                paymentDetailsId,
            }, CheckoutPaymentDetailsSetEvent)));
        })), {
            strategy: CommandStrategy.CancelPrevious,
        });
    }
    /**
     * Performs the necessary checkout preconditions.
     */
    checkoutPreconditions() {
        return combineLatest([
            this.userIdService.takeUserId(),
            this.activeCartFacade.takeActiveCartId(),
            this.activeCartFacade.isGuestCart(),
        ]).pipe(take(1), map(([userId, cartId, isGuestCart]) => {
            if (!userId ||
                !cartId ||
                (userId === OCC_USER_ID_ANONYMOUS && !isGuestCart)) {
                throw new Error('Checkout conditions not met');
            }
            return [userId, cartId];
        }));
    }
    getPaymentCardTypesState() {
        return this.paymentCardTypesQuery.getState();
    }
    getPaymentCardTypes() {
        return this.getPaymentCardTypesState().pipe(map((state) => state.data ?? []));
    }
    getPaymentDetailsState() {
        return this.checkoutQueryFacade
            .getCheckoutDetailsState()
            .pipe(map((state) => ({ ...state, data: state.data?.paymentInfo })));
    }
    createPaymentDetails(paymentDetails) {
        return this.createPaymentMethodCommand.execute(paymentDetails);
    }
    setPaymentDetails(paymentDetails) {
        return this.setPaymentMethodCommand.execute(paymentDetails);
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: CheckoutPaymentService, deps: [{ token: i1.ActiveCartFacade }, { token: i2.UserIdService }, { token: i2.QueryService }, { token: i2.CommandService }, { token: i2.EventService }, { token: i3.CheckoutPaymentConnector }, { token: i4.CheckoutQueryFacade }], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: CheckoutPaymentService }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: CheckoutPaymentService, decorators: [{
            type: Injectable
        }], ctorParameters: () => [{ type: i1.ActiveCartFacade }, { type: i2.UserIdService }, { type: i2.QueryService }, { type: i2.CommandService }, { type: i2.EventService }, { type: i3.CheckoutPaymentConnector }, { type: i4.CheckoutQueryFacade }] });
