// @spartacus/checkout/esm2022/base/core/facade/checkout-delivery-modes.service.mjs
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { Injectable } from '@angular/core';
import { CheckoutDeliveryModeClearedErrorEvent, CheckoutDeliveryModeClearedEvent, CheckoutDeliveryModeSetEvent, CheckoutSupportedDeliveryModesQueryReloadEvent, CheckoutSupportedDeliveryModesQueryResetEvent, } from '@spartacus/checkout/base/root';
import { CommandStrategy, OCC_USER_ID_ANONYMOUS, } from '@spartacus/core';
import { combineLatest } from 'rxjs';
import { filter, map, switchMap, take, tap } from 'rxjs/operators';
import * as i0 from "@angular/core";
import * as i1 from "@spartacus/cart/base/root";
import * as i2 from "@spartacus/core";
import * as i3 from "../connectors/checkout-delivery-modes/checkout-delivery-modes.connector";
import * as i4 from "@spartacus/checkout/base/root";
export class CheckoutDeliveryModesService {
    /**
     * Returns the reload events for the supportedDeliveryModes query
     */
    getCheckoutSupportedDeliveryModesQueryReloadEvents() {
        return [CheckoutSupportedDeliveryModesQueryReloadEvent];
    }
    /**
     * Return the reset events for the supportedDeliveryModes query
     */
    getCheckoutSupportedDeliveryModesQueryResetEvents() {
        return [CheckoutSupportedDeliveryModesQueryResetEvent];
    }
    constructor(activeCartFacade, userIdService, eventService, queryService, commandService, checkoutDeliveryModesConnector, checkoutQueryFacade) {
        this.activeCartFacade = activeCartFacade;
        this.userIdService = userIdService;
        this.eventService = eventService;
        this.queryService = queryService;
        this.commandService = commandService;
        this.checkoutDeliveryModesConnector = checkoutDeliveryModesConnector;
        this.checkoutQueryFacade = checkoutQueryFacade;
        this.supportedDeliveryModesQuery = this.queryService.create(() => this.checkoutPreconditions().pipe(switchMap(([userId, cartId]) => this.checkoutDeliveryModesConnector.getSupportedModes(userId, cartId))), {
            reloadOn: this.getCheckoutSupportedDeliveryModesQueryReloadEvents(),
            resetOn: this.getCheckoutSupportedDeliveryModesQueryResetEvents(),
        });
        this.setDeliveryModeCommand = this.commandService.create((deliveryModeCode) => this.checkoutPreconditions().pipe(switchMap(([userId, cartId]) => this.checkoutDeliveryModesConnector
            .setMode(userId, cartId, deliveryModeCode)
            .pipe(tap(() => {
            this.eventService.dispatch({ userId, cartId, cartCode: cartId, deliveryModeCode }, CheckoutDeliveryModeSetEvent);
        })))), {
            strategy: CommandStrategy.CancelPrevious,
        });
        this.clearDeliveryModeCommand = this.commandService.create(() => this.checkoutPreconditions().pipe(switchMap(([userId, cartId]) => this.checkoutDeliveryModesConnector
            .clearCheckoutDeliveryMode(userId, cartId)
            .pipe(tap({
            next: () => {
                this.eventService.dispatch({
                    userId,
                    cartId,
                    /**
                     * As we know the cart is not anonymous (precondition checked),
                     * we can safely use the cartId, which is actually the cart.code.
                     */
                    cartCode: cartId,
                }, CheckoutDeliveryModeClearedEvent);
            },
            error: () => {
                this.eventService.dispatch({
                    userId,
                    cartId,
                    /**
                     * As we know the cart is not anonymous (precondition checked),
                     * we can safely use the cartId, which is actually the cart.code.
                     */
                    cartCode: cartId,
                }, CheckoutDeliveryModeClearedErrorEvent);
            },
        })))), {
            strategy: CommandStrategy.CancelPrevious,
        });
    }
    /**
     * Performs the necessary checkout preconditions.
     */
    checkoutPreconditions() {
        return combineLatest([
            this.userIdService.takeUserId(),
            this.activeCartFacade.takeActiveCartId(),
            this.activeCartFacade.isGuestCart(),
        ]).pipe(take(1), map(([userId, cartId, isGuestCart]) => {
            if (!userId ||
                !cartId ||
                (userId === OCC_USER_ID_ANONYMOUS && !isGuestCart)) {
                throw new Error('Checkout conditions not met');
            }
            return [userId, cartId];
        }));
    }
    getSupportedDeliveryModesState() {
        return this.supportedDeliveryModesQuery.getState();
    }
    getSupportedDeliveryModes() {
        return this.getSupportedDeliveryModesState().pipe(filter((state) => !state.loading), map((state) => state.data ?? []));
    }
    getSelectedDeliveryModeState() {
        return this.checkoutQueryFacade
            .getCheckoutDetailsState()
            .pipe(map((state) => ({ ...state, data: state.data?.deliveryMode })));
    }
    setDeliveryMode(mode) {
        return this.setDeliveryModeCommand.execute(mode);
    }
    clearCheckoutDeliveryMode() {
        return this.clearDeliveryModeCommand.execute();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: CheckoutDeliveryModesService, deps: [{ token: i1.ActiveCartFacade }, { token: i2.UserIdService }, { token: i2.EventService }, { token: i2.QueryService }, { token: i2.CommandService }, { token: i3.CheckoutDeliveryModesConnector }, { token: i4.CheckoutQueryFacade }], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: CheckoutDeliveryModesService }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: CheckoutDeliveryModesService, decorators: [{
            type: Injectable
        }], ctorParameters: () => [{ type: i1.ActiveCartFacade }, { type: i2.UserIdService }, { type: i2.EventService }, { type: i2.QueryService }, { type: i2.CommandService }, { type: i3.CheckoutDeliveryModesConnector }, { type: i4.CheckoutQueryFacade }] });
