// @spartacus/checkout/esm2022/base/core/checkout-core.module.mjs
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { NgModule } from '@angular/core';
import { PageMetaResolver } from '@spartacus/core';
import { CheckoutDeliveryAddressConnector } from './connectors/checkout-delivery-address/checkout-delivery-address.connector';
import { CheckoutDeliveryModesConnector } from './connectors/checkout-delivery-modes/checkout-delivery-modes.connector';
import { CheckoutPaymentConnector } from './connectors/checkout-payment/checkout-payment.connector';
import { CheckoutConnector } from './connectors/checkout/checkout.connector';
import { facadeProviders } from './facade/facade-providers';
import { CheckoutPageMetaResolver } from './services/checkout-page-meta.resolver';
import * as i0 from "@angular/core";
export class CheckoutCoreModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: CheckoutCoreModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "17.0.5", ngImport: i0, type: CheckoutCoreModule }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: CheckoutCoreModule, providers: [
            ...facadeProviders,
            CheckoutDeliveryAddressConnector,
            CheckoutDeliveryModesConnector,
            CheckoutPaymentConnector,
            CheckoutConnector,
            CheckoutPageMetaResolver,
            {
                provide: PageMetaResolver,
                useExisting: CheckoutPageMetaResolver,
                multi: true,
            },
        ] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: CheckoutCoreModule, decorators: [{
            type: NgModule,
            args: [{
                    providers: [
                        ...facadeProviders,
                        CheckoutDeliveryAddressConnector,
                        CheckoutDeliveryModesConnector,
                        CheckoutPaymentConnector,
                        CheckoutConnector,
                        CheckoutPageMetaResolver,
                        {
                            provide: PageMetaResolver,
                            useExisting: CheckoutPageMetaResolver,
                            multi: true,
                        },
                    ],
                }]
        }] });
