// @spartacus/checkout/esm2022/base/occ/adapters/occ-checkout.adapter.mjs
import { Injectable, inject } from '@angular/core';
import { CHECKOUT_NORMALIZER, } from '@spartacus/checkout/base/core';
import { LoggerService, backOff, isJaloError, normalizeHttpError, } from '@spartacus/core';
import { catchError } from 'rxjs/operators';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common/http";
import * as i2 from "@spartacus/core";
export class OccCheckoutAdapter {
    constructor(http, occEndpoints, converter) {
        this.http = http;
        this.occEndpoints = occEndpoints;
        this.converter = converter;
        this.logger = inject(LoggerService);
    }
    getCheckoutDetails(userId, cartId) {
        return this.http
            .get(this.getGetCheckoutDetailsEndpoint(userId, cartId))
            .pipe(catchError((error) => {
            throw normalizeHttpError(error, this.logger);
        }), backOff({
            shouldRetry: isJaloError,
        }), this.converter.pipeable(CHECKOUT_NORMALIZER));
    }
    getGetCheckoutDetailsEndpoint(userId, cartId) {
        return this.occEndpoints.buildUrl('getCheckoutDetails', {
            urlParams: {
                userId,
                cartId,
            },
        });
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: OccCheckoutAdapter, deps: [{ token: i1.HttpClient }, { token: i2.OccEndpointsService }, { token: i2.ConverterService }], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: OccCheckoutAdapter }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: OccCheckoutAdapter, decorators: [{
            type: Injectable
        }], ctorParameters: () => [{ type: i1.HttpClient }, { type: i2.OccEndpointsService }, { type: i2.ConverterService }] });
