// @spartacus/checkout/esm2022/base/occ/adapters/occ-checkout-delivery-address.adapter.mjs
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { HttpHeaders } from '@angular/common/http';
import { Injectable, inject } from '@angular/core';
import { ADDRESS_NORMALIZER, ADDRESS_SERIALIZER, LoggerService, backOff, isJaloError, normalizeHttpError, } from '@spartacus/core';
import { catchError } from 'rxjs/operators';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common/http";
import * as i2 from "@spartacus/core";
export class OccCheckoutDeliveryAddressAdapter {
    constructor(http, occEndpoints, converter) {
        this.http = http;
        this.occEndpoints = occEndpoints;
        this.converter = converter;
        this.logger = inject(LoggerService);
    }
    createAddress(userId, cartId, address) {
        address = this.converter.convert(address, ADDRESS_SERIALIZER);
        return this.http
            .post(this.getCreateDeliveryAddressEndpoint(userId, cartId), address, {
            headers: new HttpHeaders().set('Content-Type', 'application/json'),
        })
            .pipe(catchError((error) => {
            throw normalizeHttpError(error, this.logger);
        }), backOff({
            shouldRetry: isJaloError,
        }), this.converter.pipeable(ADDRESS_NORMALIZER));
    }
    getCreateDeliveryAddressEndpoint(userId, cartId) {
        return this.occEndpoints.buildUrl('createDeliveryAddress', {
            urlParams: {
                userId,
                cartId,
            },
        });
    }
    setAddress(userId, cartId, addressId) {
        return this.http
            .put(this.getSetDeliveryAddressEndpoint(userId, cartId, addressId), {})
            .pipe(catchError((error) => {
            throw normalizeHttpError(error, this.logger);
        }), backOff({
            shouldRetry: isJaloError,
        }));
    }
    getSetDeliveryAddressEndpoint(userId, cartId, addressId) {
        return this.occEndpoints.buildUrl('setDeliveryAddress', {
            urlParams: { userId, cartId },
            queryParams: { addressId },
        });
    }
    clearCheckoutDeliveryAddress(userId, cartId) {
        return this.http
            .delete(this.getRemoveDeliveryAddressEndpoint(userId, cartId))
            .pipe(catchError((error) => {
            throw normalizeHttpError(error, this.logger);
        }), backOff({
            shouldRetry: isJaloError,
        }));
    }
    getRemoveDeliveryAddressEndpoint(userId, cartId) {
        return this.occEndpoints.buildUrl('removeDeliveryAddress', {
            urlParams: {
                userId,
                cartId,
            },
        });
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: OccCheckoutDeliveryAddressAdapter, deps: [{ token: i1.HttpClient }, { token: i2.OccEndpointsService }, { token: i2.ConverterService }], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: OccCheckoutDeliveryAddressAdapter }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: OccCheckoutDeliveryAddressAdapter, decorators: [{
            type: Injectable
        }], ctorParameters: () => [{ type: i1.HttpClient }, { type: i2.OccEndpointsService }, { type: i2.ConverterService }] });
