// @spartacus/checkout/esm2022/base/occ/adapters/occ-checkout-delivery-modes.adapter.mjs
import { Injectable, inject } from '@angular/core';
import { DELIVERY_MODE_NORMALIZER, } from '@spartacus/checkout/base/core';
import { LoggerService, backOff, isJaloError, normalizeHttpError, } from '@spartacus/core';
import { catchError, map } from 'rxjs/operators';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common/http";
import * as i2 from "@spartacus/core";
export class OccCheckoutDeliveryModesAdapter {
    constructor(http, occEndpoints, converter) {
        this.http = http;
        this.occEndpoints = occEndpoints;
        this.converter = converter;
        this.logger = inject(LoggerService);
    }
    setMode(userId, cartId, deliveryModeId) {
        return this.http
            .put(this.getSetDeliveryModeEndpoint(userId, cartId, deliveryModeId), {})
            .pipe(catchError((error) => {
            throw normalizeHttpError(error, this.logger);
        }), backOff({
            shouldRetry: isJaloError,
        }));
    }
    getSetDeliveryModeEndpoint(userId, cartId, deliveryModeId) {
        return this.occEndpoints.buildUrl('setDeliveryMode', {
            urlParams: {
                userId,
                cartId,
            },
            queryParams: { deliveryModeId },
        });
    }
    getSupportedModes(userId, cartId) {
        return this.http
            .get(this.getDeliveryModesEndpoint(userId, cartId))
            .pipe(catchError((error) => {
            throw normalizeHttpError(error, this.logger);
        }), backOff({
            shouldRetry: isJaloError,
        }), map((listResponse) => listResponse.deliveryModes ?? []), this.converter.pipeableMany(DELIVERY_MODE_NORMALIZER));
    }
    getDeliveryModesEndpoint(userId, cartId) {
        return this.occEndpoints.buildUrl('deliveryModes', {
            urlParams: { userId, cartId },
        });
    }
    clearCheckoutDeliveryMode(userId, cartId) {
        return this.http
            .delete(this.getClearDeliveryModeEndpoint(userId, cartId))
            .pipe(catchError((error) => {
            throw normalizeHttpError(error, this.logger);
        }), backOff({
            shouldRetry: isJaloError,
        }));
    }
    getClearDeliveryModeEndpoint(userId, cartId) {
        return this.occEndpoints.buildUrl('clearDeliveryMode', {
            urlParams: { userId, cartId },
        });
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: OccCheckoutDeliveryModesAdapter, deps: [{ token: i1.HttpClient }, { token: i2.OccEndpointsService }, { token: i2.ConverterService }], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: OccCheckoutDeliveryModesAdapter }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: OccCheckoutDeliveryModesAdapter, decorators: [{
            type: Injectable
        }], ctorParameters: () => [{ type: i1.HttpClient }, { type: i2.OccEndpointsService }, { type: i2.ConverterService }] });
