// @spartacus/asm/esm2022/customer-360/occ/adapters/occ-asm-customer-360.adapter.mjs
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { HttpHeaders } from '@angular/common/http';
import { Injectable, inject } from '@angular/core';
import { ASM_CUSTOMER_360_NORMALIZER, } from '@spartacus/asm/customer-360/core';
import { InterceptorUtil, LoggerService, USE_CUSTOMER_SUPPORT_AGENT_TOKEN, normalizeHttpError, } from '@spartacus/core';
import { throwError } from 'rxjs';
import { catchError } from 'rxjs/operators';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common/http";
import * as i2 from "@spartacus/core";
export class OccAsmCustomer360Adapter {
    constructor(http, occEndpointsService, converterService, baseSiteService) {
        this.http = http;
        this.occEndpointsService = occEndpointsService;
        this.converterService = converterService;
        this.baseSiteService = baseSiteService;
        this.logger = inject(LoggerService);
        this.baseSiteService
            .getActive()
            .subscribe((value) => (this.activeBaseSite = value));
    }
    getHeaders() {
        return InterceptorUtil.createHeader(USE_CUSTOMER_SUPPORT_AGENT_TOKEN, true, new HttpHeaders());
    }
    getAsmCustomer360Data(request) {
        const headers = InterceptorUtil.createHeader(USE_CUSTOMER_SUPPORT_AGENT_TOKEN, true, new HttpHeaders());
        const url = this.occEndpointsService.buildUrl('asmCustomer360', {
            urlParams: {
                baseSiteId: this.activeBaseSite,
                userId: request.options.userId ?? '',
            },
        }, {
            baseSite: false,
            prefix: false,
        });
        const requestBody = {
            customer360Queries: request.queries,
        };
        return this.http
            .post(url, requestBody, { headers })
            .pipe(catchError((error) => throwError(normalizeHttpError(error, this.logger))), this.converterService.pipeable(ASM_CUSTOMER_360_NORMALIZER));
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: OccAsmCustomer360Adapter, deps: [{ token: i1.HttpClient }, { token: i2.OccEndpointsService }, { token: i2.ConverterService }, { token: i2.BaseSiteService }], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: OccAsmCustomer360Adapter }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: OccAsmCustomer360Adapter, decorators: [{
            type: Injectable
        }], ctorParameters: () => [{ type: i1.HttpClient }, { type: i2.OccEndpointsService }, { type: i2.ConverterService }, { type: i2.BaseSiteService }] });
