// @spartacus/pdf-invoices/esm2022/components/pdf-invoices-components.module.mjs
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { ReactiveFormsModule } from '@angular/forms';
import { AuthGuard, I18nModule, provideDefaultConfig, UrlModule, } from '@spartacus/core';
import { DatePickerModule, IconModule, ListNavigationModule, MediaModule, SpinnerModule, } from '@spartacus/storefront';
import { InvoicesListComponent } from './invoices-list/invoices-list.component';
import * as i0 from "@angular/core";
export class PDFInvoicesComponentsModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: PDFInvoicesComponentsModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "17.0.5", ngImport: i0, type: PDFInvoicesComponentsModule, declarations: [InvoicesListComponent], imports: [CommonModule,
            DatePickerModule,
            I18nModule,
            ReactiveFormsModule,
            ListNavigationModule,
            UrlModule,
            IconModule,
            MediaModule,
            SpinnerModule], exports: [InvoicesListComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: PDFInvoicesComponentsModule, providers: [
            provideDefaultConfig({
                cmsComponents: {
                    AccountOrderDetailsPDFInvoicesComponent: {
                        component: InvoicesListComponent,
                        guards: [AuthGuard],
                    },
                },
            }),
        ], imports: [CommonModule,
            DatePickerModule,
            I18nModule,
            ReactiveFormsModule,
            ListNavigationModule,
            UrlModule,
            IconModule,
            MediaModule,
            SpinnerModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: PDFInvoicesComponentsModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        DatePickerModule,
                        I18nModule,
                        ReactiveFormsModule,
                        ListNavigationModule,
                        UrlModule,
                        IconModule,
                        MediaModule,
                        SpinnerModule,
                    ],
                    declarations: [InvoicesListComponent],
                    exports: [InvoicesListComponent],
                    providers: [
                        provideDefaultConfig({
                            cmsComponents: {
                                AccountOrderDetailsPDFInvoicesComponent: {
                                    component: InvoicesListComponent,
                                    guards: [AuthGuard],
                                },
                            },
                        }),
                    ],
                }]
        }] });
