// @spartacus/asm/esm2022/components/customer-emulation/customer-emulation.component.mjs
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { Component, ViewChild, } from '@angular/core';
import { BehaviorSubject, Subscription } from 'rxjs';
import { filter } from 'rxjs/operators';
import * as i0 from "@angular/core";
import * as i1 from "../services/asm-component.service";
import * as i2 from "@spartacus/user/account/root";
import * as i3 from "@spartacus/storefront";
import * as i4 from "@spartacus/core";
import * as i5 from "@angular/common";
import * as i6 from "../asm-bind-cart/asm-bind-cart.component";
export class CustomerEmulationComponent {
    constructor(asmComponentService, userAccountFacade, launchDialogService, featureModules) {
        this.asmComponentService = asmComponentService;
        this.userAccountFacade = userAccountFacade;
        this.launchDialogService = launchDialogService;
        this.featureModules = featureModules;
        this.isAsmCustomer360Configured = false;
        this.isAsmCustomer360Loaded$ = new BehaviorSubject(false);
        this.subscription = new Subscription();
    }
    ngOnInit() {
        this.isAsmCustomer360Configured =
            this.featureModules?.isConfigured('asmCustomer360');
        if (this.isAsmCustomer360Configured) {
            // trigger lazy loading of the Customer 360 feature:
            this.featureModules?.resolveFeature('asmCustomer360').subscribe(() => {
                this.isAsmCustomer360Loaded$.next(true);
            });
        }
        this.subscription.add(this.userAccountFacade.get().subscribe((user) => {
            if (user) {
                this.customer = user;
            }
        }));
        this.isCustomerEmulationSessionInProgress$ =
            this.asmComponentService.isCustomerEmulationSessionInProgress();
    }
    logoutCustomer() {
        this.asmComponentService.logoutCustomer();
    }
    openAsmCustomer360() {
        this.subscription.add(this.isAsmCustomer360Loaded$
            .pipe(filter((isReady) => Boolean(isReady)))
            .subscribe(() => {
            const data = { customer: this.customer };
            this.launchDialogService?.openDialogAndSubscribe("ASM_CUSTOMER_360" /* LAUNCH_CALLER.ASM_CUSTOMER_360 */, this.asmCustomer360LauncherElement, data);
            this.subscription.add(this.launchDialogService?.dialogClose
                .pipe(filter((result) => Boolean(result)))
                .subscribe((event) => {
                this.asmComponentService.handleAsmDialogAction(event);
            }));
        }));
    }
    ngOnDestroy() {
        this.subscription.unsubscribe();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: CustomerEmulationComponent, deps: [{ token: i1.AsmComponentService }, { token: i2.UserAccountFacade }, { token: i3.LaunchDialogService }, { token: i4.FeatureModulesService }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "17.0.5", type: CustomerEmulationComponent, selector: "cx-customer-emulation", viewQueries: [{ propertyName: "asmCustomer360LauncherElement", first: true, predicate: ["asmCustomer360Launcher"], descendants: true }], ngImport: i0, template: "<ng-container *ngIf=\"isCustomerEmulationSessionInProgress$ | async\">\n  <div class=\"cx-asm-customerInfo\">\n    <label class=\"cx-asm-name\">{{ customer?.name }}</label>\n    <label class=\"cx-asm-uid\">{{ customer?.uid }}</label>\n  </div>\n  <cx-asm-bind-cart></cx-asm-bind-cart>\n  <button\n    *ngIf=\"isAsmCustomer360Configured && customer\"\n    #asmCustomer360Launcher\n    class=\"cx-360-button\"\n    (click)=\"openAsmCustomer360()\"\n  >\n    {{ 'asm.asmCustomer360Button' | cxTranslate }}\n  </button>\n  <button formcontrolname=\"logoutCustomer\" (click)=\"logoutCustomer()\">\n    {{ 'asm.endSession' | cxTranslate }}\n  </button>\n</ng-container>\n", dependencies: [{ kind: "directive", type: i5.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { kind: "component", type: i6.AsmBindCartComponent, selector: "cx-asm-bind-cart" }, { kind: "pipe", type: i5.AsyncPipe, name: "async" }, { kind: "pipe", type: i4.TranslatePipe, name: "cxTranslate" }] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: CustomerEmulationComponent, decorators: [{
            type: Component,
            args: [{ selector: 'cx-customer-emulation', template: "<ng-container *ngIf=\"isCustomerEmulationSessionInProgress$ | async\">\n  <div class=\"cx-asm-customerInfo\">\n    <label class=\"cx-asm-name\">{{ customer?.name }}</label>\n    <label class=\"cx-asm-uid\">{{ customer?.uid }}</label>\n  </div>\n  <cx-asm-bind-cart></cx-asm-bind-cart>\n  <button\n    *ngIf=\"isAsmCustomer360Configured && customer\"\n    #asmCustomer360Launcher\n    class=\"cx-360-button\"\n    (click)=\"openAsmCustomer360()\"\n  >\n    {{ 'asm.asmCustomer360Button' | cxTranslate }}\n  </button>\n  <button formcontrolname=\"logoutCustomer\" (click)=\"logoutCustomer()\">\n    {{ 'asm.endSession' | cxTranslate }}\n  </button>\n</ng-container>\n" }]
        }], ctorParameters: () => [{ type: i1.AsmComponentService }, { type: i2.UserAccountFacade }, { type: i3.LaunchDialogService }, { type: i4.FeatureModulesService }], propDecorators: { asmCustomer360LauncherElement: [{
                type: ViewChild,
                args: ['asmCustomer360Launcher']
            }] } });
