// @spartacus/checkout/esm2022/base/core/connectors/checkout-payment/checkout-payment.connector.mjs
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { Injectable } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "./checkout-payment.adapter";
export class CheckoutPaymentConnector {
    constructor(adapter) {
        this.adapter = adapter;
    }
    createPaymentDetails(userId, cartId, paymentDetails) {
        return this.adapter.createPaymentDetails(userId, cartId, paymentDetails);
    }
    setPaymentDetails(userId, cartId, paymentDetailsId) {
        return this.adapter.setPaymentDetails(userId, cartId, paymentDetailsId);
    }
    getPaymentCardTypes() {
        return this.adapter.getPaymentCardTypes();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: CheckoutPaymentConnector, deps: [{ token: i1.CheckoutPaymentAdapter }], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: CheckoutPaymentConnector }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: CheckoutPaymentConnector, decorators: [{
            type: Injectable
        }], ctorParameters: () => [{ type: i1.CheckoutPaymentAdapter }] });
