// @spartacus/pdf-invoices/esm2022/occ/pdf-invoices-occ.module.mjs
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { provideDefaultConfig } from '@spartacus/core';
import { PDFInvoicesAdapter } from '@spartacus/pdf-invoices/core';
import { OccPDFInvoicesAdapter } from './adapters/occ-pdf-invoices.adapter';
import { defaultOccPDFInvoicesConfig } from './config/default-occ-pdf-invoices-config';
import * as i0 from "@angular/core";
export class PDFInvoicesOccModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: PDFInvoicesOccModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "17.0.5", ngImport: i0, type: PDFInvoicesOccModule, imports: [CommonModule] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: PDFInvoicesOccModule, providers: [
            provideDefaultConfig(defaultOccPDFInvoicesConfig),
            {
                provide: PDFInvoicesAdapter,
                useClass: OccPDFInvoicesAdapter,
            },
        ], imports: [CommonModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: PDFInvoicesOccModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    providers: [
                        provideDefaultConfig(defaultOccPDFInvoicesConfig),
                        {
                            provide: PDFInvoicesAdapter,
                            useClass: OccPDFInvoicesAdapter,
                        },
                    ],
                }]
        }] });
