// @spartacus/cart/esm2022/quick-order/components/quick-order-components.module.mjs
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { NgModule } from '@angular/core';
import { RouterModule } from '@angular/router';
import { QuickOrderOrderEntriesContextToken } from '@spartacus/cart/quick-order/root';
import { CartQuickOrderFormModule } from './cart-quick-order-form/cart-quick-order-form.module';
import { QuickOrderOrderEntriesContext } from './page-context/quick-order-order-entries.context';
import { QuickOrderListModule } from './quick-order/quick-order-list.module';
import * as i0 from "@angular/core";
export class QuickOrderComponentsModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: QuickOrderComponentsModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "17.0.5", ngImport: i0, type: QuickOrderComponentsModule, imports: [RouterModule, QuickOrderListModule, CartQuickOrderFormModule] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: QuickOrderComponentsModule, providers: [
            {
                provide: QuickOrderOrderEntriesContextToken,
                useExisting: QuickOrderOrderEntriesContext,
            },
        ], imports: [RouterModule, QuickOrderListModule, CartQuickOrderFormModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: QuickOrderComponentsModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [RouterModule, QuickOrderListModule, CartQuickOrderFormModule],
                    providers: [
                        {
                            provide: QuickOrderOrderEntriesContextToken,
                            useExisting: QuickOrderOrderEntriesContext,
                        },
                    ],
                }]
        }] });
