// @spartacus/storefront/esm2022/shared/components/popover/popover.service.mjs
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { Injectable, Optional, inject } from '@angular/core';
import { FeatureConfigService } from '@spartacus/core';
import { PopoverEvent } from './popover.model';
import * as i0 from "@angular/core";
export class PopoverService {
    constructor() {
        this.featureConfigService = inject(FeatureConfigService, {
            optional: true,
        });
    }
    /**
     * For a11y improvements method returns different `FocusConfig`
     * based on which event popover was triggered.
     */
    getFocusConfig(event, appendToBody) {
        let config = {};
        if (event === PopoverEvent.OPEN_BY_KEYBOARD && appendToBody) {
            config = {
                trap: true,
                block: true,
                focusOnEscape: false,
                autofocus: true,
            };
        }
        return config;
    }
    // TODO: (CXSPA-6594) - Remove feature flag next major release.
    setFocusOnElement(element, focusConfig, appendToBody) {
        if (this.featureConfigService?.isEnabled('a11yPopoverFocus')) {
            setTimeout(() => {
                element.nativeElement.focus();
            }, 0);
        }
        else {
            if (focusConfig && appendToBody) {
                element.nativeElement.focus();
            }
        }
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: PopoverService, deps: [], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: PopoverService, providedIn: 'root' }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: PopoverService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }], propDecorators: { featureConfigService: [{
                type: Optional
            }] } });
