// @spartacus/cart/esm2022/base/components/clear-cart/clear-cart-button/clear-cart.module.mjs
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { I18nModule, provideDefaultConfig } from '@spartacus/core';
import { ClearCartComponent } from './clear-cart.component';
import { ClearCartDialogModule } from '../clear-cart-dialog/clear-cart-dialog.module';
import { defaultClearCartLayoutConfig } from '../clear-cart-dialog/default-clear-cart-layout.config';
import * as i0 from "@angular/core";
export class ClearCartModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: ClearCartModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "17.0.5", ngImport: i0, type: ClearCartModule, declarations: [ClearCartComponent], imports: [CommonModule, I18nModule, ClearCartDialogModule], exports: [ClearCartComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: ClearCartModule, providers: [
            provideDefaultConfig({
                cmsComponents: {
                    ClearCartComponent: {
                        component: ClearCartComponent,
                    },
                },
            }),
            provideDefaultConfig(defaultClearCartLayoutConfig),
        ], imports: [CommonModule, I18nModule, ClearCartDialogModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: ClearCartModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [ClearCartComponent],
                    exports: [ClearCartComponent],
                    imports: [CommonModule, I18nModule, ClearCartDialogModule],
                    providers: [
                        provideDefaultConfig({
                            cmsComponents: {
                                ClearCartComponent: {
                                    component: ClearCartComponent,
                                },
                            },
                        }),
                        provideDefaultConfig(defaultClearCartLayoutConfig),
                    ],
                }]
        }] });
