// @spartacus/storefront/esm2022/cms-components/product/carousel/product-carousel/product-carousel.component.mjs
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { ChangeDetectionStrategy, Component } from '@angular/core';
import { filter, map } from 'rxjs/operators';
import * as i0 from "@angular/core";
import * as i1 from "../../../../cms-structure/page/model/cms-component-data";
import * as i2 from "@spartacus/core";
import * as i3 from "@angular/common";
import * as i4 from "../../../../shared/components/carousel/carousel.component";
import * as i5 from "../product-carousel-item/product-carousel-item.component";
export class ProductCarouselComponent {
    constructor(componentData, productService) {
        this.componentData = componentData;
        this.productService = productService;
        this.PRODUCT_SCOPE = ["list" /* ProductScope.LIST */, "stock" /* ProductScope.STOCK */];
        this.PRODUCT_SCOPE_ITEM = ["list_item" /* ProductScope.LIST_ITEM */];
        this.componentData$ = this.componentData.data$.pipe(filter((data) => Boolean(data)));
        /**
         * returns an Observable string for the title.
         */
        this.title$ = this.componentData$.pipe(map((data) => data.title));
        /**
         * Observable that holds an Array of Observables. This is done, so that
         * the component UI could consider to lazy load the UI components when they're
         * in the viewpoint.
         * If the inner component mapping exists (e.g. ProductAddToCartComponent), we aim to leverage the full scope;
         * otherwise, we prefer to minimize the scope for better performance.
         */
        this.items$ = this.componentData$.pipe(map((data) => {
            const componentMappingExist = !!data.composition?.inner?.length;
            const codes = data.productCodes?.trim().split(' ') ?? [];
            return { componentMappingExist, codes };
        }), map(({ componentMappingExist, codes }) => {
            const productScope = componentMappingExist
                ? [...this.PRODUCT_SCOPE]
                : [...this.PRODUCT_SCOPE_ITEM];
            return codes.map((code) => this.productService.get(code, productScope));
        }));
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: ProductCarouselComponent, deps: [{ token: i1.CmsComponentData }, { token: i2.ProductService }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "17.0.5", type: ProductCarouselComponent, selector: "cx-product-carousel", ngImport: i0, template: "<cx-carousel\n  *ngIf=\"items$ | async as items\"\n  role=\"region\"\n  [attr.aria-label]=\"\n    'productCarousel.carouselLabel' | cxTranslate: { title: title$ | async }\n  \"\n  [items]=\"items\"\n  [title]=\"title$ | async\"\n  [template]=\"carouselItem\"\n  itemWidth=\"285px\"\n>\n</cx-carousel>\n\n<ng-template #carouselItem let-item=\"item\">\n  <cx-product-carousel-item [item]=\"item\"></cx-product-carousel-item>\n</ng-template>\n", dependencies: [{ kind: "directive", type: i3.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { kind: "component", type: i4.CarouselComponent, selector: "cx-carousel", inputs: ["title", "items", "template", "itemWidth", "hideIndicators", "indicatorIcon", "previousIcon", "nextIcon"] }, { kind: "component", type: i5.ProductCarouselItemComponent, selector: "cx-product-carousel-item", inputs: ["item"] }, { kind: "pipe", type: i3.AsyncPipe, name: "async" }, { kind: "pipe", type: i2.TranslatePipe, name: "cxTranslate" }], changeDetection: i0.ChangeDetectionStrategy.OnPush }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: ProductCarouselComponent, decorators: [{
            type: Component,
            args: [{ selector: 'cx-product-carousel', changeDetection: ChangeDetectionStrategy.OnPush, template: "<cx-carousel\n  *ngIf=\"items$ | async as items\"\n  role=\"region\"\n  [attr.aria-label]=\"\n    'productCarousel.carouselLabel' | cxTranslate: { title: title$ | async }\n  \"\n  [items]=\"items\"\n  [title]=\"title$ | async\"\n  [template]=\"carouselItem\"\n  itemWidth=\"285px\"\n>\n</cx-carousel>\n\n<ng-template #carouselItem let-item=\"item\">\n  <cx-product-carousel-item [item]=\"item\"></cx-product-carousel-item>\n</ng-template>\n" }]
        }], ctorParameters: () => [{ type: i1.CmsComponentData }, { type: i2.ProductService }] });
