// @spartacus/asm/esm2022/root/facade/asm-customer-list.facade.mjs
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { Injectable } from '@angular/core';
import { facadeFactory } from '@spartacus/core';
import { ASM_FEATURE } from '../feature-name';
import * as i0 from "@angular/core";
export class AsmCustomerListFacade {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: AsmCustomerListFacade, deps: [], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: AsmCustomerListFacade, providedIn: 'root', useFactory: () => facadeFactory({
            facade: AsmCustomerListFacade,
            feature: ASM_FEATURE,
            methods: [
                'getCustomerLists',
                'getCustomerListsState',
                'customerListCustomersSearch',
                'getCustomerListCustomersSearchResults',
                'getCustomerListCustomersSearchResultsLoading',
                'customerListCustomersSearchReset',
                'getCustomerListCustomersSearchResultsError',
            ],
        }) }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: AsmCustomerListFacade, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                    useFactory: () => facadeFactory({
                        facade: AsmCustomerListFacade,
                        feature: ASM_FEATURE,
                        methods: [
                            'getCustomerLists',
                            'getCustomerListsState',
                            'customerListCustomersSearch',
                            'getCustomerListCustomersSearchResults',
                            'getCustomerListCustomersSearchResultsLoading',
                            'customerListCustomersSearchReset',
                            'getCustomerListCustomersSearchResultsError',
                        ],
                    }),
                }]
        }] });
