// @spartacus/asm/esm2022/root/facade/asm-create-customer.facade.mjs
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { Injectable } from '@angular/core';
import { facadeFactory } from '@spartacus/core';
import { ASM_FEATURE } from '../feature-name';
import * as i0 from "@angular/core";
export class AsmCreateCustomerFacade {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: AsmCreateCustomerFacade, deps: [], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: AsmCreateCustomerFacade, providedIn: 'root', useFactory: () => facadeFactory({
            facade: AsmCreateCustomerFacade,
            feature: ASM_FEATURE,
            methods: ['createCustomer'],
        }) }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: AsmCreateCustomerFacade, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                    useFactory: () => facadeFactory({
                        facade: AsmCreateCustomerFacade,
                        feature: ASM_FEATURE,
                        methods: ['createCustomer'],
                    }),
                }]
        }] });
