// @spartacus/setup/esm2022/ssr/logger/services/express-logger.service.mjs
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { Injectable, inject } from '@angular/core';
import { formatWithOptions } from 'util';
import { REQUEST } from '../../tokens/express.tokens';
import { EXPRESS_SERVER_LOGGER } from '../loggers';
import * as i0 from "@angular/core";
/**
 * Custom `LoggerService` used in ExpressJS.
 *
 * It converts the input arguments to a final string message similar as the native `console`
 * does (using the native function `format` from `node:util`) and passes this message
 * to a concrete server logger, used in ExpressJS.
 *
 * Besides the message, it also passes the current `request` of ExpressJS as an additional
 * context to the concrete server logger.
 */
export class ExpressLoggerService {
    constructor() {
        this.request = inject(REQUEST);
        this.serverLogger = inject(EXPRESS_SERVER_LOGGER);
    }
    log(...args) {
        this.serverLogger.log(this.formatLogMessage(...args), {
            request: this.request,
        });
    }
    warn(...args) {
        this.serverLogger.warn(this.formatLogMessage(...args), {
            request: this.request,
        });
    }
    error(...args) {
        this.serverLogger.error(this.formatLogMessage(...args), {
            request: this.request,
        });
    }
    info(...args) {
        this.serverLogger.info(this.formatLogMessage(...args), {
            request: this.request,
        });
    }
    debug(...args) {
        this.serverLogger.debug(this.formatLogMessage(...args), {
            request: this.request,
        });
    }
    formatLogMessage(message, ...optionalParams) {
        return formatWithOptions(
        // Prevent automatically breaking a long string message into multiple lines.
        // Otherwise, multi-line logs would be treated on the server as separate log
        { breakLength: Infinity }, message, ...optionalParams);
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: ExpressLoggerService, deps: [], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: ExpressLoggerService, providedIn: 'root' }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: ExpressLoggerService, decorators: [{
            type: Injectable,
            args: [{ providedIn: 'root' }]
        }] });
