// @spartacus/checkout/esm2022/base/occ/checkout-occ.module.mjs
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { CheckoutAdapter, CheckoutDeliveryAddressAdapter, CheckoutDeliveryModesAdapter, CheckoutPaymentAdapter, } from '@spartacus/checkout/base/core';
import { provideDefaultConfig } from '@spartacus/core';
import { OccCheckoutDeliveryAddressAdapter } from './adapters/occ-checkout-delivery-address.adapter';
import { OccCheckoutDeliveryModesAdapter } from './adapters/occ-checkout-delivery-modes.adapter';
import { OccCheckoutPaymentAdapter } from './adapters/occ-checkout-payment.adapter';
import { OccCheckoutAdapter } from './adapters/occ-checkout.adapter';
import { defaultOccCheckoutConfig } from './config/default-occ-checkout-config';
import * as i0 from "@angular/core";
export class CheckoutOccModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: CheckoutOccModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "17.0.5", ngImport: i0, type: CheckoutOccModule, imports: [CommonModule] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: CheckoutOccModule, providers: [
            provideDefaultConfig(defaultOccCheckoutConfig),
            {
                provide: CheckoutAdapter,
                useClass: OccCheckoutAdapter,
            },
            {
                provide: CheckoutDeliveryAddressAdapter,
                useClass: OccCheckoutDeliveryAddressAdapter,
            },
            {
                provide: CheckoutDeliveryModesAdapter,
                useClass: OccCheckoutDeliveryModesAdapter,
            },
            {
                provide: CheckoutPaymentAdapter,
                useClass: OccCheckoutPaymentAdapter,
            },
        ], imports: [CommonModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: CheckoutOccModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    providers: [
                        provideDefaultConfig(defaultOccCheckoutConfig),
                        {
                            provide: CheckoutAdapter,
                            useClass: OccCheckoutAdapter,
                        },
                        {
                            provide: CheckoutDeliveryAddressAdapter,
                            useClass: OccCheckoutDeliveryAddressAdapter,
                        },
                        {
                            provide: CheckoutDeliveryModesAdapter,
                            useClass: OccCheckoutDeliveryModesAdapter,
                        },
                        {
                            provide: CheckoutPaymentAdapter,
                            useClass: OccCheckoutPaymentAdapter,
                        },
                    ],
                }]
        }] });
