// @spartacus/asm/esm2022/components/asm-bind-cart/asm-bind-cart.component.mjs
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { ChangeDetectionStrategy, Component, ViewChild, } from '@angular/core';
import { FormControl, Validators } from '@angular/forms';
import { GlobalMessageType, OCC_CART_ID_CURRENT, } from '@spartacus/core';
import { BehaviorSubject, combineLatest, defer, EMPTY, iif, Subscription, } from 'rxjs';
import { concatMap, filter, finalize, map, shareReplay, take, tap, } from 'rxjs/operators';
import { BIND_CART_DIALOG_ACTION } from '../asm-bind-cart-dialog/asm-bind-cart-dialog.component';
import { SAVE_CART_DIALOG_ACTION } from '../asm-save-cart-dialog/asm-save-cart-dialog.component';
import * as i0 from "@angular/core";
import * as i1 from "@spartacus/core";
import * as i2 from "@spartacus/cart/base/root";
import * as i3 from "@spartacus/asm/root";
import * as i4 from "@spartacus/storefront";
import * as i5 from "@spartacus/cart/saved-cart/root";
import * as i6 from "../services/asm-component.service";
import * as i7 from "@angular/common";
import * as i8 from "@angular/forms";
import * as i9 from "../dot-spinner/dot-spinner.component";
export class AsmBindCartComponent {
    constructor(globalMessageService, activeCartFacade, multiCartFacade, asmBindCartFacade, launchDialogService, savedCartFacade, asmComponentService, routing) {
        this.globalMessageService = globalMessageService;
        this.activeCartFacade = activeCartFacade;
        this.multiCartFacade = multiCartFacade;
        this.asmBindCartFacade = asmBindCartFacade;
        this.launchDialogService = launchDialogService;
        this.savedCartFacade = savedCartFacade;
        this.asmComponentService = asmComponentService;
        this.routing = routing;
        this.activeCartValidator = (control) => {
            if (control.value === this.activeCartId) {
                return { activeCartError: true };
            }
            if (!!this.deepLinkCartId && control.value !== this.deepLinkCartId) {
                this.resetDeeplinkCart();
            }
            return null;
        };
        this.cartId = new FormControl('', [
            Validators.required,
            Validators.minLength(1),
            this.activeCartValidator,
        ]);
        this.loading$ = new BehaviorSubject(false);
        this.valid$ = this.cartId.statusChanges.pipe(map((status) => status === 'VALID'), shareReplay(1));
        this.activeCartId = '';
        this.deepLinkCartId = '';
        this.displayBindCartBtn$ = new BehaviorSubject(true);
        this.displaySaveCartBtn$ = new BehaviorSubject(false);
        this.subscription = new Subscription();
    }
    ngOnInit() {
        this.subscribeForDeeplinkCart();
        this.subscription.add(this.activeCartFacade.getActiveCartId().subscribe((response) => {
            this.activeCartId = response ?? '';
            this.cartId.setValue(this.deepLinkCartId || this.activeCartId);
        }));
    }
    resetInput() {
        if (!this.cartId.value) {
            this.cartId.setValue(this.activeCartId);
        }
    }
    /**
     * Bind the input cart number to the customer
     */
    bindCartToCustomer() {
        const anonymousCartId = this.cartId.value;
        const subscription = combineLatest([
            this.loading$.asObservable(),
            this.valid$,
        ])
            .pipe(take(1), filter(([loading, valid]) => !loading && valid), tap(() => this.loading$.next(true)), concatMap(() => this.activeCartFacade.getActive().pipe(map((cart) => cart.deliveryItemsQuantity ?? 0), take(1))), concatMap((cartItemCount) => iif(() => Boolean(this.activeCartId && cartItemCount), this.openDialog(this.activeCartId, anonymousCartId), this.simpleBindCart(anonymousCartId))), finalize(() => this.loading$.next(false)))
            .subscribe({
            next: () => {
                this.globalMessageService.add({ key: 'asm.bindCart.success' }, GlobalMessageType.MSG_TYPE_CONFIRMATION);
            },
            error: (error) => {
                this.globalMessageService.add(error.details?.[0].message ?? '', GlobalMessageType.MSG_TYPE_ERROR);
            },
        });
        this.subscription.add(subscription);
    }
    onSaveInactiveCart() {
        this.asmComponentService?.setShowDeeplinkCartInfoAlert(false);
        const customerId = this.asmComponentService?.getSearchParameter('customerId');
        this.multiCartFacade.loadCart({
            cartId: this.deepLinkCartId,
            userId: customerId,
        });
        this.multiCartFacade
            .getCartEntity(this.deepLinkCartId)
            .pipe(filter((state) => state.loading === false && state.success === true), take(1), map((state) => state.value), filter((cart) => !!cart))
            .subscribe((cart) => {
            this.openASMSaveCartDialog(cart);
        });
        this.afterCloseASMSaveCartDialog();
    }
    clearText() {
        this.cartId.setValue('');
        this.resetDeeplinkCart();
    }
    resetDeeplinkCart() {
        this.deepLinkCartId = '';
        this.displayBindCartBtn$.next(true);
        this.displaySaveCartBtn$.next(false);
        this.asmComponentService?.setShowDeeplinkCartInfoAlert(false);
    }
    ngOnDestroy() {
        this.subscription.unsubscribe();
    }
    /**
     * Binds cart on subscription and reloads cart
     */
    simpleBindCart(anonymousCartId) {
        return defer(() => this.asmBindCartFacade.bindCart(anonymousCartId)).pipe(tap(() => this.multiCartFacade.reloadCart(OCC_CART_ID_CURRENT)));
    }
    /**
     * Opens dialog and passes non-cancel result to select action
     */
    openDialog(activeCartId, anonymousCartId) {
        return defer(() => {
            this.launchDialogService.openDialogAndSubscribe("ASM_BIND_CART" /* LAUNCH_CALLER.ASM_BIND_CART */, this.bindToCartElemRef);
            return this.launchDialogService.dialogClose.pipe(filter((result) => Boolean(result)), take(1));
        }).pipe(filter((dialogResult) => Boolean(dialogResult)), concatMap((dialogResult) => {
            return this.selectBindAction(activeCartId, anonymousCartId, dialogResult);
        }));
    }
    selectBindAction(activeCartId, anonymousCartId, action) {
        switch (action) {
            case BIND_CART_DIALOG_ACTION.REPLACE:
                return this.replaceCart(activeCartId, anonymousCartId);
            case BIND_CART_DIALOG_ACTION.CANCEL:
            default:
                return EMPTY;
        }
    }
    replaceCart(previousActiveCartId, anonymousCartId) {
        return this.simpleBindCart(anonymousCartId).pipe(tap(() => {
            this.savedCartFacade.saveCart({
                cartId: previousActiveCartId,
                saveCartName: previousActiveCartId,
                // TODO(#12660): Remove default value once backend is updated
                saveCartDescription: '-',
            });
        }));
    }
    subscribeForDeeplinkCart() {
        this.subscription.add(this.asmComponentService
            ?.isEmulatedByDeepLink()
            .pipe(filter((emulated) => emulated &&
            !!this.asmComponentService?.getSearchParameter('cartId')))
            .subscribe(() => {
            const cartType = this.asmComponentService?.getSearchParameter('cartType');
            if (cartType === 'inactive' || cartType === 'active') {
                this.displayBindCartBtn$.next(false);
                this.displaySaveCartBtn$.next(cartType === 'inactive');
                this.deepLinkCartId = this.asmComponentService?.getSearchParameter('cartId');
                this.cartId.setValue(this.deepLinkCartId);
                this.asmComponentService?.setShowDeeplinkCartInfoAlert(true);
                this.asmComponentService?.handleDeepLinkNavigation();
            }
        }));
    }
    openASMSaveCartDialog(inactiveCart) {
        this.launchDialogService.openDialogAndSubscribe("ASM_SAVE_CART" /* LAUNCH_CALLER.ASM_SAVE_CART */, this.saveInactiveCartElemRef, inactiveCart);
    }
    afterCloseASMSaveCartDialog() {
        this.launchDialogService.dialogClose
            .pipe(filter((result) => result === SAVE_CART_DIALOG_ACTION.SAVE), take(1), tap(() => this.loading$.next(true)))
            .subscribe();
        this.savedCartFacade
            .getSaveCartProcessSuccess()
            .pipe(filter((success) => success), take(1), tap(() => this.loading$.next(false)))
            .subscribe(() => {
            this.goToSavedCartDetails(this.deepLinkCartId);
            this.displaySaveCartBtn$.next(false);
        });
        this.savedCartFacade
            .getSaveCartProcessError()
            .pipe(filter((error) => error), take(1), tap(() => this.loading$.next(false)))
            .subscribe();
    }
    goToSavedCartDetails(cartId) {
        this.routing?.go({
            cxRoute: 'savedCartsDetails',
            params: { savedCartId: cartId },
        });
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: AsmBindCartComponent, deps: [{ token: i1.GlobalMessageService }, { token: i2.ActiveCartFacade }, { token: i2.MultiCartFacade }, { token: i3.AsmBindCartFacade }, { token: i4.LaunchDialogService }, { token: i5.SavedCartFacade }, { token: i6.AsmComponentService }, { token: i1.RoutingService }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "17.0.5", type: AsmBindCartComponent, selector: "cx-asm-bind-cart", viewQueries: [{ propertyName: "bindToCartElemRef", first: true, predicate: ["bindToCart"], descendants: true }, { propertyName: "saveInactiveCartElemRef", first: true, predicate: ["saveInactiveCart"], descendants: true }], ngImport: i0, template: "<form>\n  <label for=\"cartNumber\">{{ 'asm.bindCart.cartNumber' | cxTranslate }} </label>\n  <div\n    role=\"search\"\n    [attr.aria-label]=\"'asm.bindCart.assignCartId' | cxTranslate\"\n    class=\"cx-asm-assignCart\"\n    [class.active]=\"valid$ | async\"\n    (click)=\"cartIdElement.focus()\"\n  >\n    <input\n      autocomplete=\"off\"\n      #cartIdElement\n      formcontrolname=\"cartNumber\"\n      [formControl]=\"cartId\"\n      (keydown.enter)=\"bindCartToCustomer()\"\n      (blur)=\"resetInput()\"\n      [attr.aria-label]=\"'asm.bindCart.enterCartId' | cxTranslate\"\n    />\n    <button\n      class=\"cx-asm-reset\"\n      [attr.aria-label]=\"'asm.bindCart.resetCartId' | cxTranslate\"\n      [class.visible]=\"cartId.value?.length > 0\"\n      (click)=\"clearText()\"\n    >\n      <cx-icon class=\"cx-icon fas fa-times-circle\"></cx-icon>\n    </button>\n  </div>\n  <button\n    #bindToCart\n    *ngIf=\"displayBindCartBtn$ | async\"\n    class=\"cx-asm-bindCartToCustomer\"\n    [disabled]=\"!(valid$ | async)\"\n    type=\"submit\"\n    [class.cx-asm-active]=\"valid$ | async\"\n    [class.cx-bind-loading]=\"loading$ | async\"\n    (click)=\"bindCartToCustomer()\"\n  >\n    <span [attr.aria-hidden]=\"loading$ | async\">\n      {{ 'asm.bindCart.bindCartToCustomer' | cxTranslate }}\n    </span>\n    <cx-dot-spinner\n      [attr.aria-hidden]=\"!(loading$ | async)\"\n      [attr.aria-label]=\"'common.loading' | cxTranslate\"\n    ></cx-dot-spinner>\n  </button>\n\n  <button\n    id=\"asm-save-inactive-cart-btn\"\n    #saveInactiveCart\n    *ngIf=\"displaySaveCartBtn$ | async\"\n    class=\"cx-asm-bindCartToCustomer cx-asm-active\"\n    type=\"submit\"\n    [class.cx-bind-loading]=\"loading$ | async\"\n    (click)=\"onSaveInactiveCart()\"\n  >\n    <span [attr.aria-hidden]=\"loading$ | async\">\n      {{ 'asm.saveCart.saveCartBtn' | cxTranslate }}\n    </span>\n    <cx-dot-spinner\n      [attr.aria-hidden]=\"!(loading$ | async)\"\n      [attr.aria-label]=\"'common.loading' | cxTranslate\"\n    ></cx-dot-spinner>\n  </button>\n</form>\n", dependencies: [{ kind: "directive", type: i7.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { kind: "directive", type: i8.ɵNgNoValidate, selector: "form:not([ngNoForm]):not([ngNativeValidate])" }, { kind: "directive", type: i8.DefaultValueAccessor, selector: "input:not([type=checkbox])[formControlName],textarea[formControlName],input:not([type=checkbox])[formControl],textarea[formControl],input:not([type=checkbox])[ngModel],textarea[ngModel],[ngDefaultControl]" }, { kind: "directive", type: i8.NgControlStatus, selector: "[formControlName],[ngModel],[formControl]" }, { kind: "directive", type: i8.NgControlStatusGroup, selector: "[formGroupName],[formArrayName],[ngModelGroup],[formGroup],form:not([ngNoForm]),[ngForm]" }, { kind: "directive", type: i8.FormControlDirective, selector: "[formControl]", inputs: ["formControl", "disabled", "ngModel"], outputs: ["ngModelChange"], exportAs: ["ngForm"] }, { kind: "component", type: i4.IconComponent, selector: "cx-icon,[cxIcon]", inputs: ["cxIcon", "type"] }, { kind: "directive", type: i8.NgForm, selector: "form:not([ngNoForm]):not([formGroup]),ng-form,[ngForm]", inputs: ["ngFormOptions"], outputs: ["ngSubmit"], exportAs: ["ngForm"] }, { kind: "component", type: i9.DotSpinnerComponent, selector: "cx-dot-spinner" }, { kind: "pipe", type: i7.AsyncPipe, name: "async" }, { kind: "pipe", type: i1.TranslatePipe, name: "cxTranslate" }], changeDetection: i0.ChangeDetectionStrategy.OnPush }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: AsmBindCartComponent, decorators: [{
            type: Component,
            args: [{ selector: 'cx-asm-bind-cart', changeDetection: ChangeDetectionStrategy.OnPush, template: "<form>\n  <label for=\"cartNumber\">{{ 'asm.bindCart.cartNumber' | cxTranslate }} </label>\n  <div\n    role=\"search\"\n    [attr.aria-label]=\"'asm.bindCart.assignCartId' | cxTranslate\"\n    class=\"cx-asm-assignCart\"\n    [class.active]=\"valid$ | async\"\n    (click)=\"cartIdElement.focus()\"\n  >\n    <input\n      autocomplete=\"off\"\n      #cartIdElement\n      formcontrolname=\"cartNumber\"\n      [formControl]=\"cartId\"\n      (keydown.enter)=\"bindCartToCustomer()\"\n      (blur)=\"resetInput()\"\n      [attr.aria-label]=\"'asm.bindCart.enterCartId' | cxTranslate\"\n    />\n    <button\n      class=\"cx-asm-reset\"\n      [attr.aria-label]=\"'asm.bindCart.resetCartId' | cxTranslate\"\n      [class.visible]=\"cartId.value?.length > 0\"\n      (click)=\"clearText()\"\n    >\n      <cx-icon class=\"cx-icon fas fa-times-circle\"></cx-icon>\n    </button>\n  </div>\n  <button\n    #bindToCart\n    *ngIf=\"displayBindCartBtn$ | async\"\n    class=\"cx-asm-bindCartToCustomer\"\n    [disabled]=\"!(valid$ | async)\"\n    type=\"submit\"\n    [class.cx-asm-active]=\"valid$ | async\"\n    [class.cx-bind-loading]=\"loading$ | async\"\n    (click)=\"bindCartToCustomer()\"\n  >\n    <span [attr.aria-hidden]=\"loading$ | async\">\n      {{ 'asm.bindCart.bindCartToCustomer' | cxTranslate }}\n    </span>\n    <cx-dot-spinner\n      [attr.aria-hidden]=\"!(loading$ | async)\"\n      [attr.aria-label]=\"'common.loading' | cxTranslate\"\n    ></cx-dot-spinner>\n  </button>\n\n  <button\n    id=\"asm-save-inactive-cart-btn\"\n    #saveInactiveCart\n    *ngIf=\"displaySaveCartBtn$ | async\"\n    class=\"cx-asm-bindCartToCustomer cx-asm-active\"\n    type=\"submit\"\n    [class.cx-bind-loading]=\"loading$ | async\"\n    (click)=\"onSaveInactiveCart()\"\n  >\n    <span [attr.aria-hidden]=\"loading$ | async\">\n      {{ 'asm.saveCart.saveCartBtn' | cxTranslate }}\n    </span>\n    <cx-dot-spinner\n      [attr.aria-hidden]=\"!(loading$ | async)\"\n      [attr.aria-label]=\"'common.loading' | cxTranslate\"\n    ></cx-dot-spinner>\n  </button>\n</form>\n" }]
        }], ctorParameters: () => [{ type: i1.GlobalMessageService }, { type: i2.ActiveCartFacade }, { type: i2.MultiCartFacade }, { type: i3.AsmBindCartFacade }, { type: i4.LaunchDialogService }, { type: i5.SavedCartFacade }, { type: i6.AsmComponentService }, { type: i1.RoutingService }], propDecorators: { bindToCartElemRef: [{
                type: ViewChild,
                args: ['bindToCart']
            }], saveInactiveCartElemRef: [{
                type: ViewChild,
                args: ['saveInactiveCart']
            }] } });
