// @spartacus/cart/esm2022/base/root/events/cart.events.mjs
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { CxEvent } from '@spartacus/core';
/**
 * Base cart event. Most cart events should have these properties.
 */
export class CartEvent extends CxEvent {
}
export class CreateCartEvent extends CartEvent {
    /**
     * Event's type
     */
    static { this.type = 'CreateCartEvent'; }
}
export class CreateCartSuccessEvent extends CartEvent {
    /**
     * Event's type
     */
    static { this.type = 'CreateCartSuccessEvent'; }
}
export class CreateCartFailEvent extends CartEvent {
    /**
     * Event's type
     */
    static { this.type = 'CreateCartFailEvent'; }
}
export class CartAddEntryEvent extends CartEvent {
    /**
     * Event's type
     */
    static { this.type = 'CartAddEntryEvent'; }
}
export class CartAddEntrySuccessEvent extends CartEvent {
    /**
     * Event's type
     */
    static { this.type = 'CartAddEntrySuccessEvent'; }
}
export class CartAddEntryFailEvent extends CartEvent {
    /**
     * Event's type
     */
    static { this.type = 'CartAddEntryFailEvent'; }
}
export class CartRemoveEntryFailEvent extends CartEvent {
    /**
     * Event's type
     */
    static { this.type = 'CartRemoveEntryFailEvent'; }
}
export class CartRemoveEntrySuccessEvent extends CartEvent {
    /**
     * Event's type
     */
    static { this.type = 'CartRemoveEntrySuccessEvent'; }
}
export class CartUpdateEntrySuccessEvent extends CartEvent {
    /**
     * Event's type
     */
    static { this.type = 'CartUpdateEntrySuccessEvent'; }
}
export class CartUpdateEntryFailEvent extends CartEvent {
    /**
     * Event's type
     */
    static { this.type = 'CartUpdateEntryFailEvent'; }
}
export class CartUiEventAddToCart extends CxEvent {
    /**
     * Event's type
     */
    static { this.type = 'CartUiEventAddToCart'; }
}
/**
 * Fired when the cart has been successfully merged.
 */
export class MergeCartSuccessEvent extends CartEvent {
    /**
     * Event's type
     */
    static { this.type = 'MergeCartSuccessEvent'; }
}
/**
 * Triggers the loading of the cart.
 */
export class LoadCartEvent extends CartEvent {
    /**
     * Event's type
     */
    static { this.type = 'LoadCartEvent'; }
}
/** Removes the cart. */
export class RemoveCartEvent extends CartEvent {
    /**
     * Event's type
     */
    static { this.type = 'RemoveCartEvent'; }
}
/**
 * Fired when the cart has been deleted.
 */
export class DeleteCartEvent extends CartEvent {
    /**
     * Event's type
     */
    static { this.type = 'DeleteCartEvent'; }
}
export class DeleteCartSuccessEvent extends CartEvent {
    /**
     * Event's type
     */
    static { this.type = 'DeleteCartSuccessEvent'; }
}
export class DeleteCartFailEvent extends CartEvent {
    /**
     * Event's type
     */
    static { this.type = 'DeleteCartFailEvent'; }
}
export class AddCartVoucherEvent extends CartEvent {
    /**
     * Event's type
     */
    static { this.type = 'AddCartVoucherEvent'; }
}
export class AddCartVoucherSuccessEvent extends AddCartVoucherEvent {
    /**
     * Event's type
     */
    static { this.type = 'AddCartVoucherSuccessEvent'; }
}
export class AddCartVoucherFailEvent extends AddCartVoucherEvent {
    /**
     * Event's type
     */
    static { this.type = 'AddCartVoucherFailEvent'; }
}
export class RemoveCartVoucherEvent extends CartEvent {
    /**
     * Event's type
     */
    static { this.type = 'RemoveCartVoucherEvent'; }
}
export class RemoveCartVoucherSuccessEvent extends RemoveCartVoucherEvent {
    /**
     * Event's type
     */
    static { this.type = 'RemoveCartVoucherSuccessEvent'; }
}
export class RemoveCartVoucherFailEvent extends RemoveCartVoucherEvent {
    /**
     * Event's type
     */
    static { this.type = 'RemoveCartVoucherFailEvent'; }
}
