// @spartacus/storefront/esm2022/shared/components/card/card.module.mjs
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { FeaturesConfigModule, I18nModule } from '@spartacus/core';
import { IconModule } from '../../../cms-components/misc/icon/icon.module';
import { KeyboardFocusModule } from '../../../layout/a11y/keyboard-focus/keyboard-focus.module';
import { AtMessageModule } from '../assistive-technology-message/assistive-technology-message.module';
import { TruncateTextPopoverModule } from '../truncate-text-popover/truncate-text-popover.module';
import { CardComponent } from './card.component';
import * as i0 from "@angular/core";
export class CardModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: CardModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "17.0.5", ngImport: i0, type: CardModule, declarations: [CardComponent], imports: [CommonModule,
            AtMessageModule,
            I18nModule,
            IconModule,
            TruncateTextPopoverModule,
            KeyboardFocusModule,
            FeaturesConfigModule], exports: [CardComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: CardModule, imports: [CommonModule,
            AtMessageModule,
            I18nModule,
            IconModule,
            TruncateTextPopoverModule,
            KeyboardFocusModule,
            FeaturesConfigModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: CardModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        AtMessageModule,
                        I18nModule,
                        IconModule,
                        TruncateTextPopoverModule,
                        KeyboardFocusModule,
                        FeaturesConfigModule,
                    ],
                    declarations: [CardComponent],
                    exports: [CardComponent],
                }]
        }] });
