// @spartacus/cart/esm2022/base/components/cart-shared/cart-shared.module.mjs
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { ReactiveFormsModule } from '@angular/forms';
import { RouterModule } from '@angular/router';
import { CartOutlets } from '@spartacus/cart/base/root';
import { FeaturesConfigModule, I18nModule, UrlModule } from '@spartacus/core';
import { AtMessageModule, IconModule, ItemCounterModule, MediaModule, OutletModule, PromotionsModule, provideOutlet, } from '@spartacus/storefront';
import { AddToCartModule } from '../add-to-cart/add-to-cart.module';
import { CartCouponModule } from '../cart-coupon/cart-coupon.module';
import { CartItemValidationWarningModule } from '../validation/cart-item-warning/cart-item-validation-warning.module';
import { CartItemListRowComponent } from './cart-item-list-row/cart-item-list-row.component';
import { CartItemListComponent } from './cart-item-list/cart-item-list.component';
import { CartItemComponent } from './cart-item/cart-item.component';
import { OrderSummaryComponent } from './order-summary/order-summary.component';
import * as i0 from "@angular/core";
export class CartSharedModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: CartSharedModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "17.0.5", ngImport: i0, type: CartSharedModule, declarations: [CartItemComponent,
            OrderSummaryComponent,
            CartItemListComponent,
            CartItemListRowComponent], imports: [AtMessageModule,
            CartCouponModule,
            CartItemValidationWarningModule,
            CommonModule,
            I18nModule,
            IconModule,
            ItemCounterModule,
            MediaModule,
            OutletModule,
            PromotionsModule,
            ReactiveFormsModule,
            RouterModule,
            UrlModule,
            AddToCartModule,
            FeaturesConfigModule], exports: [CartItemComponent,
            CartItemListRowComponent,
            CartItemListComponent,
            OrderSummaryComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: CartSharedModule, providers: [
            provideOutlet({
                id: CartOutlets.ORDER_SUMMARY,
                component: OrderSummaryComponent,
            }),
            provideOutlet({
                id: CartOutlets.CART_ITEM_LIST,
                component: CartItemListComponent,
            }),
        ], imports: [AtMessageModule,
            CartCouponModule,
            CartItemValidationWarningModule,
            CommonModule,
            I18nModule,
            IconModule,
            ItemCounterModule,
            MediaModule,
            OutletModule,
            PromotionsModule,
            ReactiveFormsModule,
            RouterModule,
            UrlModule,
            AddToCartModule,
            FeaturesConfigModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: CartSharedModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        AtMessageModule,
                        CartCouponModule,
                        CartItemValidationWarningModule,
                        CommonModule,
                        I18nModule,
                        IconModule,
                        ItemCounterModule,
                        MediaModule,
                        OutletModule,
                        PromotionsModule,
                        ReactiveFormsModule,
                        RouterModule,
                        UrlModule,
                        AddToCartModule,
                        FeaturesConfigModule,
                    ],
                    providers: [
                        provideOutlet({
                            id: CartOutlets.ORDER_SUMMARY,
                            component: OrderSummaryComponent,
                        }),
                        provideOutlet({
                            id: CartOutlets.CART_ITEM_LIST,
                            component: CartItemListComponent,
                        }),
                    ],
                    declarations: [
                        CartItemComponent,
                        OrderSummaryComponent,
                        CartItemListComponent,
                        CartItemListRowComponent,
                    ],
                    exports: [
                        CartItemComponent,
                        CartItemListRowComponent,
                        CartItemListComponent,
                        OrderSummaryComponent,
                    ],
                }]
        }] });
