// @spartacus/pdf-invoices/esm2022/core/pdf-invoices-core.module.mjs
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { NgModule } from '@angular/core';
import { HttpErrorHandler } from '@spartacus/core';
import { PDFInvoicesFacade } from '@spartacus/pdf-invoices/root';
import { PDFInvoicesConnector } from './connectors/pdf-invoices.connector';
import { PDFInvoicesBadRequestHandler } from './http-interceptors/bad-request/pdf-invoices-badrequest.handler';
import { PDFInvoicesService } from './services';
import * as i0 from "@angular/core";
export class PDFInvoicesCoreModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: PDFInvoicesCoreModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "17.0.5", ngImport: i0, type: PDFInvoicesCoreModule }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: PDFInvoicesCoreModule, providers: [
            PDFInvoicesService,
            {
                provide: PDFInvoicesFacade,
                useExisting: PDFInvoicesService,
            },
            {
                provide: HttpErrorHandler,
                useExisting: PDFInvoicesBadRequestHandler,
                multi: true,
            },
            PDFInvoicesConnector,
        ] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: PDFInvoicesCoreModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [],
                    providers: [
                        PDFInvoicesService,
                        {
                            provide: PDFInvoicesFacade,
                            useExisting: PDFInvoicesService,
                        },
                        {
                            provide: HttpErrorHandler,
                            useExisting: PDFInvoicesBadRequestHandler,
                            multi: true,
                        },
                        PDFInvoicesConnector,
                    ],
                }]
        }] });
