// @spartacus/cart/esm2022/base/components/cart-shared/cart-item-list/cart-item-list.component.mjs
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { ChangeDetectionStrategy, Component, Input, Optional, } from '@angular/core';
import { UntypedFormControl, UntypedFormGroup } from '@angular/forms';
import { PromotionLocation, CartOutlets, } from '@spartacus/cart/base/root';
import { Subscription } from 'rxjs';
import { map, startWith, tap } from 'rxjs/operators';
import * as i0 from "@angular/core";
import * as i1 from "@spartacus/cart/base/root";
import * as i2 from "@spartacus/core";
import * as i3 from "@spartacus/storefront";
import * as i4 from "@angular/common";
import * as i5 from "../cart-item-list-row/cart-item-list-row.component";
export class CartItemListComponent {
    set items(items) {
        this._setItems(items);
    }
    get items() {
        return this._items;
    }
    set setLoading(value) {
        if (!this.readonly) {
            // Whenever the cart is loading, we disable the complete form
            // to avoid any user interaction with the cart.
            value
                ? this.form.disable({ emitEvent: false })
                : this.form.enable({ emitEvent: false });
            this.cd.markForCheck();
        }
    }
    constructor(activeCartService, selectiveCartService, userIdService, multiCartService, cd, outlet) {
        this.activeCartService = activeCartService;
        this.selectiveCartService = selectiveCartService;
        this.userIdService = userIdService;
        this.multiCartService = multiCartService;
        this.cd = cd;
        this.outlet = outlet;
        this.subscription = new Subscription();
        this.readonly = false;
        this.hasHeader = true;
        this.options = {
            isSaveForLater: false,
            optionalBtn: null,
            displayAddToCart: false,
        };
        this._items = [];
        this.form = new UntypedFormGroup({});
        this.promotionLocation = PromotionLocation.ActiveCart;
        this.CartOutlets = CartOutlets;
    }
    ngOnInit() {
        this.subscription.add(this.getInputsFromContext());
        this.subscription.add(this.userIdService
            ?.getUserId()
            .subscribe((userId) => (this.userId = userId)));
    }
    _setItems(items, options) {
        this.resolveItems(items, options);
        this.createForm();
    }
    getInputsFromContext() {
        return this.outlet?.context$.subscribe((context) => {
            let contextRequiresRerender = false;
            if (context.readonly !== undefined) {
                contextRequiresRerender = this.readonly !== context.readonly;
                this.readonly = context.readonly;
            }
            if (context.hasHeader !== undefined) {
                this.hasHeader = context.hasHeader;
            }
            if (context.options !== undefined) {
                this.options = context.options;
            }
            if (context.cartId !== undefined) {
                this.cartId = context.cartId;
            }
            if (context.promotionLocation !== undefined) {
                this.promotionLocation = context.promotionLocation;
            }
            if (context.cartIsLoading !== undefined) {
                this.setLoading = context.cartIsLoading;
            }
            if (context.items !== undefined) {
                this.cd.markForCheck();
                this._setItems(context.items, {
                    forceRerender: contextRequiresRerender,
                });
            }
        });
    }
    /**
     * Resolves items passed to component input and updates 'items' field
     */
    resolveItems(items, options) {
        if (!items) {
            this._items = [];
            return;
        }
        // The items we're getting from the input do not have a consistent model.
        // In case of a `consignmentEntry`, we need to normalize the data from the orderEntry.
        if (items.every((item) => item.hasOwnProperty('orderEntry'))) {
            this.normalizeConsignmentEntries(items);
        }
        else {
            this.rerenderChangedItems(items, options);
        }
    }
    normalizeConsignmentEntries(items) {
        this._items = items.map((consignmentEntry) => {
            const entry = Object.assign({}, consignmentEntry.orderEntry);
            entry.quantity = consignmentEntry.quantity;
            return entry;
        });
    }
    /**
     * We'd like to avoid the unnecessary re-renders of unchanged cart items after the data reload.
     * OCC cart entries don't have any unique identifier that we could use in Angular `trackBy`.
     * So we update each array element to the new object only when it's any different to the previous one.
     */
    rerenderChangedItems(items, options) {
        let offset = 0;
        for (let i = 0; i - offset < Math.max(items.length, this._items.length); i++) {
            const index = i - offset;
            if (options?.forceRerender ||
                JSON.stringify(this._items?.[index]) !== JSON.stringify(items[index])) {
                if (this._items[index]) {
                    this.form?.removeControl(this.getControlName(this._items[index]));
                }
                if (!items[index]) {
                    this._items.splice(index, 1);
                    offset++;
                }
                else {
                    this._items[index] = items[index];
                }
            }
        }
    }
    /**
     * Creates form models for list items
     */
    createForm() {
        this._items.forEach((item) => {
            const controlName = this.getControlName(item);
            const control = this.form.get(controlName);
            if (control) {
                if (control.get('quantity')?.value !== item.quantity) {
                    control.patchValue({ quantity: item.quantity }, { emitEvent: false });
                }
            }
            else {
                const group = new UntypedFormGroup({
                    entryNumber: new UntypedFormControl(item.entryNumber),
                    quantity: new UntypedFormControl(item.quantity, { updateOn: 'blur' }),
                });
                this.form.addControl(controlName, group);
            }
            // If we disable form group before adding, disabled status will reset
            // Which forces us to disable control after including to form object
            if (!item.updateable || this.readonly) {
                this.form.controls[controlName].disable();
            }
        });
    }
    getControlName(item) {
        return item.entryNumber?.toString() || '';
    }
    removeEntry(item) {
        if (this.options.isSaveForLater) {
            this.selectiveCartService.removeEntry(item);
        }
        else if (this.cartId && this.userId) {
            this.multiCartService.removeEntry(this.userId, this.cartId, item.entryNumber);
        }
        else {
            this.activeCartService.removeEntry(item);
        }
        delete this.form.controls[this.getControlName(item)];
    }
    getControl(item) {
        return this.form.get(this.getControlName(item))?.valueChanges.pipe(
        // eslint-disable-next-line import/no-deprecated
        startWith(null), tap((value) => {
            if (item.updateable && value && !this.readonly) {
                if (this.options.isSaveForLater) {
                    this.selectiveCartService.updateEntry(value.entryNumber, value.quantity);
                }
                else if (this.cartId && this.userId) {
                    this.multiCartService.updateEntry(this.userId, this.cartId, value.entryNumber, value.quantity);
                }
                else {
                    this.activeCartService.updateEntry(value.entryNumber, value.quantity);
                }
            }
        }), map(() => this.form.get(this.getControlName(item))));
    }
    ngOnDestroy() {
        this.subscription?.unsubscribe();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: CartItemListComponent, deps: [{ token: i1.ActiveCartFacade }, { token: i1.SelectiveCartFacade }, { token: i2.UserIdService }, { token: i1.MultiCartFacade }, { token: i0.ChangeDetectorRef }, { token: i3.OutletContextData, optional: true }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "17.0.5", type: CartItemListComponent, selector: "cx-cart-item-list", inputs: { readonly: "readonly", hasHeader: "hasHeader", options: "options", cartId: "cartId", items: "items", promotionLocation: "promotionLocation", setLoading: ["cartIsLoading", "setLoading"] }, ngImport: i0, template: "<table role=\"table\">\n  <caption class=\"cx-visually-hidden\">\n    {{\n      'cartItems.caption' | cxTranslate\n    }}\n  </caption>\n  <thead *ngIf=\"hasHeader\">\n    <tr role=\"row\" class=\"cx-item-list-header\">\n      <th role=\"columnheader\" class=\"cx-item-list-desc\">\n        {{ 'cartItems.item' | cxTranslate }}\n      </th>\n      <th role=\"columnheader\" class=\"cx-item-list-qty\">\n        {{ 'cartItems.quantity' | cxTranslate }}\n      </th>\n      <ng-container *ngIf=\"options.isSaveForLater; else totalHeader\">\n        <th role=\"columnheader\" class=\"cx-item-list-total\">\n          {{ 'saveForLaterItems.stock' | cxTranslate }}\n        </th>\n      </ng-container>\n      <ng-container\n        *ngIf=\"!readonly || options.isSaveForLater || options.displayAddToCart\"\n      >\n        <th role=\"columnheader\" class=\"cx-item-list-actions\">\n          {{ 'cartItems.actions' | cxTranslate }}\n        </th>\n      </ng-container>\n    </tr>\n  </thead>\n  <tbody class=\"cx-item-list-items\">\n    <ng-container *ngFor=\"let item of items; let i = index\">\n      <ng-container\n        *ngIf=\"getControl(item) | async as control\"\n        [class.is-changed]=\"control.get('quantity').disabled\"\n      >\n        <tr\n          cx-cart-item-list-row\n          role=\"row\"\n          class=\"cx-item-list-row\"\n          [item]=\"item\"\n          [quantityControl]=\"control.get('quantity')\"\n          [readonly]=\"readonly\"\n          [promotionLocation]=\"promotionLocation\"\n          [options]=\"options\"\n        ></tr>\n      </ng-container>\n    </ng-container>\n  </tbody>\n</table>\n\n<ng-template #totalHeader>\n  <th role=\"columnheader\" class=\"cx-item-list-total\">\n    {{ 'cartItems.total' | cxTranslate }}\n  </th>\n</ng-template>\n", dependencies: [{ kind: "directive", type: i4.NgForOf, selector: "[ngFor][ngForOf]", inputs: ["ngForOf", "ngForTrackBy", "ngForTemplate"] }, { kind: "directive", type: i4.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { kind: "component", type: i5.CartItemListRowComponent, selector: "[cx-cart-item-list-row], cx-cart-item-list-row" }, { kind: "pipe", type: i4.AsyncPipe, name: "async" }, { kind: "pipe", type: i2.TranslatePipe, name: "cxTranslate" }], changeDetection: i0.ChangeDetectionStrategy.OnPush }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: CartItemListComponent, decorators: [{
            type: Component,
            args: [{ selector: 'cx-cart-item-list', changeDetection: ChangeDetectionStrategy.OnPush, template: "<table role=\"table\">\n  <caption class=\"cx-visually-hidden\">\n    {{\n      'cartItems.caption' | cxTranslate\n    }}\n  </caption>\n  <thead *ngIf=\"hasHeader\">\n    <tr role=\"row\" class=\"cx-item-list-header\">\n      <th role=\"columnheader\" class=\"cx-item-list-desc\">\n        {{ 'cartItems.item' | cxTranslate }}\n      </th>\n      <th role=\"columnheader\" class=\"cx-item-list-qty\">\n        {{ 'cartItems.quantity' | cxTranslate }}\n      </th>\n      <ng-container *ngIf=\"options.isSaveForLater; else totalHeader\">\n        <th role=\"columnheader\" class=\"cx-item-list-total\">\n          {{ 'saveForLaterItems.stock' | cxTranslate }}\n        </th>\n      </ng-container>\n      <ng-container\n        *ngIf=\"!readonly || options.isSaveForLater || options.displayAddToCart\"\n      >\n        <th role=\"columnheader\" class=\"cx-item-list-actions\">\n          {{ 'cartItems.actions' | cxTranslate }}\n        </th>\n      </ng-container>\n    </tr>\n  </thead>\n  <tbody class=\"cx-item-list-items\">\n    <ng-container *ngFor=\"let item of items; let i = index\">\n      <ng-container\n        *ngIf=\"getControl(item) | async as control\"\n        [class.is-changed]=\"control.get('quantity').disabled\"\n      >\n        <tr\n          cx-cart-item-list-row\n          role=\"row\"\n          class=\"cx-item-list-row\"\n          [item]=\"item\"\n          [quantityControl]=\"control.get('quantity')\"\n          [readonly]=\"readonly\"\n          [promotionLocation]=\"promotionLocation\"\n          [options]=\"options\"\n        ></tr>\n      </ng-container>\n    </ng-container>\n  </tbody>\n</table>\n\n<ng-template #totalHeader>\n  <th role=\"columnheader\" class=\"cx-item-list-total\">\n    {{ 'cartItems.total' | cxTranslate }}\n  </th>\n</ng-template>\n" }]
        }], ctorParameters: () => [{ type: i1.ActiveCartFacade }, { type: i1.SelectiveCartFacade }, { type: i2.UserIdService }, { type: i1.MultiCartFacade }, { type: i0.ChangeDetectorRef }, { type: i3.OutletContextData, decorators: [{
                    type: Optional
                }] }], propDecorators: { readonly: [{
                type: Input
            }], hasHeader: [{
                type: Input
            }], options: [{
                type: Input
            }], cartId: [{
                type: Input
            }], items: [{
                type: Input,
                args: ['items']
            }], promotionLocation: [{
                type: Input
            }], setLoading: [{
                type: Input,
                args: ['cartIsLoading']
            }] } });
