// @spartacus/cart/esm2022/base/root/models/cart.model.mjs
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
export var PromotionLocation;
(function (PromotionLocation) {
    PromotionLocation["ActiveCart"] = "CART";
    PromotionLocation["Checkout"] = "CHECKOUT";
    PromotionLocation["Order"] = "ORDER";
    PromotionLocation["SaveForLater"] = "SAVE_FOR_LATER";
    PromotionLocation["SavedCart"] = "SAVED_CART";
})(PromotionLocation || (PromotionLocation = {}));
/**
 * The possible types of 'abstract orders'.
 */
export var AbstractOrderType;
(function (AbstractOrderType) {
    /**
     * Active cart
     */
    AbstractOrderType["CART"] = "Cart";
    AbstractOrderType["ORDER"] = "Order";
    AbstractOrderType["QUOTE"] = "Quote";
    AbstractOrderType["SAVED_CART"] = "SavedCart";
})(AbstractOrderType || (AbstractOrderType = {}));
export var CartType;
(function (CartType) {
    CartType["ACTIVE"] = "Active";
    CartType["WISH_LIST"] = "WishList";
    CartType["SELECTIVE"] = "Selective";
    CartType["NEW_CREATED"] = "NewCreated";
})(CartType || (CartType = {}));
export var CartValidationStatusCode;
(function (CartValidationStatusCode) {
    CartValidationStatusCode["NO_STOCK"] = "noStock";
    CartValidationStatusCode["LOW_STOCK"] = "lowStock";
    CartValidationStatusCode["REVIEW_CONFIGURATION"] = "reviewConfiguration";
    CartValidationStatusCode["PRICING_ERROR"] = "pricingError";
    CartValidationStatusCode["UNRESOLVABLE_ISSUES"] = "unresolvableIssues";
})(CartValidationStatusCode || (CartValidationStatusCode = {}));
