// @spartacus/storefinder/esm2022/root/facade/store-finder.facade.mjs
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { Injectable } from '@angular/core';
import { facadeFactory, } from '@spartacus/core';
import { STORE_FINDER_FEATURE } from '../feature-name';
import * as i0 from "@angular/core";
/**
 * Store the Point of Service a user wants to collect a product from before it is added to the cart.
 */
export class StoreFinderFacade {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: StoreFinderFacade, deps: [], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: StoreFinderFacade, providedIn: 'root', useFactory: () => facadeFactory({
            facade: StoreFinderFacade,
            feature: STORE_FINDER_FEATURE,
            methods: [
                'getStoresLoading',
                'getStoresLoaded',
                'getFindStoresEntities',
                'getViewAllStoresLoading',
                'getViewAllStoresEntities',
                'findStoresAction',
                'viewAllStores',
                'viewStoreById',
                'callFindStoresAction',
                'getStoreLatitude',
                'getStoreLongitude',
                'getDirections',
                'getFindStoreEntityById',
            ],
            async: true,
        }) }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: StoreFinderFacade, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                    useFactory: () => facadeFactory({
                        facade: StoreFinderFacade,
                        feature: STORE_FINDER_FEATURE,
                        methods: [
                            'getStoresLoading',
                            'getStoresLoaded',
                            'getFindStoresEntities',
                            'getViewAllStoresLoading',
                            'getViewAllStoresEntities',
                            'findStoresAction',
                            'viewAllStores',
                            'viewStoreById',
                            'callFindStoresAction',
                            'getStoreLatitude',
                            'getStoreLongitude',
                            'getDirections',
                            'getFindStoreEntityById',
                        ],
                        async: true,
                    }),
                }]
        }] });
