// @spartacus/user/fesm2022/spartacus-user-account-assets.mjs
var loginForm = {
	forgotPassword: "Forgot password?",
	signIn: "Sign In",
	register: "Register",
	dontHaveAccount: "Don’t have an account?",
	guestCheckout: "Guest Checkout",
	emailAddress: {
		label: "Email address",
		placeholder: "Enter Email Address"
	},
	password: {
		label: "Password",
		placeholder: "Enter Your Password"
	},
	wrongEmailFormat: "This is not a valid email format."
};
var miniLogin = {
	userGreeting: "Hi, {{name}}",
	signInRegister: "Sign In / Register"
};
var myAccountV2User = {
	signOut: "Sign Out"
};
var userAccount = {
	loginForm: loginForm,
	miniLogin: miniLogin,
	myAccountV2User: myAccountV2User
};

/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
const en = {
    userAccount,
};

/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
const userAccountTranslations = {
    en,
};
const userAccountTranslationChunksConfig = {
    userAccount: ['loginForm', 'miniLogin', 'myAccountV2User'],
};

/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */

/**
 * Generated bundle index. Do not edit.
 */

export { userAccountTranslationChunksConfig, userAccountTranslations };
