// @spartacus/order/esm2022/core/store/effects/consignment-tracking-by-id.effect.mjs
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { inject, Injectable } from '@angular/core';
import { Actions, createEffect, ofType } from '@ngrx/effects';
import { LoggerService, normalizeHttpError } from '@spartacus/core';
import { of } from 'rxjs';
import { catchError, map, switchMap } from 'rxjs/operators';
import { OrderHistoryConnector } from '../../connectors';
import { OrderActions } from '../actions';
import * as i0 from "@angular/core";
export class ConsignmentTrackingByIdEffects {
    constructor() {
        this.logger = inject(LoggerService);
        this.actions$ = inject(Actions);
        this.orderConnector = inject(OrderHistoryConnector);
        this.loadConsignmentTrackingById$ = createEffect(() => this.actions$.pipe(ofType(OrderActions.LOAD_CONSIGNMENT_TRACKING_BY_ID), map((action) => action.payload), switchMap((payload) => {
            return this.orderConnector
                .getConsignmentTracking(payload.orderCode, payload.consignmentCode, payload.userId)
                .pipe(map((tracking) => new OrderActions.LoadConsignmentTrackingByIdSuccess({
                orderCode: payload.orderCode,
                consignmentCode: payload.consignmentCode,
                consignmentTracking: tracking,
            })), catchError((error) => of(new OrderActions.LoadConsignmentTrackingByIdFail({
                orderCode: payload.orderCode,
                consignmentCode: payload.consignmentCode,
                error: normalizeHttpError(error, this.logger),
            }))));
        })));
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: ConsignmentTrackingByIdEffects, deps: [], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: ConsignmentTrackingByIdEffects }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: ConsignmentTrackingByIdEffects, decorators: [{
            type: Injectable
        }] });
