// @spartacus/order/esm2022/core/store/effects/order-by-id.effect.mjs
import { inject, Injectable } from '@angular/core';
import { Actions, createEffect, ofType } from '@ngrx/effects';
import { LoggerService, normalizeHttpError } from '@spartacus/core';
import { of } from 'rxjs';
import { catchError, concatMap, map } from 'rxjs/operators';
import { OrderHistoryConnector } from '../../connectors/order-history.connector';
import { OrderActions } from '../actions/index';
import * as i0 from "@angular/core";
export class OrderByIdEffect {
    constructor() {
        this.logger = inject(LoggerService);
        this.actions$ = inject(Actions);
        this.orderConnector = inject(OrderHistoryConnector);
        this.loadOrderById$ = createEffect(() => this.actions$.pipe(ofType(OrderActions.LOAD_ORDER_BY_ID), map((action) => action.payload), concatMap(({ userId, code }) => {
            return this.orderConnector.get(userId, code).pipe(map((order) => {
                return new OrderActions.LoadOrderByIdSuccess(order);
            }), catchError((error) => {
                return of(new OrderActions.LoadOrderByIdFail({
                    code,
                    error: normalizeHttpError(error, this.logger),
                }));
            }));
        })));
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: OrderByIdEffect, deps: [], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: OrderByIdEffect }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: OrderByIdEffect, decorators: [{
            type: Injectable
        }] });
