// @spartacus/user/esm2022/profile/components/update-password/update-password.module.mjs
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { CommonModule } from '@angular/common';
import { NgModule, inject } from '@angular/core';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { RouterModule } from '@angular/router';
import { AuthGuard, AuthRedirectService, AuthService, FeaturesConfigModule, GlobalMessageService, I18nModule, RoutingService, UrlModule, provideDefaultConfig, provideDefaultConfigFactory, } from '@spartacus/core';
import { FormErrorsModule, MessageComponentModule, PasswordVisibilityToggleModule, SpinnerModule, } from '@spartacus/storefront';
import { UserPasswordFacade } from '@spartacus/user/profile/root';
import { UpdatePasswordComponentService } from './update-password-component.service';
import { UpdatePasswordComponent } from './update-password.component';
import { MyAccountV2PasswordComponent } from './my-account-v2-password.component';
import { USE_MY_ACCOUNT_V2_PASSWORD } from './use-my-account-v2-password';
import * as i0 from "@angular/core";
const myAccountV2PasswordMapping = {
    cmsComponents: {
        UpdatePasswordComponent: {
            component: MyAccountV2PasswordComponent,
        },
    },
};
export class UpdatePasswordModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: UpdatePasswordModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "17.0.5", ngImport: i0, type: UpdatePasswordModule, declarations: [UpdatePasswordComponent, MyAccountV2PasswordComponent], imports: [CommonModule,
            FormsModule,
            ReactiveFormsModule,
            SpinnerModule,
            I18nModule,
            FormErrorsModule,
            UrlModule,
            RouterModule,
            PasswordVisibilityToggleModule,
            FeaturesConfigModule,
            MessageComponentModule], exports: [UpdatePasswordComponent, MyAccountV2PasswordComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: UpdatePasswordModule, providers: [
            provideDefaultConfig({
                cmsComponents: {
                    UpdatePasswordComponent: {
                        component: UpdatePasswordComponent,
                        guards: [AuthGuard],
                        providers: [
                            {
                                provide: UpdatePasswordComponentService,
                                useClass: UpdatePasswordComponentService,
                                deps: [
                                    UserPasswordFacade,
                                    RoutingService,
                                    GlobalMessageService,
                                    AuthRedirectService,
                                    AuthService,
                                ],
                            },
                        ],
                    },
                },
            }),
            provideDefaultConfigFactory(() => inject(USE_MY_ACCOUNT_V2_PASSWORD) ? myAccountV2PasswordMapping : {}),
        ], imports: [CommonModule,
            FormsModule,
            ReactiveFormsModule,
            SpinnerModule,
            I18nModule,
            FormErrorsModule,
            UrlModule,
            RouterModule,
            PasswordVisibilityToggleModule,
            FeaturesConfigModule,
            MessageComponentModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: UpdatePasswordModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        FormsModule,
                        ReactiveFormsModule,
                        SpinnerModule,
                        I18nModule,
                        FormErrorsModule,
                        UrlModule,
                        RouterModule,
                        PasswordVisibilityToggleModule,
                        FeaturesConfigModule,
                        MessageComponentModule,
                    ],
                    providers: [
                        provideDefaultConfig({
                            cmsComponents: {
                                UpdatePasswordComponent: {
                                    component: UpdatePasswordComponent,
                                    guards: [AuthGuard],
                                    providers: [
                                        {
                                            provide: UpdatePasswordComponentService,
                                            useClass: UpdatePasswordComponentService,
                                            deps: [
                                                UserPasswordFacade,
                                                RoutingService,
                                                GlobalMessageService,
                                                AuthRedirectService,
                                                AuthService,
                                            ],
                                        },
                                    ],
                                },
                            },
                        }),
                        provideDefaultConfigFactory(() => inject(USE_MY_ACCOUNT_V2_PASSWORD) ? myAccountV2PasswordMapping : {}),
                    ],
                    declarations: [UpdatePasswordComponent, MyAccountV2PasswordComponent],
                    exports: [UpdatePasswordComponent, MyAccountV2PasswordComponent],
                }]
        }] });
