// @spartacus/cart/esm2022/saved-cart/components/add-to-saved-cart/add-to-saved-cart.module.mjs
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { RouterModule } from '@angular/router';
import { I18nModule, provideDefaultConfig, UrlModule, } from '@spartacus/core';
import { AddToSavedCartComponent } from './add-to-saved-cart.component';
import * as i0 from "@angular/core";
export class AddToSavedCartModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: AddToSavedCartModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "17.0.5", ngImport: i0, type: AddToSavedCartModule, declarations: [AddToSavedCartComponent], imports: [CommonModule, RouterModule, I18nModule, UrlModule], exports: [AddToSavedCartComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: AddToSavedCartModule, providers: [
            provideDefaultConfig({
                cmsComponents: {
                    AddToSavedCartsComponent: {
                        component: AddToSavedCartComponent,
                    },
                },
            }),
        ], imports: [CommonModule, RouterModule, I18nModule, UrlModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: AddToSavedCartModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, RouterModule, I18nModule, UrlModule],
                    providers: [
                        provideDefaultConfig({
                            cmsComponents: {
                                AddToSavedCartsComponent: {
                                    component: AddToSavedCartComponent,
                                },
                            },
                        }),
                    ],
                    exports: [AddToSavedCartComponent],
                    declarations: [AddToSavedCartComponent],
                }]
        }] });
