// @spartacus/product/esm2022/variants/components/product-variants-components.module.mjs
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { NgModule } from '@angular/core';
import { ProductVariantsContainerModule } from './product-variants-container/product-variants-container.module';
import { ProductVariantColorSelectorModule } from './variant-color-selector/product-variant-color-selector.module';
import { ProductVariantSizeSelectorModule } from './variant-size-selector/product-variant-size-selector.module';
import { ProductVariantStyleSelectorModule } from './variant-style-selector/product-variant-style-selector.module';
import { provideDefaultConfig } from '@spartacus/core';
import { ProductVariantsContainerComponent } from './product-variants-container/product-variants-container.component';
import { ProductVariantsGuard } from './guards/product-variants.guard';
import * as i0 from "@angular/core";
export class ProductVariantsComponentsModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: ProductVariantsComponentsModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "17.0.5", ngImport: i0, type: ProductVariantsComponentsModule, imports: [ProductVariantsContainerModule,
            ProductVariantColorSelectorModule,
            ProductVariantSizeSelectorModule,
            ProductVariantStyleSelectorModule] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: ProductVariantsComponentsModule, providers: [
            provideDefaultConfig({
                cmsComponents: {
                    ProductVariantSelectorComponent: {
                        component: ProductVariantsContainerComponent,
                        guards: [ProductVariantsGuard],
                    },
                },
            }),
        ], imports: [ProductVariantsContainerModule,
            ProductVariantColorSelectorModule,
            ProductVariantSizeSelectorModule,
            ProductVariantStyleSelectorModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: ProductVariantsComponentsModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        ProductVariantsContainerModule,
                        ProductVariantColorSelectorModule,
                        ProductVariantSizeSelectorModule,
                        ProductVariantStyleSelectorModule,
                    ],
                    providers: [
                        provideDefaultConfig({
                            cmsComponents: {
                                ProductVariantSelectorComponent: {
                                    component: ProductVariantsContainerComponent,
                                    guards: [ProductVariantsGuard],
                                },
                            },
                        }),
                    ],
                }]
        }] });
