// @spartacus/cart/esm2022/base/components/cart-proceed-to-checkout/cart-proceed-to-checkout.module.mjs
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { RouterModule } from '@angular/router';
import { I18nModule, provideDefaultConfig, UrlModule, } from '@spartacus/core';
import { ProgressButtonModule } from '@spartacus/storefront';
import { CartProceedToCheckoutComponent } from './cart-proceed-to-checkout.component';
import * as i0 from "@angular/core";
export class CartProceedToCheckoutModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: CartProceedToCheckoutModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "17.0.5", ngImport: i0, type: CartProceedToCheckoutModule, declarations: [CartProceedToCheckoutComponent], imports: [CommonModule,
            ProgressButtonModule,
            RouterModule,
            I18nModule,
            UrlModule], exports: [CartProceedToCheckoutComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: CartProceedToCheckoutModule, providers: [
            provideDefaultConfig({
                cmsComponents: {
                    CartProceedToCheckoutComponent: {
                        component: CartProceedToCheckoutComponent,
                    },
                },
            }),
        ], imports: [CommonModule,
            ProgressButtonModule,
            RouterModule,
            I18nModule,
            UrlModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: CartProceedToCheckoutModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        ProgressButtonModule,
                        RouterModule,
                        I18nModule,
                        UrlModule,
                    ],
                    providers: [
                        provideDefaultConfig({
                            cmsComponents: {
                                CartProceedToCheckoutComponent: {
                                    component: CartProceedToCheckoutComponent,
                                },
                            },
                        }),
                    ],
                    declarations: [CartProceedToCheckoutComponent],
                    exports: [CartProceedToCheckoutComponent],
                }]
        }] });
