// @spartacus/storefront/esm2022/cms-components/content/paragraph/paragraph.component.mjs
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { ChangeDetectionStrategy, Component, HostListener, inject, } from '@angular/core';
import { DomSanitizer } from '@angular/platform-browser';
import * as i0 from "@angular/core";
import * as i1 from "../../../cms-structure/page/model/cms-component-data";
import * as i2 from "@angular/router";
import * as i3 from "@angular/common";
import * as i4 from "../../../shared/pipes/suplement-hash-anchors/supplement-hash-anchors.pipe";
export class ParagraphComponent {
    handleClick(event) {
        if (event.target instanceof HTMLAnchorElement) {
            const element = event.target;
            const href = element?.getAttribute('href');
            const documentUrlObject = new URL(element.ownerDocument.URL);
            // Use router for internal link navigation
            if (href && documentUrlObject.host === element.host) {
                event.preventDefault();
                this.router.navigateByUrl(href.replace(documentUrlObject.origin, ''));
            }
        }
    }
    constructor(component, router) {
        this.component = component;
        this.router = router;
        this.sanitizer = inject(DomSanitizer);
    }
    bypassSecurityTrustHtml(html = '') {
        return this.sanitizer.bypassSecurityTrustHtml(html);
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: ParagraphComponent, deps: [{ token: i1.CmsComponentData }, { token: i2.Router }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "17.0.5", type: ParagraphComponent, selector: "cx-paragraph", host: { listeners: { "click": "handleClick($event)" } }, ngImport: i0, template: "<div\n  *ngIf=\"component.data$ | async as data\"\n  [innerHTML]=\"bypassSecurityTrustHtml(data.content | cxSupplementHashAnchors)\"\n></div>\n", dependencies: [{ kind: "directive", type: i3.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { kind: "pipe", type: i3.AsyncPipe, name: "async" }, { kind: "pipe", type: i4.SupplementHashAnchorsPipe, name: "cxSupplementHashAnchors" }], changeDetection: i0.ChangeDetectionStrategy.OnPush }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: ParagraphComponent, decorators: [{
            type: Component,
            args: [{ selector: 'cx-paragraph', changeDetection: ChangeDetectionStrategy.OnPush, template: "<div\n  *ngIf=\"component.data$ | async as data\"\n  [innerHTML]=\"bypassSecurityTrustHtml(data.content | cxSupplementHashAnchors)\"\n></div>\n" }]
        }], ctorParameters: () => [{ type: i1.CmsComponentData }, { type: i2.Router }], propDecorators: { handleClick: [{
                type: HostListener,
                args: ['click', ['$event']]
            }] } });
