// @spartacus/checkout/esm2022/scheduled-replenishment/components/checkout-schedule-replenishment-order/checkout-schedule-replenishment-order.module.mjs
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { RouterModule } from '@angular/router';
import { CartNotEmptyGuard, CheckoutAuthGuard, } from '@spartacus/checkout/base/components';
import { FeaturesConfigModule, I18nModule, provideDefaultConfig, } from '@spartacus/core';
import { IconModule } from '@spartacus/storefront';
import { CheckoutScheduleReplenishmentOrderComponent } from './checkout-schedule-replenishment-order.component';
import * as i0 from "@angular/core";
export class CheckoutScheduleReplenishmentOrderModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: CheckoutScheduleReplenishmentOrderModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "17.0.5", ngImport: i0, type: CheckoutScheduleReplenishmentOrderModule, declarations: [CheckoutScheduleReplenishmentOrderComponent], imports: [CommonModule,
            RouterModule,
            I18nModule,
            IconModule,
            FeaturesConfigModule], exports: [CheckoutScheduleReplenishmentOrderComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: CheckoutScheduleReplenishmentOrderModule, providers: [
            provideDefaultConfig({
                cmsComponents: {
                    CheckoutScheduleReplenishmentOrder: {
                        component: CheckoutScheduleReplenishmentOrderComponent,
                        guards: [CheckoutAuthGuard, CartNotEmptyGuard],
                    },
                },
            }),
        ], imports: [CommonModule,
            RouterModule,
            I18nModule,
            IconModule,
            FeaturesConfigModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: CheckoutScheduleReplenishmentOrderModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        RouterModule,
                        I18nModule,
                        IconModule,
                        FeaturesConfigModule,
                    ],
                    providers: [
                        provideDefaultConfig({
                            cmsComponents: {
                                CheckoutScheduleReplenishmentOrder: {
                                    component: CheckoutScheduleReplenishmentOrderComponent,
                                    guards: [CheckoutAuthGuard, CartNotEmptyGuard],
                                },
                            },
                        }),
                    ],
                    declarations: [CheckoutScheduleReplenishmentOrderComponent],
                    exports: [CheckoutScheduleReplenishmentOrderComponent],
                }]
        }] });
