// @spartacus/user/esm2022/profile/components/register/register.module.mjs
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { ReactiveFormsModule, UntypedFormBuilder } from '@angular/forms';
import { RouterModule } from '@angular/router';
import { NgSelectModule } from '@ng-select/ng-select';
import { FeaturesConfigModule, GlobalMessageService, I18nModule, NotAuthGuard, UrlModule, provideDefaultConfig, } from '@spartacus/core';
import { FormErrorsModule, NgSelectA11yModule, PasswordVisibilityToggleModule, SpinnerModule, } from '@spartacus/storefront';
import { UserRegisterFacade } from '@spartacus/user/profile/root';
import { RegisterComponentService } from './register-component.service';
import { RegisterComponent } from './register.component';
import * as i0 from "@angular/core";
export class RegisterComponentModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: RegisterComponentModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "17.0.5", ngImport: i0, type: RegisterComponentModule, declarations: [RegisterComponent], imports: [CommonModule,
            ReactiveFormsModule,
            RouterModule,
            UrlModule,
            I18nModule,
            SpinnerModule,
            FormErrorsModule,
            NgSelectModule,
            NgSelectA11yModule,
            PasswordVisibilityToggleModule,
            FeaturesConfigModule] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: RegisterComponentModule, providers: [
            provideDefaultConfig({
                cmsComponents: {
                    RegisterCustomerComponent: {
                        component: RegisterComponent,
                        guards: [NotAuthGuard],
                        providers: [
                            {
                                provide: RegisterComponentService,
                                useClass: RegisterComponentService,
                                deps: [
                                    UserRegisterFacade,
                                    GlobalMessageService,
                                    UntypedFormBuilder,
                                ],
                            },
                        ],
                    },
                },
            }),
        ], imports: [CommonModule,
            ReactiveFormsModule,
            RouterModule,
            UrlModule,
            I18nModule,
            SpinnerModule,
            FormErrorsModule,
            NgSelectModule,
            NgSelectA11yModule,
            PasswordVisibilityToggleModule,
            FeaturesConfigModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: RegisterComponentModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        ReactiveFormsModule,
                        RouterModule,
                        UrlModule,
                        I18nModule,
                        SpinnerModule,
                        FormErrorsModule,
                        NgSelectModule,
                        NgSelectA11yModule,
                        PasswordVisibilityToggleModule,
                        FeaturesConfigModule,
                    ],
                    providers: [
                        provideDefaultConfig({
                            cmsComponents: {
                                RegisterCustomerComponent: {
                                    component: RegisterComponent,
                                    guards: [NotAuthGuard],
                                    providers: [
                                        {
                                            provide: RegisterComponentService,
                                            useClass: RegisterComponentService,
                                            deps: [
                                                UserRegisterFacade,
                                                GlobalMessageService,
                                                UntypedFormBuilder,
                                            ],
                                        },
                                    ],
                                },
                            },
                        }),
                    ],
                    declarations: [RegisterComponent],
                }]
        }] });
