// @spartacus/cart/esm2022/wish-list/components/add-to-wishlist/add-to-wish-list.module.mjs
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { RouterModule } from '@angular/router';
import { I18nModule, provideDefaultConfig, UrlModule, } from '@spartacus/core';
import { IconModule, AtMessageModule } from '@spartacus/storefront';
import { AddToWishListComponent } from './add-to-wish-list.component';
import * as i0 from "@angular/core";
export class AddToWishListModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: AddToWishListModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "17.0.5", ngImport: i0, type: AddToWishListModule, declarations: [AddToWishListComponent], imports: [CommonModule,
            I18nModule,
            IconModule,
            RouterModule,
            UrlModule,
            AtMessageModule], exports: [AddToWishListComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: AddToWishListModule, providers: [
            provideDefaultConfig({
                cmsComponents: {
                    AddToWishListComponent: {
                        component: AddToWishListComponent,
                    },
                },
            }),
        ], imports: [CommonModule,
            I18nModule,
            IconModule,
            RouterModule,
            UrlModule,
            AtMessageModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: AddToWishListModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        I18nModule,
                        IconModule,
                        RouterModule,
                        UrlModule,
                        AtMessageModule,
                    ],
                    providers: [
                        provideDefaultConfig({
                            cmsComponents: {
                                AddToWishListComponent: {
                                    component: AddToWishListComponent,
                                },
                            },
                        }),
                    ],
                    declarations: [AddToWishListComponent],
                    exports: [AddToWishListComponent],
                }]
        }] });
