// @spartacus/checkout/esm2022/scheduled-replenishment/components/checkout-place-order/checkout-place-order.module.mjs
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { ReactiveFormsModule } from '@angular/forms';
import { RouterModule } from '@angular/router';
import { CartNotEmptyGuard, CheckoutAuthGuard, } from '@spartacus/checkout/base/components';
import { I18nModule, provideDefaultConfig, UrlModule, } from '@spartacus/core';
import { AtMessageModule } from '@spartacus/storefront';
import { CheckoutScheduledReplenishmentPlaceOrderComponent } from './checkout-place-order.component';
import * as i0 from "@angular/core";
export class CheckoutScheduledReplenishmentPlaceOrderModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: CheckoutScheduledReplenishmentPlaceOrderModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "17.0.5", ngImport: i0, type: CheckoutScheduledReplenishmentPlaceOrderModule, declarations: [CheckoutScheduledReplenishmentPlaceOrderComponent], imports: [AtMessageModule,
            CommonModule,
            RouterModule,
            UrlModule,
            I18nModule,
            ReactiveFormsModule], exports: [CheckoutScheduledReplenishmentPlaceOrderComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: CheckoutScheduledReplenishmentPlaceOrderModule, providers: [
            provideDefaultConfig({
                cmsComponents: {
                    CheckoutPlaceOrder: {
                        component: CheckoutScheduledReplenishmentPlaceOrderComponent,
                        guards: [CheckoutAuthGuard, CartNotEmptyGuard],
                    },
                },
            }),
        ], imports: [AtMessageModule,
            CommonModule,
            RouterModule,
            UrlModule,
            I18nModule,
            ReactiveFormsModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: CheckoutScheduledReplenishmentPlaceOrderModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        AtMessageModule,
                        CommonModule,
                        RouterModule,
                        UrlModule,
                        I18nModule,
                        ReactiveFormsModule,
                    ],
                    providers: [
                        provideDefaultConfig({
                            cmsComponents: {
                                CheckoutPlaceOrder: {
                                    component: CheckoutScheduledReplenishmentPlaceOrderComponent,
                                    guards: [CheckoutAuthGuard, CartNotEmptyGuard],
                                },
                            },
                        }),
                    ],
                    declarations: [CheckoutScheduledReplenishmentPlaceOrderComponent],
                    exports: [CheckoutScheduledReplenishmentPlaceOrderComponent],
                }]
        }] });
