// @spartacus/user/esm2022/account/components/my-account-v2-user/my-account-v2-user.module.mjs
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { AuthGuard, I18nModule, UrlModule, provideDefaultConfig, } from '@spartacus/core';
import { RouterModule } from '@angular/router';
import { MyAccountV2UserComponent } from './my-account-v2-user.component';
import * as i0 from "@angular/core";
export class MyAccountV2UserModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: MyAccountV2UserModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "17.0.5", ngImport: i0, type: MyAccountV2UserModule, declarations: [MyAccountV2UserComponent], imports: [CommonModule, RouterModule, UrlModule, I18nModule], exports: [MyAccountV2UserComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: MyAccountV2UserModule, providers: [
            provideDefaultConfig({
                cmsComponents: {
                    MyAccountViewUserComponent: {
                        component: MyAccountV2UserComponent,
                        guards: [AuthGuard],
                    },
                },
            }),
        ], imports: [CommonModule, RouterModule, UrlModule, I18nModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: MyAccountV2UserModule, decorators: [{
            type: NgModule,
            args: [{
                    providers: [
                        provideDefaultConfig({
                            cmsComponents: {
                                MyAccountViewUserComponent: {
                                    component: MyAccountV2UserComponent,
                                    guards: [AuthGuard],
                                },
                            },
                        }),
                    ],
                    declarations: [MyAccountV2UserComponent],
                    exports: [MyAccountV2UserComponent],
                    imports: [CommonModule, RouterModule, UrlModule, I18nModule],
                }]
        }] });
