// @spartacus/storefront/esm2022/cms-components/navigation/category-navigation/category-navigation.module.mjs
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { I18nModule, provideDefaultConfig } from '@spartacus/core';
import { NavigationModule } from '../navigation/navigation.module';
import { CategoryNavigationComponent } from './category-navigation.component';
import * as i0 from "@angular/core";
export class CategoryNavigationModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: CategoryNavigationModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "17.0.5", ngImport: i0, type: CategoryNavigationModule, declarations: [CategoryNavigationComponent], imports: [CommonModule, NavigationModule, I18nModule], exports: [CategoryNavigationComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: CategoryNavigationModule, providers: [
            provideDefaultConfig({
                cmsComponents: {
                    CategoryNavigationComponent: {
                        component: CategoryNavigationComponent,
                        data: {
                            resetMenuOnClose: true,
                        },
                    },
                },
            }),
        ], imports: [CommonModule, NavigationModule, I18nModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: CategoryNavigationModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, NavigationModule, I18nModule],
                    providers: [
                        provideDefaultConfig({
                            cmsComponents: {
                                CategoryNavigationComponent: {
                                    component: CategoryNavigationComponent,
                                    data: {
                                        resetMenuOnClose: true,
                                    },
                                },
                            },
                        }),
                    ],
                    declarations: [CategoryNavigationComponent],
                    exports: [CategoryNavigationComponent],
                }]
        }] });
