// @spartacus/core/esm2022/src/auth/user-auth/http-interceptors/auth.interceptor.mjs
import { Injectable } from '@angular/core';
import { EMPTY, of } from 'rxjs';
import { catchError, map, switchMap, take } from 'rxjs/operators';
import * as i0 from "@angular/core";
import * as i1 from "../services/auth-http-header.service";
import * as i2 from "../services/auth-config.service";
/**
 * Responsible for catching auth errors and providing `Authorization` header for API calls.
 * Uses AuthHttpHeaderService for request manipulation and error handling. Interceptor only hooks into request send/received events.
 */
export class AuthInterceptor {
    constructor(authHttpHeaderService, authConfigService) {
        this.authHttpHeaderService = authHttpHeaderService;
        this.authConfigService = authConfigService;
    }
    intercept(httpRequest, next) {
        const shouldCatchError = this.authHttpHeaderService.shouldCatchError(httpRequest);
        const shouldAddAuthorizationHeader = this.authHttpHeaderService.shouldAddAuthorizationHeader(httpRequest);
        const token$ = shouldAddAuthorizationHeader
            ? // emits sync, unless there is refresh or logout in progress, in which case it emits async
                this.authHttpHeaderService.getStableToken().pipe(take(1))
            : of(undefined);
        const requestAndToken$ = token$.pipe(map((token) => ({
            token,
            request: this.authHttpHeaderService.alterRequest(httpRequest, token),
        })));
        return requestAndToken$.pipe(switchMap(({ request, token }) => next.handle(request).pipe(catchError((errResponse) => {
            switch (errResponse.status) {
                case 401: // Unauthorized
                    if (this.isExpiredToken(errResponse) && shouldCatchError) {
                        // request failed because of the expired access token
                        // we should get refresh the token and retry the request, or logout if the refresh is missing / expired
                        return this.authHttpHeaderService.handleExpiredAccessToken(request, next, token);
                    }
                    else if (
                    // Refresh the expired token
                    // Check if the OAuth endpoint was called and the error is because the refresh token expired
                    this.errorIsInvalidToken(errResponse)) {
                        this.authHttpHeaderService.handleExpiredRefreshToken();
                        return EMPTY;
                    }
                    break;
                case 400: // Bad Request
                    if (this.errorIsInvalidGrant(errResponse) &&
                        request.body.get('grant_type') === 'refresh_token') {
                        this.authHttpHeaderService.handleExpiredRefreshToken();
                    }
                    break;
            }
            throw errResponse;
        }))));
    }
    errorIsInvalidToken(errResponse) {
        return ((errResponse.url?.includes(this.authConfigService.getTokenEndpoint()) &&
            errResponse.error.error === 'invalid_token') ??
            false);
    }
    errorIsInvalidGrant(errResponse) {
        return ((errResponse.url?.includes(this.authConfigService.getTokenEndpoint()) &&
            errResponse.error.error === 'invalid_grant') ??
            false);
    }
    isExpiredToken(resp) {
        return resp.error?.errors?.[0]?.type === 'InvalidTokenError';
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: AuthInterceptor, deps: [{ token: i1.AuthHttpHeaderService }, { token: i2.AuthConfigService }], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: AuthInterceptor, providedIn: 'root' }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: AuthInterceptor, decorators: [{
            type: Injectable,
            args: [{ providedIn: 'root' }]
        }], ctorParameters: () => [{ type: i1.AuthHttpHeaderService }, { type: i2.AuthConfigService }] });
