// @spartacus/checkout/esm2022/base/root/facade/checkout-delivery-modes.facade.mjs
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { Injectable } from '@angular/core';
import { facadeFactory } from '@spartacus/core';
import { CHECKOUT_CORE_FEATURE } from '../feature-name';
import * as i0 from "@angular/core";
export class CheckoutDeliveryModesFacade {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: CheckoutDeliveryModesFacade, deps: [], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: CheckoutDeliveryModesFacade, providedIn: 'root', useFactory: () => facadeFactory({
            facade: CheckoutDeliveryModesFacade,
            feature: CHECKOUT_CORE_FEATURE,
            methods: [
                'getSupportedDeliveryModesState',
                'getSupportedDeliveryModes',
                'setDeliveryMode',
                'getSelectedDeliveryModeState',
                'clearCheckoutDeliveryMode',
            ],
            // TODO:#deprecation-checkout - remove once we remove ngrx
            async: true,
        }) }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: CheckoutDeliveryModesFacade, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                    useFactory: () => facadeFactory({
                        facade: CheckoutDeliveryModesFacade,
                        feature: CHECKOUT_CORE_FEATURE,
                        methods: [
                            'getSupportedDeliveryModesState',
                            'getSupportedDeliveryModes',
                            'setDeliveryMode',
                            'getSelectedDeliveryModeState',
                            'clearCheckoutDeliveryMode',
                        ],
                        // TODO:#deprecation-checkout - remove once we remove ngrx
                        async: true,
                    }),
                }]
        }] });
