// @spartacus/checkout/esm2022/base/components/checkout-review/checkout-review-overview/checkout-review-overview.module.mjs
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { I18nModule, provideDefaultConfig } from '@spartacus/core';
import { PromotionsModule } from '@spartacus/storefront';
import { CartNotEmptyGuard } from '../../guards/cart-not-empty.guard';
import { CheckoutAuthGuard } from '../../guards/checkout-auth.guard';
import { CheckoutReviewOverviewComponent } from './checkout-review-overview.component';
import * as i0 from "@angular/core";
export class CheckoutReviewOverviewModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: CheckoutReviewOverviewModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "17.0.5", ngImport: i0, type: CheckoutReviewOverviewModule, declarations: [CheckoutReviewOverviewComponent], imports: [CommonModule, PromotionsModule, I18nModule] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: CheckoutReviewOverviewModule, providers: [
            provideDefaultConfig({
                cmsComponents: {
                    CheckoutReviewOverview: {
                        component: CheckoutReviewOverviewComponent,
                        guards: [CheckoutAuthGuard, CartNotEmptyGuard],
                    },
                },
            }),
        ], imports: [CommonModule, PromotionsModule, I18nModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: CheckoutReviewOverviewModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [CheckoutReviewOverviewComponent],
                    imports: [CommonModule, PromotionsModule, I18nModule],
                    providers: [
                        provideDefaultConfig({
                            cmsComponents: {
                                CheckoutReviewOverview: {
                                    component: CheckoutReviewOverviewComponent,
                                    guards: [CheckoutAuthGuard, CartNotEmptyGuard],
                                },
                            },
                        }),
                    ],
                }]
        }] });
