// @spartacus/product/esm2022/future-stock/occ/adapters/occ-future-stock.adapter.mjs
import { Injectable, inject } from '@angular/core';
import { LoggerService, normalizeHttpError, } from '@spartacus/core';
import { FUTURE_STOCK_LIST_NORMALIZER, FUTURE_STOCK_NORMALIZER, } from '@spartacus/product/future-stock/core';
import { catchError } from 'rxjs/operators';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common/http";
import * as i2 from "@spartacus/core";
export class OccFutureStockAdapter {
    constructor(http, occEndpoints, converter) {
        this.http = http;
        this.occEndpoints = occEndpoints;
        this.converter = converter;
        this.logger = inject(LoggerService);
    }
    getFutureStock(userId, productCode) {
        return this.http
            .get(this.getFutureStockEndpoint(userId, productCode))
            .pipe(catchError((error) => {
            throw normalizeHttpError(error, this.logger);
        }), this.converter.pipeable(FUTURE_STOCK_NORMALIZER));
    }
    getFutureStocks(userId, productCodes) {
        return this.http
            .get(this.getFutureStocksEndpoint(userId, productCodes))
            .pipe(catchError((error) => {
            throw normalizeHttpError(error, this.logger);
        }), this.converter.pipeable(FUTURE_STOCK_LIST_NORMALIZER));
    }
    getFutureStockEndpoint(userId, productCode) {
        return this.occEndpoints.buildUrl('futureStock', {
            urlParams: { userId, productCode },
        });
    }
    getFutureStocksEndpoint(userId, productCodes) {
        const params = {};
        params['productCodes'] = productCodes;
        return this.occEndpoints.buildUrl('futureStocks', {
            urlParams: { userId },
            queryParams: params,
        });
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: OccFutureStockAdapter, deps: [{ token: i1.HttpClient }, { token: i2.OccEndpointsService }, { token: i2.ConverterService }], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: OccFutureStockAdapter }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: OccFutureStockAdapter, decorators: [{
            type: Injectable
        }], ctorParameters: () => [{ type: i1.HttpClient }, { type: i2.OccEndpointsService }, { type: i2.ConverterService }] });
