// @spartacus/order/esm2022/components/order-history/order-history.module.mjs
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { CommonModule } from '@angular/common';
import { inject, NgModule } from '@angular/core';
import { FormsModule } from '@angular/forms';
import { RouterModule } from '@angular/router';
import { NgSelectModule } from '@ng-select/ng-select';
import { AuthGuard, I18nModule, provideDefaultConfig, provideDefaultConfigFactory, UrlModule, } from '@spartacus/core';
import { ListNavigationModule, MediaModule, SpinnerModule, } from '@spartacus/storefront';
import { OrderHistoryComponent } from './order-history.component';
import { USE_MY_ACCOUNT_V2_ORDER } from '@spartacus/order/root';
import { MyAccountV2OrderHistoryComponent, MyAccountV2OrderConsolidatedInformationComponent, MyAccountV2ConsignmentEntriesComponent, } from './my-account-v2';
import * as i0 from "@angular/core";
const myAccountV2CmsMapping = {
    cmsComponents: {
        AccountOrderHistoryComponent: {
            component: MyAccountV2OrderHistoryComponent,
            //guards: inherited from standard config,
        },
    },
};
const moduleComponents = [
    MyAccountV2OrderHistoryComponent,
    MyAccountV2OrderConsolidatedInformationComponent,
    MyAccountV2ConsignmentEntriesComponent,
];
export class OrderHistoryModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: OrderHistoryModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "17.0.5", ngImport: i0, type: OrderHistoryModule, declarations: [OrderHistoryComponent, MyAccountV2OrderHistoryComponent,
            MyAccountV2OrderConsolidatedInformationComponent,
            MyAccountV2ConsignmentEntriesComponent], imports: [CommonModule,
            RouterModule,
            FormsModule,
            NgSelectModule,
            ListNavigationModule,
            UrlModule,
            I18nModule,
            SpinnerModule,
            MediaModule], exports: [OrderHistoryComponent, MyAccountV2OrderHistoryComponent,
            MyAccountV2OrderConsolidatedInformationComponent,
            MyAccountV2ConsignmentEntriesComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: OrderHistoryModule, providers: [
            provideDefaultConfig({
                cmsComponents: {
                    AccountOrderHistoryComponent: {
                        component: OrderHistoryComponent,
                        guards: [AuthGuard],
                    },
                },
            }),
            provideDefaultConfigFactory(() => inject(USE_MY_ACCOUNT_V2_ORDER) ? myAccountV2CmsMapping : {}),
        ], imports: [CommonModule,
            RouterModule,
            FormsModule,
            NgSelectModule,
            ListNavigationModule,
            UrlModule,
            I18nModule,
            SpinnerModule,
            MediaModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: OrderHistoryModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        RouterModule,
                        FormsModule,
                        NgSelectModule,
                        ListNavigationModule,
                        UrlModule,
                        I18nModule,
                        SpinnerModule,
                        MediaModule,
                    ],
                    declarations: [OrderHistoryComponent, ...moduleComponents],
                    exports: [OrderHistoryComponent, ...moduleComponents],
                    providers: [
                        provideDefaultConfig({
                            cmsComponents: {
                                AccountOrderHistoryComponent: {
                                    component: OrderHistoryComponent,
                                    guards: [AuthGuard],
                                },
                            },
                        }),
                        provideDefaultConfigFactory(() => inject(USE_MY_ACCOUNT_V2_ORDER) ? myAccountV2CmsMapping : {}),
                    ],
                }]
        }] });
