// @spartacus/asm/esm2022/components/services/asm-component.service.mjs
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { Injectable, inject } from '@angular/core';
import { ASM_ENABLED_LOCAL_STORAGE_KEY, } from '@spartacus/asm/root';
import { RoutingService } from '@spartacus/core';
import { BehaviorSubject } from 'rxjs';
import { AsmDialogActionType, } from '@spartacus/asm/customer-360/root';
import * as i0 from "@angular/core";
import * as i1 from "@spartacus/core";
import * as i2 from "@spartacus/asm/root";
export class AsmComponentService {
    constructor(authService, csAgentAuthService, winRef, asmEnablerService, asmDeepLinkService) {
        this.authService = authService;
        this.csAgentAuthService = csAgentAuthService;
        this.winRef = winRef;
        this.asmEnablerService = asmEnablerService;
        this.asmDeepLinkService = asmDeepLinkService;
        this.isEmulatedByDeepLink$ = new BehaviorSubject(false);
        this.showDeeplinkCartInfoAlert$ = new BehaviorSubject(false);
        this.routingService = inject(RoutingService);
    }
    /**
     * Returns a deep link parameter value if it is in the url.
     */
    getSearchParameter(key) {
        return this.asmDeepLinkService?.getSearchParameter(key);
    }
    isEmulatedByDeepLink() {
        return this.isEmulatedByDeepLink$;
    }
    setEmulatedByDeepLink(emulated) {
        this.isEmulatedByDeepLink$.next(emulated);
    }
    setShowDeeplinkCartInfoAlert(display) {
        this.showDeeplinkCartInfoAlert$.next(display);
    }
    shouldShowDeeplinkCartInfoAlert() {
        return this.showDeeplinkCartInfoAlert$;
    }
    logoutCustomerSupportAgentAndCustomer() {
        this.csAgentAuthService.logoutCustomerSupportAgent();
    }
    logoutCustomer() {
        this.authService.logout();
    }
    isCustomerEmulationSessionInProgress() {
        return this.csAgentAuthService.isCustomerEmulated();
    }
    /**
     * We're currently only removing the persisted storage in the browser
     * to ensure the ASM experience isn't loaded on the next visit. There are a few
     * optimizations we could think of:
     * - drop the `asm` parameter from the URL, in case it's still there
     * - remove the generated UI from the DOM (outlets currently do not support this)
     */
    unload() {
        if (this.winRef.localStorage) {
            this.winRef.localStorage.removeItem(ASM_ENABLED_LOCAL_STORAGE_KEY);
        }
    }
    /**
     * check whether try to emulate customer from deeplink
     */
    isEmulateInURL() {
        return this.asmDeepLinkService?.isEmulateInURL() || false;
    }
    /**
     * Returns valid deep link parameters in the url.
     */
    getDeepLinkUrlParams() {
        return this.asmDeepLinkService?.getParamsInUrl();
    }
    /**
     * Handles the navigation based on deep link parameters in the URL
     * or passed parameter.
     */
    handleDeepLinkNavigation(parameters = this.getDeepLinkUrlParams()) {
        this.asmDeepLinkService?.handleNavigation(parameters);
    }
    handleAsmDialogAction(event) {
        if (typeof event === 'object' &&
            event.actionType === AsmDialogActionType.NAVIGATE) {
            this.routingService.go(event.route);
        }
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: AsmComponentService, deps: [{ token: i1.AuthService }, { token: i2.CsAgentAuthService }, { token: i1.WindowRef }, { token: i2.AsmEnablerService }, { token: i2.AsmDeepLinkService }], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: AsmComponentService, providedIn: 'root' }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: AsmComponentService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }], ctorParameters: () => [{ type: i1.AuthService }, { type: i2.CsAgentAuthService }, { type: i1.WindowRef }, { type: i2.AsmEnablerService }, { type: i2.AsmDeepLinkService }] });
