// @spartacus/core/esm2022/src/util/command-query/command.service.mjs
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { Injectable } from '@angular/core';
import { EMPTY, ReplaySubject, Subject, Subscription, defer, zip, } from 'rxjs';
import { catchError, concatMap, finalize, mergeMap, switchMap, tap, } from 'rxjs/operators';
import * as i0 from "@angular/core";
export class Command {
}
export var CommandStrategy;
(function (CommandStrategy) {
    CommandStrategy[CommandStrategy["Parallel"] = 0] = "Parallel";
    CommandStrategy[CommandStrategy["Queue"] = 1] = "Queue";
    CommandStrategy[CommandStrategy["CancelPrevious"] = 2] = "CancelPrevious";
    CommandStrategy[CommandStrategy["ErrorPrevious"] = 3] = "ErrorPrevious";
    // SkipIfOngoing,
    // ErrorIfOngoing
})(CommandStrategy || (CommandStrategy = {}));
export class CommandService {
    constructor() {
        this.subscriptions = new Subscription();
        // Intentional empty constructor
    }
    create(commandFactory, options) {
        const commands$ = new Subject();
        const results$ = new Subject();
        // We have to specify selectively the handlers after RxJS version 7.3.0.
        // Otherwise, the `unsubscribe` handler would be forwarded implicitly
        // to `notifier$` when calling `tap(notifier$)`.
        // But we don't want to forward `unsubscribe`, in particular.
        // To see more details, please check: https://github.com/ReactiveX/rxjs/pull/6527
        const notify = (notifier$) => ({
            next: (x) => notifier$.next(x),
            error: (e) => notifier$.error(e),
            complete: () => notifier$.complete(),
        });
        let process$;
        switch (options?.strategy) {
            case CommandStrategy.CancelPrevious:
            case CommandStrategy.ErrorPrevious:
                process$ = zip(commands$, results$).pipe(switchMap(([cmd, notifier$]) => defer(() => commandFactory(cmd)).pipe(tap(notify(notifier$)), catchError(() => EMPTY), finalize(() => {
                    // do not overwrite existing existing ending state
                    if (!notifier$.closed && !notifier$.hasError) {
                        // command has not ended yet, so close notifier$ according to strategy
                        if (options.strategy === CommandStrategy.ErrorPrevious) {
                            notifier$.error(new Error('Canceled by next command'));
                        }
                        else {
                            notifier$.complete();
                        }
                    }
                }))));
                break;
            case CommandStrategy.Parallel:
                process$ = zip(commands$, results$).pipe(mergeMap(([cmd, notifier$]) => defer(() => commandFactory(cmd)).pipe(tap(notify(notifier$)), catchError(() => EMPTY))));
                break;
            case CommandStrategy.Queue:
            default:
                process$ = zip(commands$, results$).pipe(concatMap(([cmd, notifier$]) => defer(() => commandFactory(cmd)).pipe(tap(notify(notifier$)), catchError(() => EMPTY))));
                break;
        }
        this.subscriptions.add(process$.subscribe());
        const command = new (class extends Command {
            constructor() {
                super(...arguments);
                this.execute = (parameters) => {
                    const result$ = new ReplaySubject();
                    results$.next(result$);
                    commands$.next(parameters);
                    return result$;
                };
            }
        })();
        return command;
    }
    ngOnDestroy() {
        this.subscriptions.unsubscribe();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: CommandService, deps: [], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: CommandService, providedIn: 'root' }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: CommandService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }], ctorParameters: () => [] });
