// @spartacus/product/esm2022/future-stock/components/future-stock-accordion/future-stock-accordion.module.mjs
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { FutureStockAccordionComponent } from './future-stock-accordion.component';
import { IconModule } from '@spartacus/storefront';
import { I18nModule, provideDefaultConfig } from '@spartacus/core';
import * as i0 from "@angular/core";
export class FutureStockAccordionModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: FutureStockAccordionModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "17.0.5", ngImport: i0, type: FutureStockAccordionModule, declarations: [FutureStockAccordionComponent], imports: [CommonModule, I18nModule, IconModule], exports: [FutureStockAccordionComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: FutureStockAccordionModule, providers: [
            provideDefaultConfig({
                cmsComponents: {
                    FutureStockComponent: {
                        component: FutureStockAccordionComponent,
                    },
                },
            }),
        ], imports: [CommonModule, I18nModule, IconModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: FutureStockAccordionModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, I18nModule, IconModule],
                    declarations: [FutureStockAccordionComponent],
                    providers: [
                        provideDefaultConfig({
                            cmsComponents: {
                                FutureStockComponent: {
                                    component: FutureStockAccordionComponent,
                                },
                            },
                        }),
                    ],
                    exports: [FutureStockAccordionComponent],
                }]
        }] });
