// @spartacus/storefront/esm2022/cms-components/product/carousel/product-carousel-item/product-carousel-item.component.mjs
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { ChangeDetectionStrategy, Component, Input, } from '@angular/core';
import { ProductListItemContext, ProductListItemContextSource, } from '../../product-list';
import * as i0 from "@angular/core";
import * as i1 from "../../product-list";
import * as i2 from "../../../../shared/components/media/media.component";
import * as i3 from "@angular/router";
import * as i4 from "../../../../cms-structure/page/component/inner-components-host.directive";
import * as i5 from "@spartacus/core";
export class ProductCarouselItemComponent {
    constructor(productListItemContextSource) {
        this.productListItemContextSource = productListItemContextSource;
    }
    ngOnChanges(changes) {
        if (changes?.item) {
            this.productListItemContextSource.product$.next(this.item);
        }
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: ProductCarouselItemComponent, deps: [{ token: i1.ProductListItemContextSource }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "17.0.5", type: ProductCarouselItemComponent, selector: "cx-product-carousel-item", inputs: { item: "item" }, providers: [
            ProductListItemContextSource,
            {
                provide: ProductListItemContext,
                useExisting: ProductListItemContextSource,
            },
        ], usesOnChanges: true, ngImport: i0, template: "<a tabindex=\"0\" [routerLink]=\"{ cxRoute: 'product', params: item } | cxUrl\">\n  <cx-media\n    [container]=\"item.images?.PRIMARY\"\n    format=\"product\"\n    [alt]=\"item.name ?? ''\"\n  ></cx-media>\n  <h3 class=\"cx-product-name\">\n    {{ item.name }}\n  </h3>\n  <div class=\"price\">\n    {{ item.price?.formattedValue }}\n  </div>\n</a>\n<div class=\"actions\">\n  <ng-container cxInnerComponentsHost></ng-container>\n</div>\n", dependencies: [{ kind: "component", type: i2.MediaComponent, selector: "cx-media", inputs: ["container", "format", "alt", "role", "loading"], outputs: ["loaded"] }, { kind: "directive", type: i3.RouterLink, selector: "[routerLink]", inputs: ["target", "queryParams", "fragment", "queryParamsHandling", "state", "relativeTo", "preserveFragment", "skipLocationChange", "replaceUrl", "routerLink"] }, { kind: "directive", type: i4.InnerComponentsHostDirective, selector: "[cxInnerComponentsHost]" }, { kind: "pipe", type: i5.UrlPipe, name: "cxUrl" }], changeDetection: i0.ChangeDetectionStrategy.OnPush }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: ProductCarouselItemComponent, decorators: [{
            type: Component,
            args: [{ selector: 'cx-product-carousel-item', providers: [
                        ProductListItemContextSource,
                        {
                            provide: ProductListItemContext,
                            useExisting: ProductListItemContextSource,
                        },
                    ], changeDetection: ChangeDetectionStrategy.OnPush, template: "<a tabindex=\"0\" [routerLink]=\"{ cxRoute: 'product', params: item } | cxUrl\">\n  <cx-media\n    [container]=\"item.images?.PRIMARY\"\n    format=\"product\"\n    [alt]=\"item.name ?? ''\"\n  ></cx-media>\n  <h3 class=\"cx-product-name\">\n    {{ item.name }}\n  </h3>\n  <div class=\"price\">\n    {{ item.price?.formattedValue }}\n  </div>\n</a>\n<div class=\"actions\">\n  <ng-container cxInnerComponentsHost></ng-container>\n</div>\n" }]
        }], ctorParameters: () => [{ type: i1.ProductListItemContextSource }], propDecorators: { item: [{
                type: Input
            }] } });
