// @spartacus/asm/esm2022/customer-360/components/sections/asm-customer-360-section/asm-customer-360-section.component.mjs
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { Component, EventEmitter, Input, Output, } from '@angular/core';
import { Subscription } from 'rxjs';
import { AsmCustomer360SectionContextSource } from '../asm-customer-360-section-context-source.model';
import { AsmCustomer360SectionContext } from '../asm-customer-360-section-context.model';
import * as i0 from "@angular/core";
import * as i1 from "../asm-customer-360-section-context-source.model";
import * as i2 from "@angular/common";
export class AsmCustomer360SectionComponent {
    set customer(customer) {
        this.source.customer$.next(customer);
    }
    set config(config) {
        this.source.config$.next(config);
    }
    set data(data) {
        this.source.data$.next(data);
    }
    constructor(source) {
        this.source = source;
        this.navigate = new EventEmitter();
        this.subscription = new Subscription();
        this.subscription.add(source.navigate$.subscribe((urlCommand) => this.navigate.emit(urlCommand)));
        this.subscription.add(() => {
            this.source.config$.complete();
            this.source.customer$.complete();
            this.source.data$.complete();
            this.source.navigate$.complete();
        });
    }
    ngOnDestroy() {
        this.subscription.unsubscribe();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: AsmCustomer360SectionComponent, deps: [{ token: i1.AsmCustomer360SectionContextSource }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "17.0.5", type: AsmCustomer360SectionComponent, selector: "cx-asm-customer-360-section", inputs: { component: "component", customer: "customer", config: "config", data: "data" }, outputs: { navigate: "navigate" }, providers: [
            AsmCustomer360SectionContextSource,
            {
                provide: AsmCustomer360SectionContext,
                useExisting: AsmCustomer360SectionContextSource,
            },
        ], ngImport: i0, template: "<ng-container *ngComponentOutlet=\"component\"></ng-container>\n", dependencies: [{ kind: "directive", type: i2.NgComponentOutlet, selector: "[ngComponentOutlet]", inputs: ["ngComponentOutlet", "ngComponentOutletInputs", "ngComponentOutletInjector", "ngComponentOutletContent", "ngComponentOutletNgModule", "ngComponentOutletNgModuleFactory"] }] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: AsmCustomer360SectionComponent, decorators: [{
            type: Component,
            args: [{ selector: 'cx-asm-customer-360-section', providers: [
                        AsmCustomer360SectionContextSource,
                        {
                            provide: AsmCustomer360SectionContext,
                            useExisting: AsmCustomer360SectionContextSource,
                        },
                    ], template: "<ng-container *ngComponentOutlet=\"component\"></ng-container>\n" }]
        }], ctorParameters: () => [{ type: i1.AsmCustomer360SectionContextSource }], propDecorators: { component: [{
                type: Input
            }], customer: [{
                type: Input
            }], config: [{
                type: Input
            }], data: [{
                type: Input
            }], navigate: [{
                type: Output
            }] } });
