// @spartacus/cart/esm2022/saved-cart/components/details/saved-cart-details.module.mjs
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { RouterModule } from '@angular/router';
import { AddToCartModule } from '@spartacus/cart/base/components/add-to-cart';
import { AuthGuard, ConfigModule, I18nModule, UrlModule, } from '@spartacus/core';
import { CardModule, IconModule, MediaModule, OutletModule, SpinnerModule, } from '@spartacus/storefront';
import { SavedCartDetailsActionComponent } from './saved-cart-details-action/saved-cart-details-action.component';
import { SavedCartDetailsItemsComponent } from './saved-cart-details-items/saved-cart-details-items.component';
import { SavedCartDetailsOverviewComponent } from './saved-cart-details-overview/saved-cart-details-overview.component';
import * as i0 from "@angular/core";
import * as i1 from "@spartacus/core";
export class SavedCartDetailsModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: SavedCartDetailsModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "17.0.5", ngImport: i0, type: SavedCartDetailsModule, declarations: [SavedCartDetailsOverviewComponent,
            SavedCartDetailsActionComponent,
            SavedCartDetailsItemsComponent], imports: [CommonModule,
            I18nModule,
            UrlModule,
            RouterModule,
            CardModule,
            MediaModule,
            IconModule,
            SpinnerModule,
            OutletModule,
            AddToCartModule, i1.ConfigModule], exports: [SavedCartDetailsOverviewComponent,
            SavedCartDetailsActionComponent,
            SavedCartDetailsItemsComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: SavedCartDetailsModule, imports: [CommonModule,
            I18nModule,
            UrlModule,
            RouterModule,
            CardModule,
            MediaModule,
            IconModule,
            SpinnerModule,
            OutletModule,
            AddToCartModule,
            ConfigModule.withConfig({
                cmsComponents: {
                    SavedCartDetailsOverviewComponent: {
                        component: SavedCartDetailsOverviewComponent,
                        guards: [AuthGuard],
                    },
                    SavedCartDetailsItemsComponent: {
                        component: SavedCartDetailsItemsComponent,
                        guards: [AuthGuard],
                    },
                    SavedCartDetailsActionComponent: {
                        component: SavedCartDetailsActionComponent,
                        guards: [AuthGuard],
                    },
                },
            })] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: SavedCartDetailsModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        I18nModule,
                        UrlModule,
                        RouterModule,
                        CardModule,
                        MediaModule,
                        IconModule,
                        SpinnerModule,
                        OutletModule,
                        AddToCartModule,
                        ConfigModule.withConfig({
                            cmsComponents: {
                                SavedCartDetailsOverviewComponent: {
                                    component: SavedCartDetailsOverviewComponent,
                                    guards: [AuthGuard],
                                },
                                SavedCartDetailsItemsComponent: {
                                    component: SavedCartDetailsItemsComponent,
                                    guards: [AuthGuard],
                                },
                                SavedCartDetailsActionComponent: {
                                    component: SavedCartDetailsActionComponent,
                                    guards: [AuthGuard],
                                },
                            },
                        }),
                    ],
                    declarations: [
                        SavedCartDetailsOverviewComponent,
                        SavedCartDetailsActionComponent,
                        SavedCartDetailsItemsComponent,
                    ],
                    exports: [
                        SavedCartDetailsOverviewComponent,
                        SavedCartDetailsActionComponent,
                        SavedCartDetailsItemsComponent,
                    ],
                }]
        }] });
