// @spartacus/asm/esm2022/customer-360/components/asm-customer-360-product-listing/asm-customer-360-product-listing.module.mjs
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { I18nModule } from '@spartacus/core';
import { MediaModule } from '@spartacus/storefront';
import { AsmCustomer360ProductItemModule } from '../asm-customer-360-product-item/asm-customer-360-product-item.module';
import { AsmCustomer360ProductListingComponent } from './asm-customer-360-product-listing.component';
import * as i0 from "@angular/core";
export class AsmCustomer360ProductListingModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: AsmCustomer360ProductListingModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "17.0.5", ngImport: i0, type: AsmCustomer360ProductListingModule, declarations: [AsmCustomer360ProductListingComponent], imports: [CommonModule,
            I18nModule,
            MediaModule,
            AsmCustomer360ProductItemModule], exports: [AsmCustomer360ProductListingComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: AsmCustomer360ProductListingModule, imports: [CommonModule,
            I18nModule,
            MediaModule,
            AsmCustomer360ProductItemModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: AsmCustomer360ProductListingModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        I18nModule,
                        MediaModule,
                        AsmCustomer360ProductItemModule,
                    ],
                    declarations: [AsmCustomer360ProductListingComponent],
                    exports: [AsmCustomer360ProductListingComponent],
                }]
        }] });
