// @spartacus/checkout/esm2022/base/components/checkout-delivery-mode/checkout-delivery-mode.module.mjs
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { ReactiveFormsModule } from '@angular/forms';
import { I18nModule, provideDefaultConfig } from '@spartacus/core';
import { OutletModule, PageComponentModule, SpinnerModule, } from '@spartacus/storefront';
import { CartNotEmptyGuard } from '../guards/cart-not-empty.guard';
import { CheckoutAuthGuard } from '../guards/checkout-auth.guard';
import { CheckoutDeliveryModeComponent } from './checkout-delivery-mode.component';
import * as i0 from "@angular/core";
export class CheckoutDeliveryModeModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: CheckoutDeliveryModeModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "17.0.5", ngImport: i0, type: CheckoutDeliveryModeModule, declarations: [CheckoutDeliveryModeComponent], imports: [CommonModule,
            ReactiveFormsModule,
            I18nModule,
            SpinnerModule,
            OutletModule,
            PageComponentModule], exports: [CheckoutDeliveryModeComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: CheckoutDeliveryModeModule, providers: [
            provideDefaultConfig({
                cmsComponents: {
                    CheckoutDeliveryMode: {
                        component: CheckoutDeliveryModeComponent,
                        data: {
                            composition: {
                                inner: ['PickupInStoreDeliveryModeComponent'],
                            },
                        },
                        guards: [CheckoutAuthGuard, CartNotEmptyGuard],
                    },
                },
            }),
        ], imports: [CommonModule,
            ReactiveFormsModule,
            I18nModule,
            SpinnerModule,
            OutletModule,
            PageComponentModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: CheckoutDeliveryModeModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        ReactiveFormsModule,
                        I18nModule,
                        SpinnerModule,
                        OutletModule,
                        PageComponentModule,
                    ],
                    providers: [
                        provideDefaultConfig({
                            cmsComponents: {
                                CheckoutDeliveryMode: {
                                    component: CheckoutDeliveryModeComponent,
                                    data: {
                                        composition: {
                                            inner: ['PickupInStoreDeliveryModeComponent'],
                                        },
                                    },
                                    guards: [CheckoutAuthGuard, CartNotEmptyGuard],
                                },
                            },
                        }),
                    ],
                    declarations: [CheckoutDeliveryModeComponent],
                    exports: [CheckoutDeliveryModeComponent],
                }]
        }] });
