// @spartacus/user/esm2022/profile/components/update-email/update-email.module.mjs
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { CommonModule } from '@angular/common';
import { NgModule, inject } from '@angular/core';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { RouterModule } from '@angular/router';
import { AuthGuard, AuthRedirectService, AuthService, FeaturesConfigModule, GlobalMessageService, I18nModule, RoutingService, UrlModule, provideDefaultConfig, provideDefaultConfigFactory, } from '@spartacus/core';
import { FormErrorsModule, MessageComponentModule, PasswordVisibilityToggleModule, SpinnerModule, } from '@spartacus/storefront';
import { UserEmailFacade } from '@spartacus/user/profile/root';
import { UpdateEmailComponentService } from './update-email-component.service';
import { UpdateEmailComponent } from './update-email.component';
import { MyAccountV2EmailComponent } from './my-account-v2-email.component';
import { USE_MY_ACCOUNT_V2_EMAIL } from './use-my-account-v2-email.ts';
import * as i0 from "@angular/core";
const myAccountV2EmailMapping = {
    cmsComponents: {
        UpdateEmailComponent: {
            component: MyAccountV2EmailComponent,
        },
    },
};
export class UpdateEmailModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: UpdateEmailModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "17.0.5", ngImport: i0, type: UpdateEmailModule, declarations: [UpdateEmailComponent, MyAccountV2EmailComponent], imports: [CommonModule,
            FormsModule,
            ReactiveFormsModule,
            SpinnerModule,
            UrlModule,
            RouterModule,
            I18nModule,
            FormErrorsModule,
            PasswordVisibilityToggleModule,
            FeaturesConfigModule,
            MessageComponentModule], exports: [UpdateEmailComponent, MyAccountV2EmailComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: UpdateEmailModule, providers: [
            provideDefaultConfig({
                cmsComponents: {
                    UpdateEmailComponent: {
                        component: UpdateEmailComponent,
                        guards: [AuthGuard],
                        providers: [
                            {
                                provide: UpdateEmailComponentService,
                                useClass: UpdateEmailComponentService,
                                deps: [
                                    UserEmailFacade,
                                    RoutingService,
                                    GlobalMessageService,
                                    AuthService,
                                    AuthRedirectService,
                                ],
                            },
                        ],
                    },
                },
            }),
            provideDefaultConfigFactory(() => inject(USE_MY_ACCOUNT_V2_EMAIL) ? myAccountV2EmailMapping : {}),
        ], imports: [CommonModule,
            FormsModule,
            ReactiveFormsModule,
            SpinnerModule,
            UrlModule,
            RouterModule,
            I18nModule,
            FormErrorsModule,
            PasswordVisibilityToggleModule,
            FeaturesConfigModule,
            MessageComponentModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: UpdateEmailModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        FormsModule,
                        ReactiveFormsModule,
                        SpinnerModule,
                        UrlModule,
                        RouterModule,
                        I18nModule,
                        FormErrorsModule,
                        PasswordVisibilityToggleModule,
                        FeaturesConfigModule,
                        MessageComponentModule,
                    ],
                    declarations: [UpdateEmailComponent, MyAccountV2EmailComponent],
                    exports: [UpdateEmailComponent, MyAccountV2EmailComponent],
                    providers: [
                        provideDefaultConfig({
                            cmsComponents: {
                                UpdateEmailComponent: {
                                    component: UpdateEmailComponent,
                                    guards: [AuthGuard],
                                    providers: [
                                        {
                                            provide: UpdateEmailComponentService,
                                            useClass: UpdateEmailComponentService,
                                            deps: [
                                                UserEmailFacade,
                                                RoutingService,
                                                GlobalMessageService,
                                                AuthService,
                                                AuthRedirectService,
                                            ],
                                        },
                                    ],
                                },
                            },
                        }),
                        provideDefaultConfigFactory(() => inject(USE_MY_ACCOUNT_V2_EMAIL) ? myAccountV2EmailMapping : {}),
                    ],
                }]
        }] });
