// @spartacus/checkout/esm2022/b2b/components/checkout-delivery-address/checkout-delivery-address.module.mjs
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { RouterModule } from '@angular/router';
import { CartValidationGuard } from '@spartacus/cart/base/core';
import { CartNotEmptyGuard, CheckoutAuthGuard, } from '@spartacus/checkout/base/components';
import { I18nModule, provideDefaultConfig } from '@spartacus/core';
import { CardModule, SpinnerModule } from '@spartacus/storefront';
import { B2BCheckoutDeliveryAddressComponent } from './checkout-delivery-address.component';
import { AddressFormModule } from '@spartacus/user/profile/components';
import * as i0 from "@angular/core";
export class B2BCheckoutDeliveryAddressModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: B2BCheckoutDeliveryAddressModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "17.0.5", ngImport: i0, type: B2BCheckoutDeliveryAddressModule, declarations: [B2BCheckoutDeliveryAddressComponent], imports: [CommonModule,
            RouterModule,
            AddressFormModule,
            CardModule,
            SpinnerModule,
            I18nModule], exports: [B2BCheckoutDeliveryAddressComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: B2BCheckoutDeliveryAddressModule, providers: [
            provideDefaultConfig({
                cmsComponents: {
                    CheckoutDeliveryAddress: {
                        component: B2BCheckoutDeliveryAddressComponent,
                        guards: [CheckoutAuthGuard, CartNotEmptyGuard, CartValidationGuard],
                    },
                },
            }),
        ], imports: [CommonModule,
            RouterModule,
            AddressFormModule,
            CardModule,
            SpinnerModule,
            I18nModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: B2BCheckoutDeliveryAddressModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        RouterModule,
                        AddressFormModule,
                        CardModule,
                        SpinnerModule,
                        I18nModule,
                    ],
                    providers: [
                        provideDefaultConfig({
                            cmsComponents: {
                                CheckoutDeliveryAddress: {
                                    component: B2BCheckoutDeliveryAddressComponent,
                                    guards: [CheckoutAuthGuard, CartNotEmptyGuard, CartValidationGuard],
                                },
                            },
                        }),
                    ],
                    declarations: [B2BCheckoutDeliveryAddressComponent],
                    exports: [B2BCheckoutDeliveryAddressComponent],
                }]
        }] });
