// @spartacus/storefront/esm2022/cms-components/content/video/video.module.mjs
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { RouterModule } from '@angular/router';
import { I18nModule, provideDefaultConfig } from '@spartacus/core';
import { SpinnerModule } from '../../../shared/components/spinner/spinner.module';
import { GenericLinkModule } from '../../../shared/components/generic-link/generic-link.module';
import { MediaModule } from '../../../shared/components/media/media.module';
import { VideoComponent } from './video.component';
import * as i0 from "@angular/core";
export class VideoModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: VideoModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "17.0.5", ngImport: i0, type: VideoModule, declarations: [VideoComponent], imports: [CommonModule,
            RouterModule,
            GenericLinkModule,
            MediaModule,
            I18nModule,
            SpinnerModule], exports: [VideoComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: VideoModule, providers: [
            provideDefaultConfig({
                cmsComponents: {
                    VideoComponent: {
                        component: VideoComponent,
                    },
                },
            }),
        ], imports: [CommonModule,
            RouterModule,
            GenericLinkModule,
            MediaModule,
            I18nModule,
            SpinnerModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: VideoModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        RouterModule,
                        GenericLinkModule,
                        MediaModule,
                        I18nModule,
                        SpinnerModule,
                    ],
                    providers: [
                        provideDefaultConfig({
                            cmsComponents: {
                                VideoComponent: {
                                    component: VideoComponent,
                                },
                            },
                        }),
                    ],
                    declarations: [VideoComponent],
                    exports: [VideoComponent],
                }]
        }] });
