// @spartacus/storefront/esm2022/cms-components/product/stock-notification/stock-notification.module.mjs
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { StockNotificationComponent } from './stock-notification.component';
import { I18nModule, provideDefaultConfig, UrlModule, } from '@spartacus/core';
import { StockNotificationDialogComponent } from './stock-notification-dialog/stock-notification-dialog.component';
import { SpinnerModule } from '../../../shared/components/spinner/spinner.module';
import { KeyboardFocusModule } from '../../../layout/a11y/keyboard-focus/keyboard-focus.module';
import { RouterModule } from '@angular/router';
import { defaultStockNotificationLayoutConfig } from './stock-notification-dialog/default-stock-notification-layout.config';
import * as i0 from "@angular/core";
export class StockNotificationModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: StockNotificationModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "17.0.5", ngImport: i0, type: StockNotificationModule, declarations: [StockNotificationComponent, StockNotificationDialogComponent], imports: [CommonModule,
            RouterModule,
            I18nModule,
            SpinnerModule,
            UrlModule,
            KeyboardFocusModule], exports: [StockNotificationComponent, StockNotificationDialogComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: StockNotificationModule, providers: [
            provideDefaultConfig(defaultStockNotificationLayoutConfig),
            provideDefaultConfig({
                cmsComponents: {
                    StockNotificationComponent: {
                        component: StockNotificationComponent,
                    },
                },
            }),
        ], imports: [CommonModule,
            RouterModule,
            I18nModule,
            SpinnerModule,
            UrlModule,
            KeyboardFocusModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: StockNotificationModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [StockNotificationComponent, StockNotificationDialogComponent],
                    imports: [
                        CommonModule,
                        RouterModule,
                        I18nModule,
                        SpinnerModule,
                        UrlModule,
                        KeyboardFocusModule,
                    ],
                    providers: [
                        provideDefaultConfig(defaultStockNotificationLayoutConfig),
                        provideDefaultConfig({
                            cmsComponents: {
                                StockNotificationComponent: {
                                    component: StockNotificationComponent,
                                },
                            },
                        }),
                    ],
                    exports: [StockNotificationComponent, StockNotificationDialogComponent],
                }]
        }] });
