// @spartacus/cart/esm2022/wish-list/components/wish-list-components.module.mjs
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { RouterModule } from '@angular/router';
import { AuthGuard, I18nModule, provideDefaultConfig, UrlModule, } from '@spartacus/core';
import { AtMessageModule, ItemCounterModule, MediaModule, PageComponentModule, StarRatingModule, } from '@spartacus/storefront';
import { WishListItemComponent } from './wish-list-item/wish-list-item.component';
import { WishListComponent } from './wish-list/wish-list.component';
import * as i0 from "@angular/core";
export class WishListComponentsModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: WishListComponentsModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "17.0.5", ngImport: i0, type: WishListComponentsModule, declarations: [WishListComponent, WishListItemComponent], imports: [AtMessageModule,
            CommonModule,
            I18nModule,
            ItemCounterModule,
            MediaModule,
            PageComponentModule,
            RouterModule,
            StarRatingModule,
            UrlModule], exports: [WishListComponent, WishListItemComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: WishListComponentsModule, providers: [
            provideDefaultConfig({
                cmsComponents: {
                    WishListComponent: {
                        component: WishListComponent,
                        data: {
                            composition: {
                                inner: ['ProductAddToCartComponent'],
                            },
                        },
                        guards: [AuthGuard],
                    },
                },
            }),
        ], imports: [AtMessageModule,
            CommonModule,
            I18nModule,
            ItemCounterModule,
            MediaModule,
            PageComponentModule,
            RouterModule,
            StarRatingModule,
            UrlModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: WishListComponentsModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        AtMessageModule,
                        CommonModule,
                        I18nModule,
                        ItemCounterModule,
                        MediaModule,
                        PageComponentModule,
                        RouterModule,
                        StarRatingModule,
                        UrlModule,
                    ],
                    providers: [
                        provideDefaultConfig({
                            cmsComponents: {
                                WishListComponent: {
                                    component: WishListComponent,
                                    data: {
                                        composition: {
                                            inner: ['ProductAddToCartComponent'],
                                        },
                                    },
                                    guards: [AuthGuard],
                                },
                            },
                        }),
                    ],
                    declarations: [WishListComponent, WishListItemComponent],
                    exports: [WishListComponent, WishListItemComponent],
                }]
        }] });
