// @spartacus/cart/esm2022/base/root/facade/active-cart.facade.mjs
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { Injectable } from '@angular/core';
import { facadeFactory } from '@spartacus/core';
import { CART_BASE_CORE_FEATURE } from '../feature-name';
import * as i0 from "@angular/core";
export class ActiveCartFacade {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: ActiveCartFacade, deps: [], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: ActiveCartFacade, providedIn: 'root', useFactory: () => facadeFactory({
            facade: ActiveCartFacade,
            feature: CART_BASE_CORE_FEATURE,
            methods: [
                'getActive',
                'takeActive',
                'getActiveCartId',
                'takeActiveCartId',
                'getEntries',
                'getLastEntry',
                'getLoading',
                'isStable',
                'addEntry',
                'removeEntry',
                'updateEntry',
                'getEntry',
                'addEmail',
                'getAssignedUser',
                'isGuestCart',
                'addEntries',
                'requireLoadedCart',
                'reloadActiveCart',
                'hasPickupItems',
                'hasDeliveryItems',
                'getPickupEntries',
                'getDeliveryEntries',
            ],
            async: true,
        }) }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: ActiveCartFacade, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                    useFactory: () => facadeFactory({
                        facade: ActiveCartFacade,
                        feature: CART_BASE_CORE_FEATURE,
                        methods: [
                            'getActive',
                            'takeActive',
                            'getActiveCartId',
                            'takeActiveCartId',
                            'getEntries',
                            'getLastEntry',
                            'getLoading',
                            'isStable',
                            'addEntry',
                            'removeEntry',
                            'updateEntry',
                            'getEntry',
                            'addEmail',
                            'getAssignedUser',
                            'isGuestCart',
                            'addEntries',
                            'requireLoadedCart',
                            'reloadActiveCart',
                            'hasPickupItems',
                            'hasDeliveryItems',
                            'getPickupEntries',
                            'getDeliveryEntries',
                        ],
                        async: true,
                    }),
                }]
        }] });
