// @spartacus/cart/esm2022/base/root/facade/cart-voucher.facade.mjs
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { Injectable } from '@angular/core';
import { facadeFactory } from '@spartacus/core';
import { CART_BASE_CORE_FEATURE } from '../feature-name';
import * as i0 from "@angular/core";
export class CartVoucherFacade {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: CartVoucherFacade, deps: [], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: CartVoucherFacade, providedIn: 'root', useFactory: () => facadeFactory({
            facade: CartVoucherFacade,
            feature: CART_BASE_CORE_FEATURE,
            methods: [
                'addVoucher',
                'removeVoucher',
                'getAddVoucherResultError',
                'getAddVoucherResultSuccess',
                'getAddVoucherResultLoading',
                'resetAddVoucherProcessingState',
            ],
            async: true,
        }) }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: CartVoucherFacade, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                    useFactory: () => facadeFactory({
                        facade: CartVoucherFacade,
                        feature: CART_BASE_CORE_FEATURE,
                        methods: [
                            'addVoucher',
                            'removeVoucher',
                            'getAddVoucherResultError',
                            'getAddVoucherResultSuccess',
                            'getAddVoucherResultLoading',
                            'resetAddVoucherProcessingState',
                        ],
                        async: true,
                    }),
                }]
        }] });
