// @spartacus/user/esm2022/profile/components/update-profile/update-profile.module.mjs
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { CommonModule } from '@angular/common';
import { NgModule, inject } from '@angular/core';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { RouterModule } from '@angular/router';
import { NgSelectModule } from '@ng-select/ng-select';
import { AuthGuard, FeaturesConfigModule, GlobalMessageService, I18nModule, UrlModule, provideDefaultConfig, provideDefaultConfigFactory, } from '@spartacus/core';
import { FormErrorsModule, NgSelectA11yModule, SpinnerModule, } from '@spartacus/storefront';
import { UserProfileFacade } from '@spartacus/user/profile/root';
import { MyAccountV2ProfileComponent } from './my-account-v2-profile.component';
import { UpdateProfileComponentService } from './update-profile-component.service';
import { UpdateProfileComponent } from './update-profile.component';
import { USE_MY_ACCOUNT_V2_PROFILE } from './use-my-account-v2-profile';
import * as i0 from "@angular/core";
const myAccountV2ProfileMapping = {
    cmsComponents: {
        UpdateProfileComponent: {
            component: MyAccountV2ProfileComponent,
        },
    },
};
export class UpdateProfileModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: UpdateProfileModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "17.0.5", ngImport: i0, type: UpdateProfileModule, declarations: [UpdateProfileComponent, MyAccountV2ProfileComponent], imports: [CommonModule,
            FormsModule,
            ReactiveFormsModule,
            SpinnerModule,
            I18nModule,
            FormErrorsModule,
            RouterModule,
            UrlModule,
            NgSelectModule,
            NgSelectA11yModule,
            FeaturesConfigModule], exports: [UpdateProfileComponent, MyAccountV2ProfileComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: UpdateProfileModule, providers: [
            provideDefaultConfig({
                cmsComponents: {
                    UpdateProfileComponent: {
                        component: UpdateProfileComponent,
                        guards: [AuthGuard],
                        providers: [
                            {
                                provide: UpdateProfileComponentService,
                                useClass: UpdateProfileComponentService,
                                deps: [UserProfileFacade, GlobalMessageService],
                            },
                        ],
                    },
                },
            }),
            provideDefaultConfigFactory(() => inject(USE_MY_ACCOUNT_V2_PROFILE) ? myAccountV2ProfileMapping : {}),
        ], imports: [CommonModule,
            FormsModule,
            ReactiveFormsModule,
            SpinnerModule,
            I18nModule,
            FormErrorsModule,
            RouterModule,
            UrlModule,
            NgSelectModule,
            NgSelectA11yModule,
            FeaturesConfigModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: UpdateProfileModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        FormsModule,
                        ReactiveFormsModule,
                        SpinnerModule,
                        I18nModule,
                        FormErrorsModule,
                        RouterModule,
                        UrlModule,
                        NgSelectModule,
                        NgSelectA11yModule,
                        FeaturesConfigModule,
                    ],
                    declarations: [UpdateProfileComponent, MyAccountV2ProfileComponent],
                    exports: [UpdateProfileComponent, MyAccountV2ProfileComponent],
                    providers: [
                        provideDefaultConfig({
                            cmsComponents: {
                                UpdateProfileComponent: {
                                    component: UpdateProfileComponent,
                                    guards: [AuthGuard],
                                    providers: [
                                        {
                                            provide: UpdateProfileComponentService,
                                            useClass: UpdateProfileComponentService,
                                            deps: [UserProfileFacade, GlobalMessageService],
                                        },
                                    ],
                                },
                            },
                        }),
                        provideDefaultConfigFactory(() => inject(USE_MY_ACCOUNT_V2_PROFILE) ? myAccountV2ProfileMapping : {}),
                    ],
                }]
        }] });
