// @spartacus/asm/esm2022/customer-360/root/asm-customer-360-root.module.mjs
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { NgModule } from '@angular/core';
import { ASM_FEATURE } from '@spartacus/asm/root';
import { provideDefaultConfig } from '@spartacus/core';
import { PageComponentModule } from '@spartacus/storefront';
import { ASM_CUSTOMER_360_CORE_FEATURE, ASM_CUSTOMER_360_FEATURE, } from './feature-name';
import { HTTP_INTERCEPTORS } from '@angular/common/http';
import { SiteContextInterceptor } from './interceptors/site-context.interceptor';
import * as i0 from "@angular/core";
export class AsmCustomer360RootModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: AsmCustomer360RootModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "17.0.5", ngImport: i0, type: AsmCustomer360RootModule, imports: [PageComponentModule] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: AsmCustomer360RootModule, providers: [
            provideDefaultConfig({
                featureModules: {
                    [ASM_CUSTOMER_360_FEATURE]: {
                        dependencies: [ASM_FEATURE],
                    },
                    [ASM_CUSTOMER_360_CORE_FEATURE]: ASM_CUSTOMER_360_FEATURE,
                },
            }),
            {
                provide: HTTP_INTERCEPTORS,
                useExisting: SiteContextInterceptor,
                multi: true,
            },
        ], imports: [PageComponentModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: AsmCustomer360RootModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [PageComponentModule],
                    providers: [
                        provideDefaultConfig({
                            featureModules: {
                                [ASM_CUSTOMER_360_FEATURE]: {
                                    dependencies: [ASM_FEATURE],
                                },
                                [ASM_CUSTOMER_360_CORE_FEATURE]: ASM_CUSTOMER_360_FEATURE,
                            },
                        }),
                        {
                            provide: HTTP_INTERCEPTORS,
                            useExisting: SiteContextInterceptor,
                            multi: true,
                        },
                    ],
                }]
        }] });
