// @spartacus/cart/esm2022/saved-cart/root/events/saved-cart.events.mjs
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { CartEvent } from '@spartacus/cart/base/root';
/**
 * Base saved cart event. Most cart events should have these properties.
 */
export class SavedCartEvent extends CartEvent {
}
// =====================================================================
export class SaveCartEvent extends SavedCartEvent {
    /**
     * Event's type
     */
    static { this.type = 'SaveCartEvent'; }
}
export class SaveCartSuccessEvent extends SavedCartEvent {
    /**
     * Event's type
     */
    static { this.type = 'SaveCartSuccessEvent'; }
}
export class SaveCartFailEvent extends SavedCartEvent {
    /**
     * Event's type
     */
    static { this.type = 'SaveCartEvent'; }
}
export class RestoreSavedCartEvent extends SavedCartEvent {
    /**
     * Event's type
     */
    static { this.type = 'RestoreSavedCartEvent'; }
}
export class RestoreSavedCartSuccessEvent extends SavedCartEvent {
    /**
     * Event's type
     */
    static { this.type = 'RestoreSavedCartSuccessEvent'; }
}
export class RestoreSavedCartFailEvent extends SavedCartEvent {
    /**
     * Event's type
     */
    static { this.type = 'RestoreSavedCartFailEvent'; }
}
export class EditSavedCartEvent extends SavedCartEvent {
    /**
     * Event's type
     */
    static { this.type = 'EditSavedCartEvent'; }
}
export class EditSavedCartSuccessEvent extends SavedCartEvent {
    /**
     * Event's type
     */
    static { this.type = 'EditSavedCartSuccessEvent'; }
}
export class EditSavedCartFailEvent extends SavedCartEvent {
    /**
     * Event's type
     */
    static { this.type = 'EditSavedCartFailEvent'; }
}
export class CloneSavedCartEvent extends SavedCartEvent {
    /**
     * Event's type
     */
    static { this.type = 'CloneSavedCartEvent'; }
}
export class CloneSavedCartSuccessEvent extends SavedCartEvent {
    /**
     * Event's type
     */
    static { this.type = 'CloneSavedCartSuccessEvent'; }
}
export class CloneSavedCartFailEvent extends SavedCartEvent {
    /**
     * Event's type
     */
    static { this.type = 'CloneSavedCartFailEvent'; }
}
