// @spartacus/schematics/src/shared/utils/lib-utils.js
"use strict";
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.finalizeInstallation = exports.calculateCrossLibrarySort = exports.calculateCrossFeatureSort = exports.configureB2bFeatures = exports.dependencyExists = exports.addPackageJsonDependenciesForLibrary = exports.addPackageJsonDependencies = exports.installPackageJsonDependencies = exports.createNodePackageInstallationTask = exports.addLibraryStyles = exports.addFeatureTranslations = exports.createSpartacusWrapperModuleFileName = exports.createSpartacusFeatureFileName = exports.createSpartacusFeatureFolderPath = exports.checkAppStructure = exports.addLibraryFeature = exports.shouldAddFeature = void 0;
const strings_1 = require("@angular-devkit/core/src/utils/strings");
const schematics_1 = require("@angular-devkit/schematics");
const tasks_1 = require("@angular-devkit/schematics/tasks");
const dependencies_1 = require("@schematics/angular/utility/dependencies");
const constants_1 = require("../constants");
const libs_constants_1 = require("../libs-constants");
const config_utils_1 = require("./config-utils");
const feature_utils_1 = require("./feature-utils");
const graph_utils_1 = require("./graph-utils");
const import_utils_1 = require("./import-utils");
const logger_utils_1 = require("./logger-utils");
const new_module_utils_1 = require("./new-module-utils");
const package_utils_1 = require("./package-utils");
const program_1 = require("./program");
const project_tsconfig_paths_1 = require("./project-tsconfig-paths");
const styling_utils_1 = require("./styling-utils");
const workspace_utils_1 = require("./workspace-utils");
function shouldAddFeature(feature, features = []) {
    return features.includes(feature);
}
exports.shouldAddFeature = shouldAddFeature;
function addLibraryFeature(options, config) {
    return (tree, context) => {
        const spartacusFeatureModuleExistsInApp = checkAppStructure(tree, options.project);
        if (!spartacusFeatureModuleExistsInApp) {
            context.logger.info('Scaffolding the new app structure...');
            context.logger.warn('Please migrate manually the rest of your feature modules to the new app structure: https://sap.github.io/spartacus-docs/reference-app-structure/');
        }
        return (0, schematics_1.chain)([
            (0, logger_utils_1.debugLogRule)((0, logger_utils_1.formatFeatureStart)(config.library.featureName, `adding...`), options.debug),
            spartacusFeatureModuleExistsInApp ? (0, schematics_1.noop)() : (0, workspace_utils_1.scaffoldStructure)(options),
            handleFeature(options, config),
            config.styles ? addLibraryStyles(config.styles, options) : (0, schematics_1.noop)(),
            config.assets ? addLibraryAssets(config.assets, options) : (0, schematics_1.noop)(),
            (0, logger_utils_1.debugLogRule)((0, logger_utils_1.formatFeatureComplete)(config.library.featureName, `added.`), options.debug),
        ]);
    };
}
exports.addLibraryFeature = addLibraryFeature;
function checkAppStructure(tree, project) {
    const { buildPaths } = (0, project_tsconfig_paths_1.getProjectTsConfigPaths)(tree, project);
    if (!buildPaths.length) {
        throw new schematics_1.SchematicsException(`Could not find any tsconfig file. Can't find ${libs_constants_1.SPARTACUS_FEATURES_NG_MODULE}.`);
    }
    const basePath = process.cwd();
    for (const tsconfigPath of buildPaths) {
        if ((0, feature_utils_1.getSpartacusFeaturesModule)(tree, basePath, tsconfigPath)) {
            return true;
        }
    }
    return false;
}
exports.checkAppStructure = checkAppStructure;
function handleFeature(options, config) {
    return (tree, _context) => {
        const rules = [];
        rules.push((0, new_module_utils_1.ensureModuleExists)({
            name: createSpartacusFeatureFileName(config.moduleName),
            path: createSpartacusFeatureFolderPath(config.folderName),
            module: libs_constants_1.SPARTACUS_FEATURES_MODULE,
            project: options.project,
        }));
        rules.push(addRootModule(options, config));
        rules.push(addFeatureModule(options, config));
        rules.push(addFeatureTranslations(options, config));
        rules.push(addCustomConfig(options, config));
        if (config.library.b2b) {
            rules.push(configureB2bFeatures(options, (0, package_utils_1.readPackageJson)(tree)));
        }
        return (0, schematics_1.chain)(rules);
    };
}
function createSpartacusFeatureFolderPath(folderName) {
    return `app/spartacus/features/${(0, strings_1.dasherize)(folderName)}`;
}
exports.createSpartacusFeatureFolderPath = createSpartacusFeatureFolderPath;
function createSpartacusFeatureFileName(name) {
    return `${(0, strings_1.dasherize)(name)}-feature`;
}
exports.createSpartacusFeatureFileName = createSpartacusFeatureFileName;
function createSpartacusWrapperModuleFileName(name) {
    const normalizedName = name.replace('module', '').replace('Module', '');
    return `${(0, strings_1.dasherize)(normalizedName)}-wrapper`;
}
exports.createSpartacusWrapperModuleFileName = createSpartacusWrapperModuleFileName;
function addRootModule(options, config) {
    return (tree) => {
        if (!config.rootModule) {
            return tree;
        }
        const basePath = process.cwd();
        const moduleFileName = createModuleFileName(config);
        const { buildPaths } = (0, project_tsconfig_paths_1.getProjectTsConfigPaths)(tree, options.project);
        for (const tsconfigPath of buildPaths) {
            const { appSourceFiles } = (0, program_1.createProgram)(tree, basePath, tsconfigPath);
            for (const sourceFile of appSourceFiles) {
                if (!sourceFile.getFilePath().endsWith('/' + moduleFileName)) {
                    continue;
                }
                (0, new_module_utils_1.addModuleImport)(sourceFile, {
                    import: {
                        moduleSpecifier: config.rootModule.importPath,
                        namedImports: [config.rootModule.name],
                    },
                    content: config.rootModule.content || config.rootModule.name,
                });
                (0, program_1.saveAndFormat)(sourceFile);
                break;
            }
        }
        return tree;
    };
}
function addFeatureModule(options, config) {
    return (tree) => {
        const basePath = process.cwd();
        const moduleFileName = createModuleFileName(config);
        const { buildPaths } = (0, project_tsconfig_paths_1.getProjectTsConfigPaths)(tree, options.project);
        for (const tsconfigPath of buildPaths) {
            const { appSourceFiles } = (0, program_1.createProgram)(tree, basePath, tsconfigPath);
            for (const sourceFile of appSourceFiles) {
                if (!sourceFile.getFilePath().endsWith('/' + moduleFileName)) {
                    continue;
                }
                addToFeatureModule(sourceFile, appSourceFiles);
                (0, program_1.saveAndFormat)(sourceFile);
                break;
            }
        }
        return tree;
    };
    function addToFeatureModule(sourceFile, appSourceFiles) {
        const configFeatures = [].concat(config.featureModule);
        for (let i = 0; i < configFeatures.length; i++) {
            const featureModule = configFeatures[i];
            // if it's already in a wrapper module
            if ((0, feature_utils_1.findFeatureModule)(featureModule, appSourceFiles)) {
                break;
            }
            let content = `${constants_1.PROVIDE_CONFIG_FUNCTION}(<${constants_1.CMS_CONFIG}>{
            featureModules: {`;
            if (options.lazy) {
                let lazyLoadingChunkName = config.moduleName;
                if (config.lazyLoadingChunk) {
                    const namedImportsContent = config.lazyLoadingChunk.namedImports[i];
                    lazyLoadingChunkName = `[${namedImportsContent}]`;
                    (0, import_utils_1.createImports)(sourceFile, config.lazyLoadingChunk);
                }
                content =
                    content +
                        `${lazyLoadingChunkName}: {
                module: () =>
                  import('${featureModule.importPath}').then((m) => m.${featureModule.name}),
              },`;
                (0, new_module_utils_1.addModuleProvider)(sourceFile, {
                    import: [
                        {
                            moduleSpecifier: libs_constants_1.SPARTACUS_CORE,
                            namedImports: [constants_1.PROVIDE_CONFIG_FUNCTION, constants_1.CMS_CONFIG],
                        },
                    ],
                    content: content + `}})`,
                });
            }
            else {
                (0, new_module_utils_1.addModuleImport)(sourceFile, {
                    import: {
                        moduleSpecifier: featureModule.importPath,
                        namedImports: [featureModule.name],
                    },
                    content: featureModule.content || featureModule.name,
                });
            }
        }
    }
}
function addFeatureTranslations(options, config) {
    return (tree) => {
        if (!config.i18n) {
            return tree;
        }
        const basePath = process.cwd();
        const moduleFileName = createModuleFileName(config);
        const { buildPaths } = (0, project_tsconfig_paths_1.getProjectTsConfigPaths)(tree, options.project);
        for (const tsconfigPath of buildPaths) {
            const { appSourceFiles } = (0, program_1.createProgram)(tree, basePath, tsconfigPath);
            for (const sourceFile of appSourceFiles) {
                if (!sourceFile.getFilePath().endsWith('/' + moduleFileName)) {
                    continue;
                }
                (0, new_module_utils_1.addModuleProvider)(sourceFile, {
                    import: [
                        {
                            moduleSpecifier: libs_constants_1.SPARTACUS_CORE,
                            namedImports: [constants_1.PROVIDE_CONFIG_FUNCTION, constants_1.I18N_CONFIG],
                        },
                        {
                            moduleSpecifier: config.i18n.importPath,
                            namedImports: [config.i18n.chunks, config.i18n.resources],
                        },
                    ],
                    content: `${constants_1.PROVIDE_CONFIG_FUNCTION}(<${constants_1.I18N_CONFIG}>{
              i18n: {
                resources: ${config.i18n.resources},
                chunks: ${config.i18n.chunks},
              },
            })`,
                });
                (0, program_1.saveAndFormat)(sourceFile);
                break;
            }
        }
        return tree;
    };
}
exports.addFeatureTranslations = addFeatureTranslations;
function addCustomConfig(options, config) {
    return (tree) => {
        var _a;
        if (!config.customConfig) {
            return tree;
        }
        const basePath = process.cwd();
        const moduleFileName = createModuleFileName(config);
        const { buildPaths } = (0, project_tsconfig_paths_1.getProjectTsConfigPaths)(tree, options.project);
        for (const tsconfigPath of buildPaths) {
            const { appSourceFiles } = (0, program_1.createProgram)(tree, basePath, tsconfigPath);
            for (const sourceFile of appSourceFiles) {
                if (!sourceFile.getFilePath().endsWith('/' + moduleFileName)) {
                    continue;
                }
                const customConfigs = [].concat((_a = config.customConfig(options).providers) !== null && _a !== void 0 ? _a : []);
                customConfigs.forEach((customConfig) => {
                    (0, new_module_utils_1.addModuleProvider)(sourceFile, {
                        import: [
                            {
                                moduleSpecifier: libs_constants_1.SPARTACUS_CORE,
                                namedImports: [constants_1.PROVIDE_CONFIG_FUNCTION],
                            },
                            ...customConfig.import,
                        ],
                        content: `${constants_1.PROVIDE_CONFIG_FUNCTION}(${customConfig.content})`,
                    });
                });
                (0, program_1.saveAndFormat)(sourceFile);
                break;
            }
        }
        return tree;
    };
}
function addLibraryAssets(assetsConfig, options) {
    return (tree) => {
        var _a, _b, _c, _d;
        const { path, workspace: angularJson } = (0, workspace_utils_1.getWorkspace)(tree);
        const defaultProject = (0, workspace_utils_1.getDefaultProjectNameFromWorkspace)(tree);
        const project = options.project || defaultProject;
        const architect = angularJson.projects[project].architect;
        // `build` architect section
        const architectBuild = architect === null || architect === void 0 ? void 0 : architect.build;
        const buildAssets = createAssetsArray(assetsConfig, (_a = architectBuild === null || architectBuild === void 0 ? void 0 : architectBuild.options) === null || _a === void 0 ? void 0 : _a.assets);
        const buildOptions = Object.assign(Object.assign({}, architectBuild === null || architectBuild === void 0 ? void 0 : architectBuild.options), { assets: buildAssets });
        // `test` architect section
        const architectTest = architect === null || architect === void 0 ? void 0 : architect.test;
        const testAssets = createAssetsArray(assetsConfig, (_b = architectTest === null || architectTest === void 0 ? void 0 : architectTest.options) === null || _b === void 0 ? void 0 : _b.assets);
        const testOptions = Object.assign(Object.assign({}, architectTest === null || architectTest === void 0 ? void 0 : architectTest.options), { assets: testAssets });
        const updatedAngularJson = Object.assign(Object.assign({}, angularJson), { projects: Object.assign(Object.assign({}, angularJson.projects), { [project]: Object.assign(Object.assign({}, angularJson.projects[project]), { architect: Object.assign(Object.assign({}, architect), { build: Object.assign(Object.assign({}, architectBuild), { options: buildOptions }), test: Object.assign(Object.assign({}, architectTest), { options: testOptions }) }) }) }) });
        const initialContent = (_d = (_c = tree.read(path)) === null || _c === void 0 ? void 0 : _c.toString(constants_1.UTF_8)) !== null && _d !== void 0 ? _d : '';
        const toUpdate = JSON.stringify(updatedAngularJson, null, 2);
        // prevent the unnecessary Angular logs about the files being updated
        if (initialContent !== toUpdate) {
            tree.overwrite(path, toUpdate);
        }
    };
}
function createAssetsArray(assetsConfig, angularJsonAssets = []) {
    for (const asset of angularJsonAssets) {
        if (typeof asset === 'object') {
            if (asset.glob === assetsConfig.glob &&
                asset.input === `./node_modules/@spartacus/${assetsConfig.input}` &&
                asset.output === (assetsConfig.output || 'assets/')) {
                return angularJsonAssets;
            }
        }
    }
    angularJsonAssets = [
        ...angularJsonAssets,
        {
            glob: assetsConfig.glob,
            input: `./node_modules/@spartacus/${assetsConfig.input}`,
            output: assetsConfig.output || 'assets/',
        },
    ];
    return angularJsonAssets;
}
function addLibraryStyles(stylingConfig, options) {
    return (tree, _context) => {
        var _a, _b, _c, _d, _e, _f;
        const defaultProject = (0, workspace_utils_1.getDefaultProjectNameFromWorkspace)(tree);
        const project = options.project || defaultProject;
        const libraryScssPath = `${(0, workspace_utils_1.getSourceRoot)(tree, {
            project: project,
        })}/styles/spartacus/${stylingConfig.scssFileName}`;
        const libraryStylesImport = `@import "${stylingConfig.importStyle}";`;
        if (tree.exists(libraryScssPath)) {
            const initialContent = (_b = (_a = tree.read(libraryScssPath)) === null || _a === void 0 ? void 0 : _a.toString(constants_1.UTF_8)) !== null && _b !== void 0 ? _b : '';
            let content = initialContent;
            if (!content.includes(libraryStylesImport)) {
                content += `\n${libraryStylesImport}`;
            }
            // prevent the unnecessary Angular logs about the files being updated
            if (initialContent !== content) {
                tree.overwrite(libraryScssPath, content);
            }
            return tree;
        }
        const styleConfigFilePath = (0, styling_utils_1.getStylesConfigFilePath)((0, workspace_utils_1.getSourceRoot)(tree, {
            project: project,
        }));
        let libraryScssFileContent = '';
        if (tree.exists(styleConfigFilePath)) {
            const styleConfigImportPath = (0, styling_utils_1.getRelativeStyleConfigImportPath)((0, workspace_utils_1.getProject)(tree, project), libraryScssPath);
            const stylesConfigImport = `@import "${styleConfigImportPath}";`;
            libraryScssFileContent += `${stylesConfigImport}\n`;
        }
        libraryScssFileContent += `${libraryStylesImport}\n`;
        tree.create(libraryScssPath, libraryScssFileContent);
        const { path, workspace: angularJson } = (0, workspace_utils_1.getWorkspace)(tree);
        const architect = angularJson.projects[project].architect;
        // `build` architect section
        const architectBuild = architect === null || architect === void 0 ? void 0 : architect.build;
        const buildOptions = Object.assign(Object.assign({}, architectBuild === null || architectBuild === void 0 ? void 0 : architectBuild.options), { styles: [
                ...(((_c = architectBuild === null || architectBuild === void 0 ? void 0 : architectBuild.options) === null || _c === void 0 ? void 0 : _c.styles)
                    ? (_d = architectBuild === null || architectBuild === void 0 ? void 0 : architectBuild.options) === null || _d === void 0 ? void 0 : _d.styles
                    : []),
                libraryScssPath,
            ] });
        // `test` architect section
        const architectTest = architect === null || architect === void 0 ? void 0 : architect.test;
        const testOptions = Object.assign(Object.assign({}, architectTest === null || architectTest === void 0 ? void 0 : architectTest.options), { styles: [
                ...(((_e = architectTest === null || architectTest === void 0 ? void 0 : architectTest.options) === null || _e === void 0 ? void 0 : _e.styles)
                    ? (_f = architectTest === null || architectTest === void 0 ? void 0 : architectTest.options) === null || _f === void 0 ? void 0 : _f.styles
                    : []),
                libraryScssPath,
            ] });
        const updatedAngularJson = Object.assign(Object.assign({}, angularJson), { projects: Object.assign(Object.assign({}, angularJson.projects), { [project]: Object.assign(Object.assign({}, angularJson.projects[project]), { architect: Object.assign(Object.assign({}, architect), { build: Object.assign(Object.assign({}, architectBuild), { options: buildOptions }), test: Object.assign(Object.assign({}, architectTest), { options: testOptions }) }) }) }) });
        tree.overwrite(path, JSON.stringify(updatedAngularJson, null, 2));
    };
}
exports.addLibraryStyles = addLibraryStyles;
function createNodePackageInstallationTask(context) {
    return context.addTask(new tasks_1.NodePackageInstallTask());
}
exports.createNodePackageInstallationTask = createNodePackageInstallationTask;
function installPackageJsonDependencies() {
    return (tree, context) => {
        createNodePackageInstallationTask(context);
        return tree;
    };
}
exports.installPackageJsonDependencies = installPackageJsonDependencies;
function addPackageJsonDependencies(dependencies, packageJson) {
    return (tree, context) => {
        for (const dependency of dependencies) {
            if (!dependencyExists(dependency, packageJson)) {
                (0, dependencies_1.addPackageJsonDependency)(tree, dependency);
                context.logger.info(`✅️ Added '${dependency.name}' into ${dependency.type}`);
            }
        }
        return tree;
    };
}
exports.addPackageJsonDependencies = addPackageJsonDependencies;
/**
 * Adds libraries dependencies to package.json
 */
function addPackageJsonDependenciesForLibrary(dependencies, _options) {
    return (tree, _context) => {
        const packageJson = (0, package_utils_1.readPackageJson)(tree);
        const spartacusLibraries = (0, package_utils_1.createSpartacusDependencies)(dependencies);
        const thirdPartyLibraries = (0, package_utils_1.createDependencies)(dependencies);
        const libraries = spartacusLibraries.concat(thirdPartyLibraries);
        return (0, schematics_1.chain)([
            addPackageJsonDependencies(libraries, packageJson),
            installPackageJsonDependencies(),
        ]);
    };
}
exports.addPackageJsonDependenciesForLibrary = addPackageJsonDependenciesForLibrary;
function dependencyExists(dependency, packageJson) {
    var _a;
    return (_a = packageJson[dependency.type]) === null || _a === void 0 ? void 0 : _a.hasOwnProperty(dependency.name);
}
exports.dependencyExists = dependencyExists;
function configureB2bFeatures(options, packageJson) {
    return (_tree, _context) => {
        const spartacusVersion = (0, package_utils_1.getPrefixedSpartacusSchematicsVersion)();
        return (0, schematics_1.chain)([
            addB2bProviders(options),
            addPackageJsonDependencies([
                {
                    type: dependencies_1.NodeDependencyType.Default,
                    version: spartacusVersion,
                    name: libs_constants_1.SPARTACUS_SETUP,
                },
            ], packageJson),
        ]);
    };
}
exports.configureB2bFeatures = configureB2bFeatures;
function addB2bProviders(options) {
    return (tree, _context) => {
        const { buildPaths } = (0, project_tsconfig_paths_1.getProjectTsConfigPaths)(tree, options.project);
        if (!buildPaths.length) {
            throw new schematics_1.SchematicsException('Could not find any tsconfig file. Cannot configure SpartacusConfigurationModule.');
        }
        const basePath = process.cwd();
        for (const tsconfigPath of buildPaths) {
            const { appSourceFiles } = (0, program_1.createProgram)(tree, basePath, tsconfigPath);
            for (const sourceFile of appSourceFiles) {
                if (!sourceFile
                    .getFilePath()
                    .includes(`${libs_constants_1.SPARTACUS_CONFIGURATION_MODULE}.module.ts`)) {
                    continue;
                }
                (0, config_utils_1.getB2bConfiguration)().forEach((provider) => (0, new_module_utils_1.addModuleProvider)(sourceFile, provider));
                (0, program_1.saveAndFormat)(sourceFile);
                break;
            }
        }
        return tree;
    };
}
function createModuleFileName(config) {
    return `${(0, strings_1.dasherize)(config.moduleName)}-feature.module.ts`;
}
/**
 * Used a comparator function when sorting features.
 */
function calculateCrossFeatureSort(featureA, featureB) {
    return calculateSortInternal(featureA, featureB, graph_utils_1.crossFeatureInstallationOrder);
}
exports.calculateCrossFeatureSort = calculateCrossFeatureSort;
/**
 * Used a comparator function when sorting libraries.
 */
function calculateCrossLibrarySort(libraryA, libraryB) {
    return calculateSortInternal(libraryA, libraryB, graph_utils_1.crossLibraryInstallationOrder);
}
exports.calculateCrossLibrarySort = calculateCrossLibrarySort;
/**
 * Used to sort libraries or features in the correct order.
 */
function calculateSortInternal(libOrFeatureA, libOrFeatureB, order) {
    const indexA = order.indexOf(libOrFeatureA);
    const indexB = order.indexOf(libOrFeatureB);
    /**
     * In case a feature module is _not_ found in the `order`,
     * we want to sort it at the end of the list.
     */
    return (indexA > -1 ? indexA : Infinity) - (indexB > -1 ? indexB : Infinity);
}
/**
 * Performs the final steps of the installation,
 * before Angular schematics mechanism takes over.
 */
function finalizeInstallation(options, features) {
    return (_tree, context) => {
        var _a;
        if ((_a = options.internal) === null || _a === void 0 ? void 0 : _a.existingSpartacusApplication) {
            let message = `🚨 Detected Spartacus installation. Please make sure the following `;
            message += `features are installed, configured and sorted in the correct order:\n`;
            message += features.join(', ');
            context.logger.warn(message);
        }
    };
}
exports.finalizeInstallation = finalizeInstallation;
