// @spartacus/schematics/src/shared/utils/package-utils.js
"use strict";
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.updatePackageJsonDependencies = exports.prepare3rdPartyDependencies = exports.prepareSpartacusDependencies = exports.checkIfSSRIsUsed = exports.getSpartacusCurrentFeatureLevel = exports.getPrefixedSpartacusSchematicsVersion = exports.getSpartacusSchematicsVersion = exports.getMajorVersionNumber = exports.cleanSemverVersion = exports.readPackageJson = exports.mapPackageToNodeDependencies = exports.createDependencies = exports.createSpartacusDependencies = void 0;
const schematics_1 = require("@angular-devkit/schematics");
const dependencies_1 = require("@schematics/angular/utility/dependencies");
const semver_1 = __importDefault(require("semver"));
const package_json_1 = require("../../../package.json");
const dependencies_json_1 = __importDefault(require("../../dependencies.json"));
const constants_1 = require("../constants");
const libs_constants_1 = require("../libs-constants");
const file_utils_1 = require("./file-utils");
const lib_utils_1 = require("./lib-utils");
const workspace_utils_1 = require("./workspace-utils");
const DEV_DEPENDENCIES_KEYWORDS = [
    'schematics',
    'parse5',
    'typescript',
    '@angular-devkit/core',
    '@angular/compiler',
    'jsonc-parser',
];
function createSpartacusDependencies(dependencyObject) {
    const spartacusVersion = getPrefixedSpartacusSchematicsVersion();
    return createDependencies(dependencyObject, {
        skipScopes: libs_constants_1.CORE_SPARTACUS_SCOPES,
        onlyIncludeScopes: libs_constants_1.FEATURES_LIBS_SKIP_SCOPES,
        version: spartacusVersion,
    });
}
exports.createSpartacusDependencies = createSpartacusDependencies;
function createDependencies(dependencyObject, options = {
    skipScopes: libs_constants_1.FEATURES_LIBS_SKIP_SCOPES,
}) {
    var _a;
    const dependencies = [];
    for (const dependencyName in dependencyObject) {
        if (!dependencyObject.hasOwnProperty(dependencyName)) {
            continue;
        }
        if (options.skipScopes.some((scope) => dependencyName.startsWith(scope))) {
            continue;
        }
        if (
        // if `onlyIncludeScopes` is not defined, always include the dependency
        !options.onlyIncludeScopes ||
            // if defined, check if the current dependency is in the given array
            options.onlyIncludeScopes.some((scope) => dependencyName.startsWith(scope))) {
            dependencies.push(mapPackageToNodeDependencies(dependencyName, (_a = options.version) !== null && _a !== void 0 ? _a : dependencyObject[dependencyName], options.overwrite));
        }
    }
    return dependencies;
}
exports.createDependencies = createDependencies;
function mapPackageToNodeDependencies(packageName, pkgVersion, overwrite = false) {
    const type = DEV_DEPENDENCIES_KEYWORDS.some((keyword) => packageName.includes(keyword))
        ? dependencies_1.NodeDependencyType.Dev
        : dependencies_1.NodeDependencyType.Default;
    return {
        type,
        overwrite,
        name: packageName,
        version: pkgVersion,
    };
}
exports.mapPackageToNodeDependencies = mapPackageToNodeDependencies;
function readPackageJson(tree) {
    const pkgPath = '/package.json';
    const buffer = tree.read(pkgPath);
    if (!buffer) {
        throw new schematics_1.SchematicsException('Could not find package.json');
    }
    return JSON.parse(buffer.toString(constants_1.UTF_8));
}
exports.readPackageJson = readPackageJson;
function cleanSemverVersion(versionString) {
    if (isNaN(Number(versionString.charAt(0)))) {
        return versionString.substring(1, versionString.length);
    }
    return versionString;
}
exports.cleanSemverVersion = cleanSemverVersion;
function getMajorVersionNumber(versionString) {
    const cleanVersion = cleanSemverVersion(versionString);
    return Number(cleanVersion.charAt(0));
}
exports.getMajorVersionNumber = getMajorVersionNumber;
function getSpartacusSchematicsVersion() {
    return package_json_1.version;
}
exports.getSpartacusSchematicsVersion = getSpartacusSchematicsVersion;
function getPrefixedSpartacusSchematicsVersion() {
    return `~${getSpartacusSchematicsVersion()}`;
}
exports.getPrefixedSpartacusSchematicsVersion = getPrefixedSpartacusSchematicsVersion;
function getSpartacusCurrentFeatureLevel() {
    return package_json_1.version.split('.').slice(0, 2).join('.');
}
exports.getSpartacusCurrentFeatureLevel = getSpartacusCurrentFeatureLevel;
function checkIfSSRIsUsed(tree) {
    const projectName = (0, workspace_utils_1.getDefaultProjectNameFromWorkspace)(tree);
    const buffer = tree.read('angular.json');
    if (!buffer) {
        throw new schematics_1.SchematicsException('Could not find angular.json');
    }
    const angularFileBuffer = buffer.toString(constants_1.UTF_8);
    const angularJson = JSON.parse(angularFileBuffer);
    const isServerConfiguration = !!angularJson.projects[projectName].architect['server'];
    const serverFileLocation = (0, file_utils_1.getServerTsPath)(tree);
    if (!serverFileLocation) {
        return false;
    }
    const serverBuffer = tree.read(serverFileLocation);
    const serverFileBuffer = serverBuffer === null || serverBuffer === void 0 ? void 0 : serverBuffer.toString(constants_1.UTF_8);
    const isServerSideAvailable = serverFileBuffer && !!serverFileBuffer.length;
    return !!(isServerConfiguration && isServerSideAvailable);
}
exports.checkIfSSRIsUsed = checkIfSSRIsUsed;
function prepareSpartacusDependencies() {
    const spartacusVersion = getPrefixedSpartacusSchematicsVersion();
    const spartacusDependencies = [
        {
            type: dependencies_1.NodeDependencyType.Default,
            version: spartacusVersion,
            name: libs_constants_1.SPARTACUS_CORE,
        },
        {
            type: dependencies_1.NodeDependencyType.Default,
            version: spartacusVersion,
            name: libs_constants_1.SPARTACUS_STOREFRONTLIB,
        },
        {
            type: dependencies_1.NodeDependencyType.Default,
            version: spartacusVersion,
            name: libs_constants_1.SPARTACUS_ASSETS,
        },
        {
            type: dependencies_1.NodeDependencyType.Default,
            version: spartacusVersion,
            name: libs_constants_1.SPARTACUS_STYLES,
        },
    ];
    return spartacusDependencies;
}
exports.prepareSpartacusDependencies = prepareSpartacusDependencies;
function prepare3rdPartyDependencies() {
    const thirdPartyDependencies = createDependencies(Object.assign(Object.assign(Object.assign(Object.assign(Object.assign({}, dependencies_json_1.default[libs_constants_1.SPARTACUS_CORE]), dependencies_json_1.default[libs_constants_1.SPARTACUS_STOREFRONTLIB]), dependencies_json_1.default[libs_constants_1.SPARTACUS_STYLES]), dependencies_json_1.default[libs_constants_1.SPARTACUS_ASSETS]), dependencies_json_1.default[libs_constants_1.SPARTACUS_SCHEMATICS]));
    return thirdPartyDependencies;
}
exports.prepare3rdPartyDependencies = prepare3rdPartyDependencies;
function updatePackageJsonDependencies(dependencies, packageJson) {
    return (tree, context) => {
        const dependenciesToAdd = [];
        for (const dependency of dependencies) {
            const currentVersion = getCurrentDependencyVersion(dependency, packageJson);
            if (!currentVersion) {
                dependenciesToAdd.push(dependency);
                continue;
            }
            if (semver_1.default.satisfies(currentVersion, dependency.version)) {
                continue;
            }
            const versionToUpdate = semver_1.default.parse(cleanSemverVersion(dependency.version));
            if (!versionToUpdate || semver_1.default.eq(versionToUpdate, currentVersion)) {
                continue;
            }
            (0, dependencies_1.addPackageJsonDependency)(tree, Object.assign(Object.assign({}, dependency), { overwrite: true }));
            const change = semver_1.default.gt(versionToUpdate, currentVersion)
                ? 'Upgrading'
                : 'Downgrading';
            context.logger.info(`🩹 ${change} '${dependency.name}' to ${dependency.version} (was ${currentVersion.raw})`);
        }
        return (0, lib_utils_1.addPackageJsonDependencies)(dependenciesToAdd, packageJson);
    };
}
exports.updatePackageJsonDependencies = updatePackageJsonDependencies;
function getCurrentDependencyVersion(dependency, packageJson) {
    if (!(0, lib_utils_1.dependencyExists)(dependency, packageJson)) {
        return null;
    }
    const dependencies = packageJson[dependency.type];
    const currentVersion = dependencies[dependency.name];
    return semver_1.default.parse(cleanSemverVersion(currentVersion));
}
