// @spartacus/schematics/src/shared/utils/test-utils.js
"use strict";
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.getParams = exports.getSuperNode = exports.getConstructor = exports.runMigration = exports.writeFile = exports.segmentRefsFeatureModulePath = exports.epdFeatureModulePath = exports.digitalPaymentsFeatureModulePath = exports.cdsFeatureModulePath = exports.cdpFeatureModulePath = exports.cdcFeatureModulePath = exports.userProfileWrapperModulePath = exports.userAccountWrapperModulePath = exports.userFeatureModulePath = exports.trackingTagManagementFeatureModulePath = exports.trackingPersonalizationFeatureModulePath = exports.storeFinderFeatureModulePath = exports.smartEditFeatureModulePath = exports.requestedDeliveryDateFeatureModulePath = exports.qualtricsFeatureModulePath = exports.productConfiguratorRulebasedWrapperModulePath = exports.productConfiguratorFeatureModulePath = exports.productFutureStockFeatureModulePath = exports.productVariantsFeatureModulePath = exports.productImageZoomFeatureModulePath = exports.productBulkPricingFeatureModulePath = exports.pdfInvoicesFeatureModulePath = exports.organizationAccountSummaryFeatureModulePath = exports.organizationUnitOrderFeatureModulePath = exports.organizationUserRegistrationWrapperModulePath = exports.organizationUserRegistrationFeatureModulePath = exports.organizationOrderApprovalFeatureModulePath = exports.organizationAdministrationWrapperModulePath = exports.pickupInStoreFeatureModulePath = exports.organizationAdministrationFeatureModulePath = exports.orderFeatureModulePath = exports.quoteFeatureModulePath = exports.checkoutWrapperModulePath = exports.checkoutFeatureModulePath = exports.savedCartFeatureModulePath = exports.wishListFeatureModulePath = exports.quickOrderFeatureModulePath = exports.importExportFeatureModulePath = exports.customerTicketingWrapperModulePath = exports.customerTicketingFeatureModulePath = exports.cartWrapperModulePath = exports.cartBaseFeatureModulePath = exports.asmFeatureModulePath = exports.spartacusFeaturesModulePath = void 0;
const core_1 = require("@angular-devkit/core");
const ast_utils_1 = require("@schematics/angular/utility/ast-utils");
const typescript_1 = __importDefault(require("typescript"));
const file_utils_1 = require("./file-utils");
exports.spartacusFeaturesModulePath = 'src/app/spartacus/spartacus-features.module.ts';
exports.asmFeatureModulePath = 'src/app/spartacus/features/asm/asm-feature.module.ts';
exports.cartBaseFeatureModulePath = 'src/app/spartacus/features/cart/cart-base-feature.module.ts';
exports.cartWrapperModulePath = 'src/app/spartacus/features/cart/cart-base-wrapper.module.ts';
exports.customerTicketingFeatureModulePath = 'src/app/spartacus/features/customer-ticketing/customer-ticketing-feature.module.ts';
exports.customerTicketingWrapperModulePath = 'src/app/spartacus/features/customer-ticketing/customer-ticketing-wrapper.module.ts';
exports.importExportFeatureModulePath = 'src/app/spartacus/features/cart/cart-import-export-feature.module.ts';
exports.quickOrderFeatureModulePath = 'src/app/spartacus/features/cart/cart-quick-order-feature.module.ts';
exports.wishListFeatureModulePath = 'src/app/spartacus/features/cart/wish-list-feature.module.ts';
exports.savedCartFeatureModulePath = 'src/app/spartacus/features/cart/cart-saved-cart-feature.module.ts';
exports.checkoutFeatureModulePath = 'src/app/spartacus/features/checkout/checkout-feature.module.ts';
exports.checkoutWrapperModulePath = 'src/app/spartacus/features/checkout/checkout-wrapper.module.ts';
exports.quoteFeatureModulePath = 'src/app/spartacus/features/quote/quote-feature.module.ts';
exports.orderFeatureModulePath = 'src/app/spartacus/features/order/order-feature.module.ts';
exports.organizationAdministrationFeatureModulePath = 'src/app/spartacus/features/organization/organization-administration-feature.module.ts';
exports.pickupInStoreFeatureModulePath = 'src/app/spartacus/features/pickup-in-store/pickup-in-store-feature.module.ts';
exports.organizationAdministrationWrapperModulePath = 'src/app/spartacus/features/organization/administration-wrapper.module.ts';
exports.organizationOrderApprovalFeatureModulePath = 'src/app/spartacus/features/organization/organization-order-approval-feature.module.ts';
exports.organizationUserRegistrationFeatureModulePath = 'src/app/spartacus/features/organization/organization-user-registration-feature.module.ts';
exports.organizationUserRegistrationWrapperModulePath = 'src/app/spartacus/features/organization/organization-user-registration-wrapper.module.ts';
exports.organizationUnitOrderFeatureModulePath = 'src/app/spartacus/features/organization/organization-unit-order-feature.module.ts';
exports.organizationAccountSummaryFeatureModulePath = 'src/app/spartacus/features/organization/organization-account-summary-feature.module.ts';
exports.pdfInvoicesFeatureModulePath = 'src/app/spartacus/features/pdf-invoices/pdf-invoices-feature.module.ts';
exports.productBulkPricingFeatureModulePath = 'src/app/spartacus/features/product/product-bulk-pricing-feature.module.ts';
exports.productImageZoomFeatureModulePath = 'src/app/spartacus/features/product/product-image-zoom-feature.module.ts';
exports.productVariantsFeatureModulePath = 'src/app/spartacus/features/product/product-variants-feature.module.ts';
exports.productFutureStockFeatureModulePath = 'src/app/spartacus/features/product/product-future-stock-feature.module.ts';
exports.productConfiguratorFeatureModulePath = 'src/app/spartacus/features/product-configurator/product-configurator-feature.module.ts';
exports.productConfiguratorRulebasedWrapperModulePath = 'src/app/spartacus/features/product-configurator/rulebased-configurator-wrapper.module.ts';
exports.qualtricsFeatureModulePath = 'src/app/spartacus/features/qualtrics/qualtrics-feature.module.ts';
exports.requestedDeliveryDateFeatureModulePath = 'src/app/spartacus/features/requested-delivery-date/requested-delivery-date-feature.module.ts';
exports.smartEditFeatureModulePath = 'src/app/spartacus/features/smartedit/smart-edit-feature.module.ts';
exports.storeFinderFeatureModulePath = 'src/app/spartacus/features/storefinder/store-finder-feature.module.ts';
exports.trackingPersonalizationFeatureModulePath = 'src/app/spartacus/features/tracking/personalization-feature.module.ts';
exports.trackingTagManagementFeatureModulePath = 'src/app/spartacus/features/tracking/tag-management-feature.module.ts';
exports.userFeatureModulePath = 'src/app/spartacus/features/user/user-feature.module.ts';
exports.userAccountWrapperModulePath = 'src/app/spartacus/features/user/user-account-wrapper.module.ts';
exports.userProfileWrapperModulePath = 'src/app/spartacus/features/user/user-profile-wrapper.module.ts';
exports.cdcFeatureModulePath = 'src/app/spartacus/features/cdc/cdc-feature.module.ts';
exports.cdpFeatureModulePath = 'src/app/spartacus/features/cdp/cdp-feature.module.ts';
exports.cdsFeatureModulePath = 'src/app/spartacus/features/cds/cds-feature.module.ts';
exports.digitalPaymentsFeatureModulePath = 'src/app/spartacus/features/digital-payments/digital-payments-feature.module.ts';
exports.epdFeatureModulePath = 'src/app/spartacus/features/epd-visualization/epd-visualization-feature.module.ts';
exports.segmentRefsFeatureModulePath = 'src/app/spartacus/features/segment-refs/segment-refs-feature.module.ts';
function writeFile(host, filePath, contents) {
    host.sync.write((0, core_1.normalize)(filePath), core_1.virtualFs.stringToFileBuffer(contents));
}
exports.writeFile = writeFile;
function runMigration(appTree, schematicRunner, migrationScript, options = {}) {
    return schematicRunner.runSchematic(migrationScript, options, appTree);
}
exports.runMigration = runMigration;
function getConstructor(nodes) {
    const constructorNode = (0, file_utils_1.findConstructor)(nodes);
    if (!constructorNode) {
        throw new Error('No constructor node found');
    }
    return constructorNode;
}
exports.getConstructor = getConstructor;
function getSuperNode(constructorNode) {
    const superNodes = (0, ast_utils_1.findNodes)(constructorNode, typescript_1.default.SyntaxKind.SuperKeyword);
    if (!superNodes || superNodes.length === 0) {
        return undefined;
    }
    return superNodes[0];
}
exports.getSuperNode = getSuperNode;
function getParams(constructorNode, camelizedParamNames) {
    const superNode = getSuperNode(constructorNode);
    if (!superNode) {
        throw new Error('No super() node found');
    }
    const callExpressions = (0, ast_utils_1.findNodes)(constructorNode, typescript_1.default.SyntaxKind.CallExpression);
    if (!callExpressions || callExpressions.length === 0) {
        throw new Error('No call expressions found in constructor');
    }
    const params = (0, ast_utils_1.findNodes)(callExpressions[0], typescript_1.default.SyntaxKind.Identifier);
    camelizedParamNames = camelizedParamNames.map((param) => core_1.strings.camelize(param));
    return params
        .filter((n) => n.kind === typescript_1.default.SyntaxKind.Identifier)
        .map((n) => n.getText())
        .filter((text) => camelizedParamNames.includes(text));
}
exports.getParams = getParams;
