// @spartacus/core/esm2022/src/global-message/http-interceptors/http-error.interceptor.mjs
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { HttpErrorResponse, } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { shareReplay, tap } from 'rxjs/operators';
import { resolveApplicable } from '../../util/applicable';
import { getLastValueSync } from '../../util/rxjs/get-last-value-sync';
import { HttpErrorHandler } from './handlers/http-error.handler';
import * as i0 from "@angular/core";
import * as i1 from "../../lazy-loading/unified-injector";
export class HttpErrorInterceptor {
    constructor(unifiedInjector) {
        this.unifiedInjector = unifiedInjector;
        this.handlers$ = this.unifiedInjector
            .getMulti(HttpErrorHandler)
            .pipe(shareReplay(1));
    }
    intercept(request, next) {
        return next.handle(request).pipe(tap({
            error: (response) => {
                if (response instanceof HttpErrorResponse) {
                    this.handleErrorResponse(request, response);
                }
            },
        }));
    }
    handleErrorResponse(request, response) {
        const handler = this.getResponseHandler(response);
        if (handler) {
            handler.handleError(request, response);
        }
    }
    /**
     * return the error handler that matches the `HttpResponseStatus` code.
     * If no handler is available, the UNKNOWN handler is returned.
     */
    getResponseHandler(response) {
        return resolveApplicable(getLastValueSync(this.handlers$) ?? [], [
            response,
        ]);
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: HttpErrorInterceptor, deps: [{ token: i1.UnifiedInjector }], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: HttpErrorInterceptor, providedIn: 'root' }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: HttpErrorInterceptor, decorators: [{
            type: Injectable,
            args: [{ providedIn: 'root' }]
        }], ctorParameters: () => [{ type: i1.UnifiedInjector }] });
