// @spartacus/storefront/esm2022/cms-components/navigation/search-box/search-box.component.mjs
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { ChangeDetectionStrategy, Component, Input, Optional, } from '@angular/core';
import { PageType, } from '@spartacus/core';
import { of, Subscription } from 'rxjs';
import { filter, map, switchMap, tap } from 'rxjs/operators';
import { ICON_TYPE } from '../../../cms-components/misc/icon/index';
import { SearchBoxOutlets } from './search-box-outlets.model';
import { SearchBoxFeatures } from './search-box-features.model';
import * as i0 from "@angular/core";
import * as i1 from "./search-box-component.service";
import * as i2 from "../../../cms-structure/page/model/cms-component-data";
import * as i3 from "@spartacus/core";
import * as i4 from "@angular/common";
import * as i5 from "@angular/router";
import * as i6 from "../../../shared/components/media/media.component";
import * as i7 from "../../misc/icon/icon.component";
import * as i8 from "../../../cms-structure/outlet/outlet.directive";
import * as i9 from "./highlight.pipe";
const DEFAULT_SEARCH_BOX_CONFIG = {
    minCharactersBeforeRequest: 1,
    displayProducts: true,
    displaySuggestions: true,
    maxProducts: 5,
    maxSuggestions: 5,
    displayProductImages: true,
    recentSearches: true,
    maxRecentSearches: 5,
};
const SEARCHBOX_IS_ACTIVE = 'searchbox-is-active';
export class SearchBoxComponent {
    /**
     * Sets the search box input field
     */
    set queryText(value) {
        if (value) {
            this.search(value);
        }
    }
    constructor(searchBoxComponentService, componentData, winRef, routingService) {
        this.searchBoxComponentService = searchBoxComponentService;
        this.componentData = componentData;
        this.winRef = winRef;
        this.routingService = routingService;
        this.searchBoxOutlets = SearchBoxOutlets;
        this.searchBoxFeatures = SearchBoxFeatures;
        this.iconTypes = ICON_TYPE;
        this.searchBoxActive = false;
        /**
         * In some occasions we need to ignore the close event,
         * for example when we click inside the search result section.
         */
        this.ignoreCloseEvent = false;
        this.chosenWord = '';
        this.subscriptions = new Subscription();
        /**
         * Returns the SearchBox configuration. The configuration is driven by multiple
         * layers: default configuration, (optional) backend configuration and (optional)
         * input configuration.
         */
        this.config$ = (this.componentData?.data$ || of({})).pipe(map((config) => {
            const isBool = (obj, prop) => obj[prop] !== 'false' &&
                obj[prop] !== false;
            return {
                ...DEFAULT_SEARCH_BOX_CONFIG,
                ...config,
                displayProducts: isBool(config, 'displayProducts'),
                displayProductImages: isBool(config, 'displayProductImages'),
                displaySuggestions: isBool(config, 'displaySuggestions'),
                // we're merging the (optional) input of this component, but write the merged
                // result back to the input property, as the view logic depends on it.
                ...this.config,
            };
        }), tap((config) => (this.config = config)));
        this.results$ = this.config$.pipe(switchMap((config) => this.searchBoxComponentService.getResults(config)));
    }
    ngOnInit() {
        const routeStateSubscription = this.routingService
            .getRouterState()
            .pipe(filter((data) => !data.nextState))
            .subscribe((data) => {
            if (!(data.state.context?.id === 'search' &&
                data.state.context?.type === PageType.CONTENT_PAGE)) {
                this.chosenWord = '';
            }
        });
        this.subscriptions.add(routeStateSubscription);
        const chosenWordSubscription = this.searchBoxComponentService.chosenWord.subscribe((chosenWord) => {
            this.updateChosenWord(chosenWord);
        });
        this.subscriptions.add(chosenWordSubscription);
        const UIEventSubscription = this.searchBoxComponentService.sharedEvent.subscribe((event) => {
            this.propagateEvent(event);
        });
        this.subscriptions.add(UIEventSubscription);
    }
    /**
     * Closes the searchBox and opens the search result page.
     */
    search(query) {
        this.searchBoxComponentService.search(query, this.config);
        // force the searchBox to open
        this.open();
    }
    /**
     * Opens the type-ahead searchBox
     */
    open() {
        this.searchBoxComponentService.toggleBodyClass(SEARCHBOX_IS_ACTIVE, true);
        this.searchBoxActive = true;
    }
    /**
     * Dispatch UI events for Suggestion selected
     *
     * @param eventData the data for the event
     */
    dispatchSuggestionEvent(eventData) {
        this.searchBoxComponentService.dispatchSuggestionSelectedEvent(eventData);
    }
    /**
     * Dispatch UI events for Product selected
     *
     * @param eventData the data for the event
     */
    dispatchProductEvent(eventData) {
        this.searchBoxComponentService.dispatchProductSelectedEvent(eventData);
    }
    /**
     * Closes the type-ahead searchBox.
     */
    close(event, force) {
        // Use timeout to detect changes
        setTimeout(() => {
            if ((!this.ignoreCloseEvent && !this.isSearchBoxFocused()) || force) {
                this.blurSearchBox(event);
            }
        });
    }
    blurSearchBox(event) {
        this.searchBoxComponentService.toggleBodyClass(SEARCHBOX_IS_ACTIVE, false);
        this.searchBoxActive = false;
        if (event && event.target) {
            event.target.blur();
        }
    }
    // Check if focus is on searchbox or result list elements
    isSearchBoxFocused() {
        return (this.getResultElements().includes(this.getFocusedElement()) ||
            this.winRef.document.querySelector('input[aria-label="Search"]') ===
                this.getFocusedElement());
    }
    /**
     * Especially in mobile we do not want the search icon
     * to focus the input again when it's already open.
     * */
    avoidReopen(event) {
        if (this.searchBoxComponentService.hasBodyClass(SEARCHBOX_IS_ACTIVE)) {
            this.close(event);
            event.preventDefault();
        }
    }
    // Return result list as HTMLElement array
    getResultElements() {
        return Array.from(this.winRef.document.querySelectorAll('.products > li a, .suggestions > li a, .recent-searches > li a'));
    }
    // Return focused element as HTMLElement
    getFocusedElement() {
        return this.winRef.document.activeElement;
    }
    updateChosenWord(chosenWord) {
        this.chosenWord = chosenWord;
    }
    getFocusedIndex() {
        return this.getResultElements().indexOf(this.getFocusedElement());
    }
    propagateEvent(event) {
        if (event.code) {
            switch (event.code) {
                case 'Escape':
                case 'Enter':
                    this.close(event, true);
                    return;
                case 'ArrowUp':
                    this.focusPreviousChild(event);
                    return;
                case 'ArrowDown':
                    this.focusNextChild(event);
                    return;
                default:
                    return;
            }
        }
        else if (event.type === 'blur') {
            this.close(event);
        }
    }
    // Focus on previous item in results list
    focusPreviousChild(event) {
        event.preventDefault(); // Negate normal keyscroll
        const [results, focusedIndex] = [
            this.getResultElements(),
            this.getFocusedIndex(),
        ];
        // Focus on last index moving to first
        if (results.length) {
            if (focusedIndex < 1) {
                results[results.length - 1].focus();
            }
            else {
                results[focusedIndex - 1].focus();
            }
        }
    }
    // Focus on next item in results list
    focusNextChild(event) {
        this.open();
        event.preventDefault(); // Negate normal keyscroll
        const [results, focusedIndex] = [
            this.getResultElements(),
            this.getFocusedIndex(),
        ];
        // Focus on first index moving to last
        if (results.length) {
            if (focusedIndex >= results.length - 1) {
                results[0].focus();
            }
            else {
                results[focusedIndex + 1].focus();
            }
        }
    }
    /**
     * Opens the PLP with the given query.
     *
     * TODO: if there's a single product match, we could open the PDP.
     */
    launchSearchResult(event, query) {
        if (!query || query.trim().length === 0) {
            return;
        }
        this.close(event);
        this.searchBoxComponentService.launchSearchPage(query);
    }
    /**
     * Disables closing the search result list.
     */
    disableClose() {
        this.ignoreCloseEvent = true;
    }
    preventDefault(ev) {
        ev.preventDefault();
    }
    /**
     * Clears the search box input field
     */
    clear(el) {
        this.disableClose();
        el.value = '';
        this.searchBoxComponentService.clearResults();
        // Use Timeout to run after blur event to prevent the searchbox from closing on mobile
        setTimeout(() => {
            // Retain focus on input lost by clicking on icon
            el.focus();
            this.ignoreCloseEvent = false;
        });
    }
    ngOnDestroy() {
        this.subscriptions?.unsubscribe();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: SearchBoxComponent, deps: [{ token: i1.SearchBoxComponentService }, { token: i2.CmsComponentData, optional: true }, { token: i3.WindowRef }, { token: i3.RoutingService }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "17.0.5", type: SearchBoxComponent, selector: "cx-searchbox", inputs: { config: "config", queryText: "queryText" }, ngImport: i0, template: "<div [attr.aria-label]=\"'searchBox.productSearch' | cxTranslate\" role=\"search\">\n  <label class=\"searchbox\" [class.dirty]=\"!!searchInput.value\">\n    <input\n      #searchInput\n      [placeholder]=\"'searchBox.placeholder' | cxTranslate\"\n      autocomplete=\"off\"\n      aria-describedby=\"initialDescription\"\n      aria-controls=\"results\"\n      [attr.aria-label]=\"'searchBox.placeholder' | cxTranslate\"\n      (focus)=\"open()\"\n      (click)=\"open()\"\n      (input)=\"search(searchInput.value)\"\n      (blur)=\"close($any($event))\"\n      (keydown.escape)=\"close($any($event))\"\n      (keydown.enter)=\"\n        close($any($event), true);\n        launchSearchResult($any($event), searchInput.value);\n        updateChosenWord(searchInput.value)\n      \"\n      (keydown.arrowup)=\"focusPreviousChild($any($event))\"\n      (keydown.arrowdown)=\"focusNextChild($any($event))\"\n      value=\"{{ chosenWord }}\"\n    />\n\n    <button\n      [attr.aria-label]=\"'common.reset' | cxTranslate\"\n      (mousedown)=\"clear(searchInput)\"\n      (keydown.enter)=\"clear(searchInput)\"\n      class=\"reset\"\n    >\n      <cx-icon [type]=\"iconTypes.RESET\"></cx-icon>\n    </button>\n\n    <div role=\"presentation\" class=\"search-icon\">\n      <cx-icon [type]=\"iconTypes.SEARCH\"></cx-icon>\n    </div>\n\n    <button\n      [attr.aria-label]=\"'common.search' | cxTranslate\"\n      class=\"search\"\n      (click)=\"open()\"\n    >\n      <cx-icon [type]=\"iconTypes.SEARCH\"></cx-icon>\n    </button>\n  </label>\n</div>\n\n<div\n  *ngIf=\"results$ | async as result\"\n  class=\"results\"\n  id=\"results\"\n  (click)=\"close($any($event), true)\"\n  role=\"dialog\"\n>\n  <div\n    *ngIf=\"result.message\"\n    class=\"message\"\n    [innerHTML]=\"result.message\"\n  ></div>\n  <!--RESULT  SUGGESTIONS-->\n  <ng-container *cxFeature=\"searchBoxFeatures.RECENT_SEARCHES_FEATURE\">\n    <div class=\"message\" *ngIf=\"result?.suggestions?.length\">\n      {{ 'searchBox.suggestions' | cxTranslate }}\n    </div>\n  </ng-container>\n  <ul\n    class=\"suggestions\"\n    attr.aria-label=\"{{ 'searchBox.ariaLabelSuggestions' | cxTranslate }}\"\n    role=\"listbox\"\n  >\n    <li *ngFor=\"let suggestion of result.suggestions\">\n      <a\n        role=\"option\"\n        [innerHTML]=\"suggestion | cxHighlight: searchInput.value\"\n        [routerLink]=\"\n          {\n            cxRoute: 'search',\n            params: { query: suggestion }\n          } | cxUrl\n        \"\n        (keydown.arrowup)=\"focusPreviousChild($any($event))\"\n        (keydown.arrowdown)=\"focusNextChild($any($event))\"\n        (keydown.enter)=\"close($any($event), true)\"\n        (keydown.escape)=\"close($any($event), true)\"\n        (blur)=\"close($event)\"\n        (mousedown)=\"preventDefault($event)\"\n        (click)=\"\n          dispatchSuggestionEvent({\n            freeText: searchInput.value,\n            selectedSuggestion: suggestion,\n            searchSuggestions: result.suggestions ?? []\n          });\n          updateChosenWord(suggestion)\n        \"\n      >\n      </a>\n    </li>\n  </ul>\n\n  <!--  RECENT SEARCHES-->\n  <ng-container *cxFeature=\"searchBoxFeatures.RECENT_SEARCHES_FEATURE\">\n    <ng-template\n      *ngIf=\"config.recentSearches\"\n      [cxOutlet]=\"searchBoxOutlets.RECENT_SEARCHES\"\n      [cxOutletContext]=\"{\n        search: searchInput.value,\n        searchBoxActive: searchBoxActive,\n        maxRecentSearches: config.maxRecentSearches\n      }\"\n    ></ng-template>\n  </ng-container>\n\n  <!--RESULT  PRODUCTS-->\n  <ng-container *cxFeature=\"searchBoxFeatures.RECENT_SEARCHES_FEATURE\">\n    <div class=\"message\" *ngIf=\"result.products?.length\">\n      {{ 'searchBox.products' | cxTranslate }}\n    </div>\n  </ng-container>\n  <ul\n    class=\"products\"\n    *ngIf=\"result.products\"\n    attr.aria-label=\"{{ 'searchBox.ariaLabelProducts' | cxTranslate }}\"\n    role=\"listbox\"\n  >\n    <li *ngFor=\"let product of result.products\">\n      <a\n        role=\"option\"\n        [routerLink]=\"\n          {\n            cxRoute: 'product',\n            params: product\n          } | cxUrl\n        \"\n        [class.has-media]=\"config.displayProductImages\"\n        (keydown.arrowup)=\"focusPreviousChild($any($event))\"\n        (keydown.arrowdown)=\"focusNextChild($any($event))\"\n        (keydown.enter)=\"close($any($event), true)\"\n        (keydown.escape)=\"close($any($event), true)\"\n        (blur)=\"close($any($event))\"\n        (mousedown)=\"preventDefault($event)\"\n        (click)=\"\n          dispatchProductEvent({\n            freeText: searchInput.value,\n            productCode: product.code\n          })\n        \"\n      >\n        <cx-media\n          *ngIf=\"config.displayProductImages\"\n          [container]=\"product.images?.PRIMARY\"\n          format=\"thumbnail\"\n          role=\"presentation\"\n        ></cx-media>\n        <div class=\"name\" [innerHTML]=\"product.nameHtml\"></div>\n        <span class=\"price\">{{ product.price?.formattedValue }}</span>\n      </a>\n    </li>\n  </ul>\n  <span id=\"initialDescription\" class=\"cx-visually-hidden\">\n    {{ 'searchBox.initialDescription' | cxTranslate }}\n  </span>\n  <div\n    *ngIf=\"result.suggestions?.length || result.products?.length\"\n    aria-live=\"assertive\"\n    class=\"cx-visually-hidden\"\n  >\n    {{\n      'searchBox.suggestionsResult'\n        | cxTranslate: { count: result.suggestions?.length }\n    }}\n    {{\n      'searchBox.productsResult'\n        | cxTranslate: { count: result.products?.length }\n    }}\n    {{ 'searchBox.initialDescription' | cxTranslate }}\n  </div>\n</div>\n", dependencies: [{ kind: "directive", type: i4.NgForOf, selector: "[ngFor][ngForOf]", inputs: ["ngForOf", "ngForTrackBy", "ngForTemplate"] }, { kind: "directive", type: i4.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { kind: "directive", type: i5.RouterLink, selector: "[routerLink]", inputs: ["target", "queryParams", "fragment", "queryParamsHandling", "state", "relativeTo", "preserveFragment", "skipLocationChange", "replaceUrl", "routerLink"] }, { kind: "component", type: i6.MediaComponent, selector: "cx-media", inputs: ["container", "format", "alt", "role", "loading"], outputs: ["loaded"] }, { kind: "component", type: i7.IconComponent, selector: "cx-icon,[cxIcon]", inputs: ["cxIcon", "type"] }, { kind: "directive", type: i8.OutletDirective, selector: "[cxOutlet]", inputs: ["cxOutlet", "cxOutletContext", "cxOutletDefer", "cxComponentRef"], outputs: ["loaded", "cxComponentRefChange"] }, { kind: "directive", type: i3.FeatureDirective, selector: "[cxFeature]", inputs: ["cxFeature"] }, { kind: "pipe", type: i4.AsyncPipe, name: "async" }, { kind: "pipe", type: i3.UrlPipe, name: "cxUrl" }, { kind: "pipe", type: i3.TranslatePipe, name: "cxTranslate" }, { kind: "pipe", type: i9.HighlightPipe, name: "cxHighlight" }], changeDetection: i0.ChangeDetectionStrategy.OnPush }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: SearchBoxComponent, decorators: [{
            type: Component,
            args: [{ selector: 'cx-searchbox', changeDetection: ChangeDetectionStrategy.OnPush, template: "<div [attr.aria-label]=\"'searchBox.productSearch' | cxTranslate\" role=\"search\">\n  <label class=\"searchbox\" [class.dirty]=\"!!searchInput.value\">\n    <input\n      #searchInput\n      [placeholder]=\"'searchBox.placeholder' | cxTranslate\"\n      autocomplete=\"off\"\n      aria-describedby=\"initialDescription\"\n      aria-controls=\"results\"\n      [attr.aria-label]=\"'searchBox.placeholder' | cxTranslate\"\n      (focus)=\"open()\"\n      (click)=\"open()\"\n      (input)=\"search(searchInput.value)\"\n      (blur)=\"close($any($event))\"\n      (keydown.escape)=\"close($any($event))\"\n      (keydown.enter)=\"\n        close($any($event), true);\n        launchSearchResult($any($event), searchInput.value);\n        updateChosenWord(searchInput.value)\n      \"\n      (keydown.arrowup)=\"focusPreviousChild($any($event))\"\n      (keydown.arrowdown)=\"focusNextChild($any($event))\"\n      value=\"{{ chosenWord }}\"\n    />\n\n    <button\n      [attr.aria-label]=\"'common.reset' | cxTranslate\"\n      (mousedown)=\"clear(searchInput)\"\n      (keydown.enter)=\"clear(searchInput)\"\n      class=\"reset\"\n    >\n      <cx-icon [type]=\"iconTypes.RESET\"></cx-icon>\n    </button>\n\n    <div role=\"presentation\" class=\"search-icon\">\n      <cx-icon [type]=\"iconTypes.SEARCH\"></cx-icon>\n    </div>\n\n    <button\n      [attr.aria-label]=\"'common.search' | cxTranslate\"\n      class=\"search\"\n      (click)=\"open()\"\n    >\n      <cx-icon [type]=\"iconTypes.SEARCH\"></cx-icon>\n    </button>\n  </label>\n</div>\n\n<div\n  *ngIf=\"results$ | async as result\"\n  class=\"results\"\n  id=\"results\"\n  (click)=\"close($any($event), true)\"\n  role=\"dialog\"\n>\n  <div\n    *ngIf=\"result.message\"\n    class=\"message\"\n    [innerHTML]=\"result.message\"\n  ></div>\n  <!--RESULT  SUGGESTIONS-->\n  <ng-container *cxFeature=\"searchBoxFeatures.RECENT_SEARCHES_FEATURE\">\n    <div class=\"message\" *ngIf=\"result?.suggestions?.length\">\n      {{ 'searchBox.suggestions' | cxTranslate }}\n    </div>\n  </ng-container>\n  <ul\n    class=\"suggestions\"\n    attr.aria-label=\"{{ 'searchBox.ariaLabelSuggestions' | cxTranslate }}\"\n    role=\"listbox\"\n  >\n    <li *ngFor=\"let suggestion of result.suggestions\">\n      <a\n        role=\"option\"\n        [innerHTML]=\"suggestion | cxHighlight: searchInput.value\"\n        [routerLink]=\"\n          {\n            cxRoute: 'search',\n            params: { query: suggestion }\n          } | cxUrl\n        \"\n        (keydown.arrowup)=\"focusPreviousChild($any($event))\"\n        (keydown.arrowdown)=\"focusNextChild($any($event))\"\n        (keydown.enter)=\"close($any($event), true)\"\n        (keydown.escape)=\"close($any($event), true)\"\n        (blur)=\"close($event)\"\n        (mousedown)=\"preventDefault($event)\"\n        (click)=\"\n          dispatchSuggestionEvent({\n            freeText: searchInput.value,\n            selectedSuggestion: suggestion,\n            searchSuggestions: result.suggestions ?? []\n          });\n          updateChosenWord(suggestion)\n        \"\n      >\n      </a>\n    </li>\n  </ul>\n\n  <!--  RECENT SEARCHES-->\n  <ng-container *cxFeature=\"searchBoxFeatures.RECENT_SEARCHES_FEATURE\">\n    <ng-template\n      *ngIf=\"config.recentSearches\"\n      [cxOutlet]=\"searchBoxOutlets.RECENT_SEARCHES\"\n      [cxOutletContext]=\"{\n        search: searchInput.value,\n        searchBoxActive: searchBoxActive,\n        maxRecentSearches: config.maxRecentSearches\n      }\"\n    ></ng-template>\n  </ng-container>\n\n  <!--RESULT  PRODUCTS-->\n  <ng-container *cxFeature=\"searchBoxFeatures.RECENT_SEARCHES_FEATURE\">\n    <div class=\"message\" *ngIf=\"result.products?.length\">\n      {{ 'searchBox.products' | cxTranslate }}\n    </div>\n  </ng-container>\n  <ul\n    class=\"products\"\n    *ngIf=\"result.products\"\n    attr.aria-label=\"{{ 'searchBox.ariaLabelProducts' | cxTranslate }}\"\n    role=\"listbox\"\n  >\n    <li *ngFor=\"let product of result.products\">\n      <a\n        role=\"option\"\n        [routerLink]=\"\n          {\n            cxRoute: 'product',\n            params: product\n          } | cxUrl\n        \"\n        [class.has-media]=\"config.displayProductImages\"\n        (keydown.arrowup)=\"focusPreviousChild($any($event))\"\n        (keydown.arrowdown)=\"focusNextChild($any($event))\"\n        (keydown.enter)=\"close($any($event), true)\"\n        (keydown.escape)=\"close($any($event), true)\"\n        (blur)=\"close($any($event))\"\n        (mousedown)=\"preventDefault($event)\"\n        (click)=\"\n          dispatchProductEvent({\n            freeText: searchInput.value,\n            productCode: product.code\n          })\n        \"\n      >\n        <cx-media\n          *ngIf=\"config.displayProductImages\"\n          [container]=\"product.images?.PRIMARY\"\n          format=\"thumbnail\"\n          role=\"presentation\"\n        ></cx-media>\n        <div class=\"name\" [innerHTML]=\"product.nameHtml\"></div>\n        <span class=\"price\">{{ product.price?.formattedValue }}</span>\n      </a>\n    </li>\n  </ul>\n  <span id=\"initialDescription\" class=\"cx-visually-hidden\">\n    {{ 'searchBox.initialDescription' | cxTranslate }}\n  </span>\n  <div\n    *ngIf=\"result.suggestions?.length || result.products?.length\"\n    aria-live=\"assertive\"\n    class=\"cx-visually-hidden\"\n  >\n    {{\n      'searchBox.suggestionsResult'\n        | cxTranslate: { count: result.suggestions?.length }\n    }}\n    {{\n      'searchBox.productsResult'\n        | cxTranslate: { count: result.products?.length }\n    }}\n    {{ 'searchBox.initialDescription' | cxTranslate }}\n  </div>\n</div>\n" }]
        }], ctorParameters: () => [{ type: i1.SearchBoxComponentService }, { type: i2.CmsComponentData, decorators: [{
                    type: Optional
                }] }, { type: i3.WindowRef }, { type: i3.RoutingService }], propDecorators: { config: [{
                type: Input
            }], queryText: [{
                type: Input,
                args: ['queryText']
            }] } });
