// @spartacus/cart/esm2022/quick-order/root/quick-order-root.module.mjs
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { NgModule } from '@angular/core';
import { RouterModule } from '@angular/router';
import { ORDER_ENTRIES_CONTEXT } from '@spartacus/cart/base/root';
import { provideDefaultConfig, provideDefaultConfigFactory, } from '@spartacus/core';
import { CmsPageGuard, PageLayoutComponent } from '@spartacus/storefront';
import { defaultQuickOrderConfig } from './config/default-quick-order.config';
import { CART_QUICK_ORDER_CORE_FEATURE, CART_QUICK_ORDER_FEATURE, } from './feature-name';
import { QuickOrderOrderEntriesContextToken } from './tokens/context';
import * as i0 from "@angular/core";
import * as i1 from "@angular/router";
export function defaultQuickOrderComponentsConfig() {
    const config = {
        featureModules: {
            [CART_QUICK_ORDER_FEATURE]: {
                cmsComponents: ['QuickOrderComponent', 'CartQuickOrderFormComponent'],
            },
            // by default core is bundled together with components
            [CART_QUICK_ORDER_CORE_FEATURE]: CART_QUICK_ORDER_FEATURE,
        },
    };
    return config;
}
export const defaultQuickOrderRoutingConfig = {
    routing: {
        routes: {
            quickOrder: {
                paths: ['my-account/quick-order'],
            },
        },
    },
};
export class QuickOrderRootModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: QuickOrderRootModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "17.0.5", ngImport: i0, type: QuickOrderRootModule, imports: [i1.RouterModule] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: QuickOrderRootModule, providers: [
            provideDefaultConfigFactory(defaultQuickOrderComponentsConfig),
            provideDefaultConfig(defaultQuickOrderRoutingConfig),
            provideDefaultConfig(defaultQuickOrderConfig),
        ], imports: [RouterModule.forChild([
                {
                    // @ts-ignore
                    path: null,
                    canActivate: [CmsPageGuard],
                    component: PageLayoutComponent,
                    data: {
                        cxRoute: 'quickOrder',
                        cxContext: {
                            [ORDER_ENTRIES_CONTEXT]: QuickOrderOrderEntriesContextToken,
                        },
                    },
                },
            ])] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: QuickOrderRootModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        RouterModule.forChild([
                            {
                                // @ts-ignore
                                path: null,
                                canActivate: [CmsPageGuard],
                                component: PageLayoutComponent,
                                data: {
                                    cxRoute: 'quickOrder',
                                    cxContext: {
                                        [ORDER_ENTRIES_CONTEXT]: QuickOrderOrderEntriesContextToken,
                                    },
                                },
                            },
                        ]),
                    ],
                    providers: [
                        provideDefaultConfigFactory(defaultQuickOrderComponentsConfig),
                        provideDefaultConfig(defaultQuickOrderRoutingConfig),
                        provideDefaultConfig(defaultQuickOrderConfig),
                    ],
                }]
        }] });
