// @spartacus/storefront/esm2022/cms-components/navigation/navigation/navigation-ui.component.mjs
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { ChangeDetectionStrategy, Component, EventEmitter, HostBinding, HostListener, Input, Optional, } from '@angular/core';
import { NavigationEnd } from '@angular/router';
import { Subject, Subscription } from 'rxjs';
import { debounceTime, distinctUntilChanged, filter, take, } from 'rxjs/operators';
import { ICON_TYPE } from '../../misc/icon/index';
import * as i0 from "@angular/core";
import * as i1 from "@angular/router";
import * as i2 from "./../../../layout/header/hamburger-menu/hamburger-menu.service";
import * as i3 from "@spartacus/core";
import * as i4 from "@angular/common";
import * as i5 from "../../misc/icon/icon.component";
import * as i6 from "../../../shared/components/generic-link/generic-link.component";
export class NavigationUIComponent {
    onResize() {
        this.resize.next(undefined);
    }
    onArrow(e) {
        this.arrowControls.next(e);
    }
    constructor(router, renderer, elemRef, hamburgerMenuService, winRef, featureConfigService) {
        this.router = router;
        this.renderer = renderer;
        this.elemRef = elemRef;
        this.hamburgerMenuService = hamburgerMenuService;
        this.winRef = winRef;
        this.featureConfigService = featureConfigService;
        /**
         * the icon type that will be used for navigation nodes
         * with children.
         */
        this.iconType = ICON_TYPE;
        /**
         * Indicates whether the navigation should support flyout.
         * If flyout is set to true, the
         * nested child navigation nodes will only appear on hover or focus.
         */
        this.flyout = true;
        this.isOpen = false;
        this.openNodes = [];
        this.subscriptions = new Subscription();
        this.resize = new EventEmitter();
        this.arrowControls = new Subject();
        this.subscriptions.add(this.router.events
            .pipe(filter((event) => event instanceof NavigationEnd))
            .subscribe(() => this.clear()));
        this.subscriptions.add(this.resize.pipe(debounceTime(50)).subscribe(() => {
            this.alignWrappersToRightIfStickOut();
        }));
    }
    /**
     * During initialization of this component, we will check the resetMenuOnClose flag and attach a menu reset listener if needed.
     */
    ngOnInit() {
        if (this.resetMenuOnClose) {
            this.resetOnMenuCollapse();
        }
    }
    /**
     * This method performs the action of resetting the menu (close all sub menus and return to main options)
     * when the menu is closed.
     */
    resetOnMenuCollapse() {
        this.subscriptions.add(this.hamburgerMenuService?.isExpanded
            .pipe(distinctUntilChanged(), filter(Boolean))
            .subscribe(() => {
            this.reinitializeMenu();
        }));
    }
    closeIfClickedTheSameLink(navNode) {
        if (typeof navNode.url === 'string' &&
            this.winRef.nativeWindow?.location.href.includes(navNode.url)) {
            // TODO: (CXSPA-5919) Remove feature flag next major release
            if (this.featureConfigService?.isEnabled('a11yNavigationUiKeyboardControls')) {
                this.reinitializeMenu();
            }
            else {
                this.elemRef.nativeElement
                    .querySelectorAll('li.is-open:not(.back), li.is-opened')
                    .forEach((el) => {
                    this.renderer.removeClass(el, 'is-open');
                    this.renderer.removeClass(el, 'is-opened');
                });
                this.reinitializeMenu();
            }
            this.hamburgerMenuService.toggle();
        }
    }
    /**
     * This method performs the actions required to reset the state of the menu and reset any visual components.
     */
    reinitializeMenu() {
        if (this.openNodes?.length > 0) {
            // TODO: (CXSPA-5919) Remove feature flag next major release
            if (this.featureConfigService?.isEnabled('a11yNavigationUiKeyboardControls')) {
                this.elemRef.nativeElement
                    .querySelectorAll('li.is-open:not(.back), li.is-opened')
                    .forEach((el) => {
                    this.renderer.removeClass(el, 'is-open');
                    this.renderer.removeClass(el, 'is-opened');
                });
            }
            this.clear();
            if (!this.featureConfigService?.isEnabled('a11yNavigationUiKeyboardControls')) {
                this.renderer.removeClass(this.elemRef.nativeElement, 'is-open');
            }
        }
    }
    ariaCollapseNodes() {
        this.openNodes.forEach((parentNode) => {
            Array.from(parentNode.children)
                .filter((childNode) => childNode?.tagName === 'BUTTON')
                .forEach((childNode) => {
                this.renderer.setAttribute(childNode, 'aria-expanded', 'false');
            });
        });
    }
    toggleOpen(event) {
        if (event.type === 'keydown') {
            event.preventDefault();
        }
        this.ariaCollapseNodes();
        const node = event.currentTarget;
        const parentNode = node.parentNode;
        if (this.openNodes.includes(parentNode)) {
            if (event.type === 'keydown') {
                this.back();
            }
            else {
                this.openNodes = this.openNodes.filter((n) => n !== parentNode);
                this.renderer.removeClass(parentNode, 'is-open');
            }
        }
        else {
            this.openNodes.push(parentNode);
            this.renderer.setAttribute(node, 'aria-expanded', 'true');
        }
        this.updateClasses();
        event.stopImmediatePropagation();
        event.stopPropagation();
    }
    /**
     * Opens dropdown and starts keyboard navigation
     */
    onSpace(event) {
        this.hamburgerMenuService.isExpanded
            .pipe(take(1))
            .subscribe((isExpanded) => {
            if (isExpanded) {
                this.toggleOpen(event);
                return;
            }
            if (!this.openNodes.length) {
                this.toggleOpen(event);
                return;
            }
            event.preventDefault();
        });
        this.focusOnNode(event);
        this.setupArrowControls();
    }
    /**
     * Subscribes to arrow keys and enables navigation between dropdown items
     */
    setupArrowControls() {
        this.subscriptions.add(this.arrowControls.subscribe((e) => {
            e.preventDefault();
            const parentElement = e.target.parentElement
                ?.parentElement;
            const nextLink = parentElement?.nextElementSibling?.querySelector('a');
            const previousLink = parentElement?.previousElementSibling?.querySelector('a');
            e.code === 'ArrowDown' ? nextLink?.focus() : previousLink?.focus();
        }));
    }
    /**
     * Focuses on the first focusable element in the dropdown
     */
    focusOnNode(event) {
        const firstFocusableElement = event.target.nextElementSibling?.querySelector('button') ||
            event.target.nextElementSibling?.querySelector('a');
        firstFocusableElement?.focus();
    }
    back() {
        if (this.openNodes[this.openNodes.length - 1]) {
            this.renderer.removeClass(this.openNodes[this.openNodes.length - 1], 'is-open');
            this.openNodes.pop();
            this.updateClasses();
        }
    }
    clear() {
        this.openNodes = [];
        if (!this.featureConfigService?.isEnabled('a11yNavigationUiKeyboardControls')) {
            this.updateClasses();
        }
    }
    onMouseEnter(event) {
        this.alignWrapperToRightIfStickOut(event.currentTarget);
        this.focusAfterPreviousClicked(event);
    }
    getTotalDepth(node, depth = 0) {
        if (node.children && node.children.length > 0) {
            return Math.max(...node.children.map((n) => this.getTotalDepth(n, depth + 1)));
        }
        else {
            return depth;
        }
    }
    getColumnCount(length) {
        return Math.round(length / (this.wrapAfter || length));
    }
    focusAfterPreviousClicked(event) {
        const target = ((event.target || event.relatedTarget));
        if (target.ownerDocument.activeElement?.matches('nav[tabindex]') &&
            target.parentElement?.matches('.flyout')) {
            target.focus();
        }
        return target.ownerDocument;
    }
    ngOnDestroy() {
        if (this.subscriptions) {
            this.subscriptions.unsubscribe();
        }
    }
    alignWrapperToRightIfStickOut(node) {
        const wrapper = node.querySelector('.wrapper');
        const body = node.closest('body');
        if (wrapper) {
            this.renderer.removeStyle(wrapper, 'margin-left');
            if (wrapper.offsetLeft + wrapper.offsetWidth >
                body.offsetLeft + body.offsetWidth) {
                this.renderer.setStyle(wrapper, 'margin-left', `${node.offsetWidth - wrapper.offsetWidth}px`);
            }
        }
    }
    alignWrappersToRightIfStickOut() {
        const navs = this.elemRef.nativeElement.childNodes;
        Array.from(navs)
            .filter((node) => node.tagName === 'LI')
            .forEach((nav) => this.alignWrapperToRightIfStickOut(nav));
    }
    updateClasses() {
        this.openNodes.forEach((node, i) => {
            if (i + 1 < this.openNodes.length) {
                this.renderer.addClass(node, 'is-opened');
                this.renderer.removeClass(node, 'is-open');
            }
            else {
                this.renderer.removeClass(node, 'is-opened');
                this.renderer.addClass(node, 'is-open');
            }
        });
        this.isOpen = this.openNodes.length > 0;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: NavigationUIComponent, deps: [{ token: i1.Router }, { token: i0.Renderer2 }, { token: i0.ElementRef }, { token: i2.HamburgerMenuService }, { token: i3.WindowRef }, { token: i3.FeatureConfigService, optional: true }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "17.0.5", type: NavigationUIComponent, selector: "cx-navigation-ui", inputs: { node: "node", wrapAfter: "wrapAfter", resetMenuOnClose: "resetMenuOnClose", navAriaLabel: "navAriaLabel", flyout: "flyout", isOpen: "isOpen" }, host: { listeners: { "window:resize": "onResize()", "document:keyDown.arrowUp": "onArrow($event)", "document:keyDown.arrowDown": "onArrow($event)" }, properties: { "class.flyout": "this.flyout", "class.is-open": "this.isOpen" } }, ngImport: i0, template: "<nav [attr.aria-label]=\"navAriaLabel\">\n  <ul>\n    <li\n      *ngIf=\"flyout && (node?.children?.length ?? 0) > 1\"\n      class=\"back is-open\"\n    >\n      <button (click)=\"back()\">\n        <cx-icon [type]=\"iconType.CARET_LEFT\"></cx-icon>\n        {{ 'common.back' | cxTranslate }}\n      </button>\n    </li>\n\n    <ng-container *ngFor=\"let child of node?.children\">\n      <ng-container *ngTemplateOutlet=\"nav; context: { node: child, depth: 0 }\">\n      </ng-container>\n    </ng-container>\n  </ul>\n</nav>\n<!-- we generate links in a recursive manner -->\n\n<ng-template #nav let-node=\"node\" let-depth=\"depth\">\n  <li>\n    <ng-container *cxFeature=\"'!a11yNavigationUiKeyboardControls'\">\n      <cx-generic-link\n        *ngIf=\"\n          node.url && (!node.children || node.children?.length === 0);\n          else heading\n        \"\n        [url]=\"node.url\"\n        [target]=\"node.target\"\n        [style]=\"node.styleAttributes\"\n        [class]=\"node.styleClasses\"\n        (click)=\"closeIfClickedTheSameLink(node)\"\n      >\n        {{ node.title }}\n      </cx-generic-link>\n    </ng-container>\n    <ng-container *cxFeature=\"'a11yNavigationUiKeyboardControls'\">\n      <cx-generic-link\n        *ngIf=\"\n          node.url && (!node.children || node.children?.length === 0);\n          else heading\n        \"\n        [url]=\"node.url\"\n        [target]=\"node.target\"\n        [style]=\"node.styleAttributes\"\n        [class]=\"node.styleClasses\"\n        (click)=\"closeIfClickedTheSameLink(node)\"\n        [tabindex]=\"depth > 0 && !node.children ? -1 : 0\"\n        (focus)=\"depth || reinitializeMenu()\"\n        (keydown.space)=\"toggleOpen($any($event))\"\n      >\n        {{ node.title }}\n      </cx-generic-link>\n    </ng-container>\n\n    <ng-template #heading>\n      <ng-container *ngIf=\"flyout && node.children?.length > 0; else title\">\n        <cx-generic-link\n          *ngIf=\"node.url\"\n          [url]=\"node.url\"\n          [target]=\"node.target\"\n          (click)=\"closeIfClickedTheSameLink(node)\"\n        >\n          {{ node.title }}\n        </cx-generic-link>\n        <ng-container *cxFeature=\"'!a11yNavigationUiKeyboardControls'\">\n          <button\n            [attr.tabindex]=\"depth < 1 ? 0 : -1\"\n            [attr.aria-haspopup]=\"true\"\n            [attr.aria-expanded]=\"false\"\n            [attr.aria-label]=\"node.title\"\n            (click)=\"toggleOpen($any($event))\"\n            (mouseenter)=\"onMouseEnter($event)\"\n            (keydown.space)=\"toggleOpen($any($event))\"\n            (keydown.esc)=\"back()\"\n          >\n            <ng-container *ngIf=\"!node.url\">\n              {{ node.title }}\n            </ng-container>\n            <cx-icon [type]=\"iconType.CARET_DOWN\"></cx-icon>\n          </button>\n        </ng-container>\n        <ng-container *cxFeature=\"'a11yNavigationUiKeyboardControls'\">\n          <button\n            [attr.aria-haspopup]=\"true\"\n            [attr.aria-expanded]=\"false\"\n            [attr.aria-label]=\"node.title\"\n            (click)=\"toggleOpen($any($event))\"\n            (mouseenter)=\"onMouseEnter($event)\"\n            (keydown.space)=\"onSpace($any($event))\"\n            (keydown.esc)=\"back()\"\n            (keydown.arrowDown)=\"focusOnNode($any($event))\"\n            (focus)=\"depth || reinitializeMenu()\"\n          >\n            <ng-container *ngIf=\"!node.url\">\n              {{ node.title }}\n            </ng-container>\n            <cx-icon [type]=\"iconType.CARET_DOWN\"></cx-icon>\n          </button>\n        </ng-container>\n      </ng-container>\n      <ng-template #title>\n        <span *ngIf=\"node.title\" [attr.tabindex]=\"-1\">\n          {{ node.title }}\n        </span>\n      </ng-template>\n    </ng-template>\n\n    <!-- we add a wrapper to allow for better layout handling in CSS -->\n    <div class=\"wrapper\" *ngIf=\"node.children && node.children.length > 0\">\n      <ul\n        class=\"childs\"\n        [attr.depth]=\"getTotalDepth(node)\"\n        [attr.wrap-after]=\"node.children.length > wrapAfter ? wrapAfter : null\"\n        [attr.columns]=\"getColumnCount(node.children.length)\"\n      >\n        <ng-container *ngFor=\"let child of node.children\">\n          <ng-container\n            *ngTemplateOutlet=\"nav; context: { node: child, depth: depth + 1 }\"\n          >\n          </ng-container>\n        </ng-container>\n      </ul>\n    </div>\n  </li>\n</ng-template>\n", dependencies: [{ kind: "directive", type: i4.NgForOf, selector: "[ngFor][ngForOf]", inputs: ["ngForOf", "ngForTrackBy", "ngForTemplate"] }, { kind: "directive", type: i4.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { kind: "directive", type: i4.NgTemplateOutlet, selector: "[ngTemplateOutlet]", inputs: ["ngTemplateOutletContext", "ngTemplateOutlet", "ngTemplateOutletInjector"] }, { kind: "component", type: i5.IconComponent, selector: "cx-icon,[cxIcon]", inputs: ["cxIcon", "type"] }, { kind: "component", type: i6.GenericLinkComponent, selector: "cx-generic-link", inputs: ["url", "target", "id", "class", "style", "title", "ariaLabel", "tabindex"] }, { kind: "directive", type: i3.FeatureDirective, selector: "[cxFeature]", inputs: ["cxFeature"] }, { kind: "pipe", type: i3.TranslatePipe, name: "cxTranslate" }], changeDetection: i0.ChangeDetectionStrategy.OnPush }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: NavigationUIComponent, decorators: [{
            type: Component,
            args: [{ selector: 'cx-navigation-ui', changeDetection: ChangeDetectionStrategy.OnPush, template: "<nav [attr.aria-label]=\"navAriaLabel\">\n  <ul>\n    <li\n      *ngIf=\"flyout && (node?.children?.length ?? 0) > 1\"\n      class=\"back is-open\"\n    >\n      <button (click)=\"back()\">\n        <cx-icon [type]=\"iconType.CARET_LEFT\"></cx-icon>\n        {{ 'common.back' | cxTranslate }}\n      </button>\n    </li>\n\n    <ng-container *ngFor=\"let child of node?.children\">\n      <ng-container *ngTemplateOutlet=\"nav; context: { node: child, depth: 0 }\">\n      </ng-container>\n    </ng-container>\n  </ul>\n</nav>\n<!-- we generate links in a recursive manner -->\n\n<ng-template #nav let-node=\"node\" let-depth=\"depth\">\n  <li>\n    <ng-container *cxFeature=\"'!a11yNavigationUiKeyboardControls'\">\n      <cx-generic-link\n        *ngIf=\"\n          node.url && (!node.children || node.children?.length === 0);\n          else heading\n        \"\n        [url]=\"node.url\"\n        [target]=\"node.target\"\n        [style]=\"node.styleAttributes\"\n        [class]=\"node.styleClasses\"\n        (click)=\"closeIfClickedTheSameLink(node)\"\n      >\n        {{ node.title }}\n      </cx-generic-link>\n    </ng-container>\n    <ng-container *cxFeature=\"'a11yNavigationUiKeyboardControls'\">\n      <cx-generic-link\n        *ngIf=\"\n          node.url && (!node.children || node.children?.length === 0);\n          else heading\n        \"\n        [url]=\"node.url\"\n        [target]=\"node.target\"\n        [style]=\"node.styleAttributes\"\n        [class]=\"node.styleClasses\"\n        (click)=\"closeIfClickedTheSameLink(node)\"\n        [tabindex]=\"depth > 0 && !node.children ? -1 : 0\"\n        (focus)=\"depth || reinitializeMenu()\"\n        (keydown.space)=\"toggleOpen($any($event))\"\n      >\n        {{ node.title }}\n      </cx-generic-link>\n    </ng-container>\n\n    <ng-template #heading>\n      <ng-container *ngIf=\"flyout && node.children?.length > 0; else title\">\n        <cx-generic-link\n          *ngIf=\"node.url\"\n          [url]=\"node.url\"\n          [target]=\"node.target\"\n          (click)=\"closeIfClickedTheSameLink(node)\"\n        >\n          {{ node.title }}\n        </cx-generic-link>\n        <ng-container *cxFeature=\"'!a11yNavigationUiKeyboardControls'\">\n          <button\n            [attr.tabindex]=\"depth < 1 ? 0 : -1\"\n            [attr.aria-haspopup]=\"true\"\n            [attr.aria-expanded]=\"false\"\n            [attr.aria-label]=\"node.title\"\n            (click)=\"toggleOpen($any($event))\"\n            (mouseenter)=\"onMouseEnter($event)\"\n            (keydown.space)=\"toggleOpen($any($event))\"\n            (keydown.esc)=\"back()\"\n          >\n            <ng-container *ngIf=\"!node.url\">\n              {{ node.title }}\n            </ng-container>\n            <cx-icon [type]=\"iconType.CARET_DOWN\"></cx-icon>\n          </button>\n        </ng-container>\n        <ng-container *cxFeature=\"'a11yNavigationUiKeyboardControls'\">\n          <button\n            [attr.aria-haspopup]=\"true\"\n            [attr.aria-expanded]=\"false\"\n            [attr.aria-label]=\"node.title\"\n            (click)=\"toggleOpen($any($event))\"\n            (mouseenter)=\"onMouseEnter($event)\"\n            (keydown.space)=\"onSpace($any($event))\"\n            (keydown.esc)=\"back()\"\n            (keydown.arrowDown)=\"focusOnNode($any($event))\"\n            (focus)=\"depth || reinitializeMenu()\"\n          >\n            <ng-container *ngIf=\"!node.url\">\n              {{ node.title }}\n            </ng-container>\n            <cx-icon [type]=\"iconType.CARET_DOWN\"></cx-icon>\n          </button>\n        </ng-container>\n      </ng-container>\n      <ng-template #title>\n        <span *ngIf=\"node.title\" [attr.tabindex]=\"-1\">\n          {{ node.title }}\n        </span>\n      </ng-template>\n    </ng-template>\n\n    <!-- we add a wrapper to allow for better layout handling in CSS -->\n    <div class=\"wrapper\" *ngIf=\"node.children && node.children.length > 0\">\n      <ul\n        class=\"childs\"\n        [attr.depth]=\"getTotalDepth(node)\"\n        [attr.wrap-after]=\"node.children.length > wrapAfter ? wrapAfter : null\"\n        [attr.columns]=\"getColumnCount(node.children.length)\"\n      >\n        <ng-container *ngFor=\"let child of node.children\">\n          <ng-container\n            *ngTemplateOutlet=\"nav; context: { node: child, depth: depth + 1 }\"\n          >\n          </ng-container>\n        </ng-container>\n      </ul>\n    </div>\n  </li>\n</ng-template>\n" }]
        }], ctorParameters: () => [{ type: i1.Router }, { type: i0.Renderer2 }, { type: i0.ElementRef }, { type: i2.HamburgerMenuService }, { type: i3.WindowRef }, { type: i3.FeatureConfigService, decorators: [{
                    type: Optional
                }] }], propDecorators: { node: [{
                type: Input
            }], wrapAfter: [{
                type: Input
            }], resetMenuOnClose: [{
                type: Input
            }], navAriaLabel: [{
                type: Input
            }], flyout: [{
                type: Input
            }, {
                type: HostBinding,
                args: ['class.flyout']
            }], isOpen: [{
                type: Input
            }, {
                type: HostBinding,
                args: ['class.is-open']
            }], onResize: [{
                type: HostListener,
                args: ['window:resize']
            }], onArrow: [{
                type: HostListener,
                args: ['document:keyDown.arrowUp', ['$event']]
            }, {
                type: HostListener,
                args: ['document:keyDown.arrowDown', ['$event']]
            }] } });
