// @spartacus/product/esm2022/future-stock/occ/future-stock-occ.module.mjs
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { provideDefaultConfig } from '@spartacus/core';
import { FutureStockAdapter } from '@spartacus/product/future-stock/core';
import { OccFutureStockAdapter } from './adapters/occ-future-stock.adapter';
import { defaultOccFutureStockConfig } from './config/default-occ-future-stock.config';
import * as i0 from "@angular/core";
export class FutureStockOccModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: FutureStockOccModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "17.0.5", ngImport: i0, type: FutureStockOccModule, imports: [CommonModule] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: FutureStockOccModule, providers: [
            provideDefaultConfig(defaultOccFutureStockConfig),
            {
                provide: FutureStockAdapter,
                useClass: OccFutureStockAdapter,
            },
        ], imports: [CommonModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: FutureStockOccModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    providers: [
                        provideDefaultConfig(defaultOccFutureStockConfig),
                        {
                            provide: FutureStockAdapter,
                            useClass: OccFutureStockAdapter,
                        },
                    ],
                }]
        }] });
