// @spartacus/smartedit/esm2022/root/http-interceptors/cms-ticket.interceptor.mjs
import { inject, Injectable } from '@angular/core';
import { RoutingService } from '@spartacus/core';
import { switchMap, take } from 'rxjs/operators';
import * as i0 from "@angular/core";
import * as i1 from "../services/smart-edit-launcher.service";
export class CmsTicketInterceptor {
    constructor(service) {
        this.service = service;
        this.routingService = inject(RoutingService);
    }
    intercept(request, next) {
        const cmsTicketId = this.service.cmsTicketId;
        if (!cmsTicketId) {
            return next.handle(request);
        }
        if (request.url.includes('/productList')) {
            return this.setRequestForProductListPage(request, next, cmsTicketId);
        }
        if (request.url.includes('/cms/') || request.url.includes('/products/')) {
            request = request.clone({
                setParams: {
                    cmsTicketId,
                },
            });
        }
        return next.handle(request);
    }
    setRequestForProductListPage(request, next, cmsTicketId) {
        return this.routingService.getPageContext().pipe(take(1), switchMap((pageContext) => {
            request = request.clone(!!pageContext.id
                ? {
                    setParams: {
                        cmsTicketId,
                        categoryCode: pageContext.id,
                    },
                }
                : {
                    setParams: {
                        cmsTicketId,
                    },
                });
            return next.handle(request);
        }));
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: CmsTicketInterceptor, deps: [{ token: i1.SmartEditLauncherService }], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: CmsTicketInterceptor, providedIn: 'root' }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: CmsTicketInterceptor, decorators: [{
            type: Injectable,
            args: [{ providedIn: 'root' }]
        }], ctorParameters: () => [{ type: i1.SmartEditLauncherService }] });
