// @spartacus/storefront/esm2022/cms-components/navigation/scroll-to-top/scroll-to-top.module.mjs
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { I18nModule, provideDefaultConfig } from '@spartacus/core';
import { IconModule } from '../../misc/icon/icon.module';
import { ScrollToTopComponent } from './scroll-to-top.component';
import * as i0 from "@angular/core";
export class ScrollToTopModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: ScrollToTopModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "17.0.5", ngImport: i0, type: ScrollToTopModule, declarations: [ScrollToTopComponent], imports: [CommonModule, IconModule, I18nModule], exports: [ScrollToTopComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: ScrollToTopModule, providers: [
            provideDefaultConfig({
                cmsComponents: {
                    ScrollToTopComponent: {
                        component: ScrollToTopComponent,
                    },
                },
            }),
        ], imports: [CommonModule, IconModule, I18nModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: ScrollToTopModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, IconModule, I18nModule],
                    providers: [
                        provideDefaultConfig({
                            cmsComponents: {
                                ScrollToTopComponent: {
                                    component: ScrollToTopComponent,
                                },
                            },
                        }),
                    ],
                    declarations: [ScrollToTopComponent],
                    exports: [ScrollToTopComponent],
                }]
        }] });
