// @spartacus/storefront/esm2022/cms-components/navigation/scroll-to-top/scroll-to-top.component.mjs
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { ChangeDetectionStrategy, Component, HostBinding, HostListener, Optional, ViewChild, inject, } from '@angular/core';
import { FeatureConfigService, ScrollBehavior, } from '@spartacus/core';
import { take } from 'rxjs/operators';
import { ICON_TYPE } from '../../misc/icon/icon.model';
import * as i0 from "@angular/core";
import * as i1 from "@spartacus/core";
import * as i2 from "../../../cms-structure/page/model/cms-component-data";
import * as i3 from "../../../layout/a11y/index";
import * as i4 from "../../misc/icon/icon.component";
export class ScrollToTopComponent {
    constructor(winRef, componentData, selectFocusUtility) {
        this.winRef = winRef;
        this.componentData = componentData;
        this.selectFocusUtility = selectFocusUtility;
        this.iconTypes = ICON_TYPE;
        this.window = this.winRef.nativeWindow;
        this.scrollBehavior = ScrollBehavior.SMOOTH;
        this.displayThreshold = (this.window?.innerHeight ?? 400) / 2;
        this.triggedByKeypress = false;
        //TODO: (CXSPA-6522) - remove feature flag next major release.
        this.featureConfigService = inject(FeatureConfigService, {
            optional: true,
        });
    }
    ngOnInit() {
        this.setConfig();
    }
    onScroll() {
        this.switchDisplay();
    }
    /**
     * Scroll back to the top of the page, and recognize if triggerd by keyboard.
     */
    //TODO: (CXSPA-6522) - remove feature flag next major release.
    scrollToTop(event) {
        this.window?.scrollTo({
            top: 0,
            behavior: this.scrollBehavior,
        });
        if (this.featureConfigService?.isEnabled('a11yScrollToTop')) {
            this.triggedByKeypress = event?.detail === 0;
        }
        else {
            // Focus first focusable element within the html body
            this.selectFocusUtility
                .findFirstFocusable(this.winRef.document.body, { autofocus: '' })
                ?.focus();
        }
    }
    //TODO: (CXSPA-6522) - remove feature flag next major release.
    onFocusOut() {
        if (this.display &&
            this.featureConfigService?.isEnabled('a11yScrollToTop')) {
            this.switchDisplay();
        }
    }
    /**
     * After scrolling to top, pressing Tab should focus first focusable element within body.
     */
    //TODO: (CXSPA-6522) - remove feature flag next major release.
    onTab(event) {
        if (!this.featureConfigService?.isEnabled('a11yScrollToTop')) {
            return;
        }
        const tabEvent = event;
        const scrollToTopHasFocus = document.activeElement === this.button.nativeElement;
        const isAtTopOfPage = this.window?.scrollY === 0;
        if (scrollToTopHasFocus &&
            isAtTopOfPage &&
            tabEvent.key === 'Tab' &&
            !tabEvent.shiftKey) {
            event.preventDefault();
            this.selectFocusUtility
                .findFirstFocusable(this.winRef.document.body, { autofocus: '' })
                ?.focus();
        }
    }
    setConfig() {
        this.componentData.data$.pipe(take(1)).subscribe((data) => {
            this.scrollBehavior = data.scrollBehavior ?? this.scrollBehavior;
            this.displayThreshold = data.displayThreshold ?? this.displayThreshold;
        });
    }
    /**
     * This methods decides when the ScrollToTop button should be displayed.
     */
    switchDisplay() {
        const isPastThreshold = this.window && this.window.scrollY > this.displayThreshold;
        const buttonHasFocus = this.button.nativeElement === document.activeElement;
        this.display =
            isPastThreshold || (this.triggedByKeypress && buttonHasFocus);
        if (!this.display) {
            this.triggedByKeypress = false;
        }
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: ScrollToTopComponent, deps: [{ token: i1.WindowRef }, { token: i2.CmsComponentData }, { token: i3.SelectFocusUtility }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "17.0.5", type: ScrollToTopComponent, selector: "cx-scroll-to-top", host: { listeners: { "window:scroll": "onScroll($event)" }, properties: { "class.display": "this.display" } }, viewQueries: [{ propertyName: "button", first: true, predicate: ["button"], descendants: true }], ngImport: i0, template: "<button\n  #button\n  [attr.aria-label]=\"'navigation.scrollToTop' | cxTranslate\"\n  class=\"cx-scroll-to-top-btn\"\n  (click)=\"scrollToTop($event)\"\n  (focusout)=\"onFocusOut()\"\n  (keydown.Tab)=\"onTab($event)\"\n>\n  <span aria-hidden=\"true\">\n    <cx-icon class=\"caret-up-icon\" [type]=\"iconTypes.CARET_UP\"></cx-icon>\n  </span>\n</button>\n", dependencies: [{ kind: "component", type: i4.IconComponent, selector: "cx-icon,[cxIcon]", inputs: ["cxIcon", "type"] }, { kind: "pipe", type: i1.TranslatePipe, name: "cxTranslate" }], changeDetection: i0.ChangeDetectionStrategy.OnPush }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: ScrollToTopComponent, decorators: [{
            type: Component,
            args: [{ selector: 'cx-scroll-to-top', changeDetection: ChangeDetectionStrategy.OnPush, template: "<button\n  #button\n  [attr.aria-label]=\"'navigation.scrollToTop' | cxTranslate\"\n  class=\"cx-scroll-to-top-btn\"\n  (click)=\"scrollToTop($event)\"\n  (focusout)=\"onFocusOut()\"\n  (keydown.Tab)=\"onTab($event)\"\n>\n  <span aria-hidden=\"true\">\n    <cx-icon class=\"caret-up-icon\" [type]=\"iconTypes.CARET_UP\"></cx-icon>\n  </span>\n</button>\n" }]
        }], ctorParameters: () => [{ type: i1.WindowRef }, { type: i2.CmsComponentData }, { type: i3.SelectFocusUtility }], propDecorators: { display: [{
                type: HostBinding,
                args: ['class.display']
            }], button: [{
                type: ViewChild,
                args: ['button']
            }], featureConfigService: [{
                type: Optional
            }], onScroll: [{
                type: HostListener,
                args: ['window:scroll', ['$event']]
            }] } });
