// @spartacus/storefront/esm2022/cms-components/myaccount/my-account-v2/my-account-v2-navigation/my-account-v2-navigation.module.mjs
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { MyAccountV2NavigationComponent } from './my-account-v2-navigation.component';
import { AuthGuard, I18nModule, provideDefaultConfig, } from '@spartacus/core';
import { NavigationModule } from '../../../navigation/navigation/navigation.module';
import * as i0 from "@angular/core";
export class MyAccountV2NavigationModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: MyAccountV2NavigationModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "17.0.5", ngImport: i0, type: MyAccountV2NavigationModule, declarations: [MyAccountV2NavigationComponent], imports: [CommonModule, NavigationModule, I18nModule], exports: [MyAccountV2NavigationComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: MyAccountV2NavigationModule, providers: [
            provideDefaultConfig({
                cmsComponents: {
                    MyAccountSideNavigationComponent: {
                        component: MyAccountV2NavigationComponent,
                        guards: [AuthGuard],
                    },
                },
            }),
        ], imports: [CommonModule, NavigationModule, I18nModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: MyAccountV2NavigationModule, decorators: [{
            type: NgModule,
            args: [{
                    providers: [
                        provideDefaultConfig({
                            cmsComponents: {
                                MyAccountSideNavigationComponent: {
                                    component: MyAccountV2NavigationComponent,
                                    guards: [AuthGuard],
                                },
                            },
                        }),
                    ],
                    declarations: [MyAccountV2NavigationComponent],
                    exports: [MyAccountV2NavigationComponent],
                    imports: [CommonModule, NavigationModule, I18nModule],
                }]
        }] });
