// @spartacus/core/esm2022/src/occ/adapters/site-context/site-context-occ.module.mjs
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { CommonModule } from '@angular/common';
import { HTTP_INTERCEPTORS } from '@angular/common/http';
import { NgModule } from '@angular/core';
import { provideDefaultConfig } from '../../../config/config-providers';
import { BASE_SITE_NORMALIZER } from '../../../site-context/connectors/converters';
import { SiteAdapter } from '../../../site-context/connectors/site.adapter';
import { BaseSiteNormalizer } from './converters/base-site-normalizer';
import { defaultOccSiteContextConfig } from './default-occ-site-context-config';
import { OccSiteAdapter } from './occ-site.adapter';
import { SiteContextInterceptor } from './site-context.interceptor';
import * as i0 from "@angular/core";
export class SiteContextOccModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: SiteContextOccModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "17.0.5", ngImport: i0, type: SiteContextOccModule, imports: [CommonModule] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: SiteContextOccModule, providers: [
            provideDefaultConfig(defaultOccSiteContextConfig),
            {
                provide: SiteAdapter,
                useClass: OccSiteAdapter,
            },
            {
                provide: HTTP_INTERCEPTORS,
                useExisting: SiteContextInterceptor,
                multi: true,
            },
            {
                provide: BASE_SITE_NORMALIZER,
                useExisting: BaseSiteNormalizer,
                multi: true,
            },
        ], imports: [CommonModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: SiteContextOccModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    providers: [
                        provideDefaultConfig(defaultOccSiteContextConfig),
                        {
                            provide: SiteAdapter,
                            useClass: OccSiteAdapter,
                        },
                        {
                            provide: HTTP_INTERCEPTORS,
                            useExisting: SiteContextInterceptor,
                            multi: true,
                        },
                        {
                            provide: BASE_SITE_NORMALIZER,
                            useExisting: BaseSiteNormalizer,
                            multi: true,
                        },
                    ],
                }]
        }] });
