// @spartacus/storefront/esm2022/cms-components/content/pdf/pdf.module.mjs
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { I18nModule, provideDefaultConfig } from '@spartacus/core';
import { IconModule } from '../../misc/icon/icon.module';
import { PDFComponent } from './pdf.component';
import * as i0 from "@angular/core";
export class PDFModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: PDFModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "17.0.5", ngImport: i0, type: PDFModule, declarations: [PDFComponent], imports: [CommonModule, IconModule, I18nModule], exports: [PDFComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: PDFModule, providers: [
            provideDefaultConfig({
                cmsComponents: {
                    PDFDocumentComponent: {
                        component: PDFComponent,
                    },
                },
            }),
        ], imports: [CommonModule, IconModule, I18nModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: PDFModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, IconModule, I18nModule],
                    providers: [
                        provideDefaultConfig({
                            cmsComponents: {
                                PDFDocumentComponent: {
                                    component: PDFComponent,
                                },
                            },
                        }),
                    ],
                    declarations: [PDFComponent],
                    exports: [PDFComponent],
                }]
        }] });
