// @spartacus/storefront/esm2022/layout/theme/theme.module.mjs
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { APP_BOOTSTRAP_LISTENER, NgModule } from '@angular/core';
import { ThemeService } from './theme.service';
import * as i0 from "@angular/core";
export function initTheme(themeService) {
    const result = (component) => themeService.init(component);
    return result;
}
export class ThemeModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: ThemeModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "17.0.5", ngImport: i0, type: ThemeModule }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: ThemeModule, providers: [
            {
                provide: APP_BOOTSTRAP_LISTENER,
                multi: true,
                useFactory: initTheme,
                deps: [ThemeService],
            },
        ] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: ThemeModule, decorators: [{
            type: NgModule,
            args: [{
                    providers: [
                        {
                            provide: APP_BOOTSTRAP_LISTENER,
                            multi: true,
                            useFactory: initTheme,
                            deps: [ThemeService],
                        },
                    ],
                }]
        }] });
