// @spartacus/storefront/esm2022/cms-components/myaccount/my-account-v2/my-account-v2-consent-management/my-account-v2-consent-management.module.mjs
/*
 * SPDX-FileCopyrightText: 2023 SAP Spartacus team <spartacus-team@sap.com>
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { I18nModule } from '@spartacus/core';
import { SpinnerModule } from '../../../../shared/components/spinner/spinner.module';
import { IconModule } from '../../../misc/icon/icon.module';
import { MyAccountV2ConsentManagementComponent } from './components/my-account-v2-consent-management.component';
import { MyAccountV2ConsentManagementFormComponent } from './components/consent-form/my-account-v2-consent-management-form.component';
import { ConsentManagementComponentService } from '../../consent-management';
import * as i0 from "@angular/core";
export class MyAccountV2ConsentManagementModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: MyAccountV2ConsentManagementModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "17.0.5", ngImport: i0, type: MyAccountV2ConsentManagementModule, declarations: [MyAccountV2ConsentManagementComponent,
            MyAccountV2ConsentManagementFormComponent], imports: [CommonModule,
            FormsModule,
            ReactiveFormsModule,
            SpinnerModule,
            I18nModule,
            IconModule], exports: [MyAccountV2ConsentManagementComponent,
            MyAccountV2ConsentManagementFormComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: MyAccountV2ConsentManagementModule, providers: [ConsentManagementComponentService], imports: [CommonModule,
            FormsModule,
            ReactiveFormsModule,
            SpinnerModule,
            I18nModule,
            IconModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: MyAccountV2ConsentManagementModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        FormsModule,
                        ReactiveFormsModule,
                        SpinnerModule,
                        I18nModule,
                        IconModule,
                    ],
                    providers: [ConsentManagementComponentService],
                    declarations: [
                        MyAccountV2ConsentManagementComponent,
                        MyAccountV2ConsentManagementFormComponent,
                    ],
                    exports: [
                        MyAccountV2ConsentManagementComponent,
                        MyAccountV2ConsentManagementFormComponent,
                    ],
                }]
        }] });
