// @spartacus/user/esm2022/profile/components/reset-password/reset-password.module.mjs
/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { RouterModule } from '@angular/router';
import { GlobalMessageService, I18nModule, NotAuthGuard, provideDefaultConfig, RoutingService, } from '@spartacus/core';
import { FormErrorsModule, SpinnerModule, PasswordVisibilityToggleModule, } from '@spartacus/storefront';
import { UserPasswordFacade } from '@spartacus/user/profile/root';
import { ResetPasswordComponentService } from './reset-password-component.service';
import { ResetPasswordComponent } from './reset-password.component';
import * as i0 from "@angular/core";
export class ResetPasswordModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: ResetPasswordModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "17.0.5", ngImport: i0, type: ResetPasswordModule, declarations: [ResetPasswordComponent], imports: [CommonModule,
            FormsModule,
            ReactiveFormsModule,
            RouterModule,
            I18nModule,
            FormErrorsModule,
            SpinnerModule,
            PasswordVisibilityToggleModule] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: ResetPasswordModule, providers: [
            provideDefaultConfig({
                cmsComponents: {
                    ResetPasswordComponent: {
                        component: ResetPasswordComponent,
                        guards: [NotAuthGuard],
                        providers: [
                            {
                                provide: ResetPasswordComponentService,
                                useClass: ResetPasswordComponentService,
                                deps: [UserPasswordFacade, RoutingService, GlobalMessageService],
                            },
                        ],
                    },
                },
            }),
        ], imports: [CommonModule,
            FormsModule,
            ReactiveFormsModule,
            RouterModule,
            I18nModule,
            FormErrorsModule,
            SpinnerModule,
            PasswordVisibilityToggleModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.5", ngImport: i0, type: ResetPasswordModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        FormsModule,
                        ReactiveFormsModule,
                        RouterModule,
                        I18nModule,
                        FormErrorsModule,
                        SpinnerModule,
                        PasswordVisibilityToggleModule,
                    ],
                    providers: [
                        provideDefaultConfig({
                            cmsComponents: {
                                ResetPasswordComponent: {
                                    component: ResetPasswordComponent,
                                    guards: [NotAuthGuard],
                                    providers: [
                                        {
                                            provide: ResetPasswordComponentService,
                                            useClass: ResetPasswordComponentService,
                                            deps: [UserPasswordFacade, RoutingService, GlobalMessageService],
                                        },
                                    ],
                                },
                            },
                        }),
                    ],
                    declarations: [ResetPasswordComponent],
                }]
        }] });
